---
output:
  pdf_document:
    keep_tex: true
    includes:
      in_header: table_header.tex
tables: true
geometry: margin=0.75in
---

**Table 1. Summary of obesity, demographic, sequencing, and beta-diversity analysis data for the studies used in the meta-analysis.** NA indicates that those metadata were not available for that study.

```{r results='asis', echo=FALSE, warning=FALSE, eval=TRUE}
source('code/utilities.R')
get_dependencies('xtable')

options(xtable.sanitize.text.function=identity)


capwords <- function(s, strict = FALSE) {
		s <- as.character(s)
		cap <- function(s) paste(toupper(substring(s, 1, 1)),
									{s <- substring(s, 2); if(strict) tolower(s) else s},
														 sep = "", collapse = " " )
		sapply(strsplit(s, split = " "), cap, USE.NAMES = !is.null(names(s)))
}

make_study_label <- function(dataset){
	study <- ifelse(dataset=="hmp", "HMP", capwords(dataset))
	paste0(study, " (", citation[dataset], ")")
}

format_p <- function(p_value){
	char_p <- format(round(p_value, 3), nsmall=3L)

	if(p_value < 0.001){
		char_p <- "<0.001"
	}

	return(char_p)
}


region <- c(
	baxter = "V4",
	escobar = "V2",
	hmp = "V3-V5",
	ross = "V1-V3",
	schubert = "V3-V5",
	turnbaugh = "V2",
	wu = "V1-V2",
	zupancic = "V3-V5",
	goodrich = "V4",
	zeevi = "V3-V4"
)

technology <-  c(
	baxter = "MiSeq",
	escobar = "454",
	hmp = "454",
	ross = "454",
	schubert = "454",
	turnbaugh = "454",
	wu = "454",
	zupancic = "454",
	goodrich = "MiSeq",
	zeevi = "MiSeq"
)

citation <-  c(
	baxter = "23",#
	escobar = "16",#
	hmp = "21",#
	ross = "18",#
	schubert = "22",#
	turnbaugh = "6",#
	wu = "20",#
	zupancic = "15",#
	goodrich = "19",#
	zeevi = "17"#
)

get_study_summary <- function(study, beta = beta_summary){
	metadata_file <- paste0('data/', study, '/', study, '.metadata')
	metadata <- read.table(file=metadata_file, header=T, stringsAsFactors=FALSE, na.strings=c("NA", "NULL"))

	n_subjects <- nrow(metadata)

	per_obese <- format(round(100 * mean(metadata$obese, na.rm=T), 1), nsmall=1L)

	bmi <- "NA"
	if(sum(is.na(metadata$bmi)) != nrow(metadata)){
		mean_bmi <- format(signif(mean(metadata$bmi, na.rm=T), 3), nsmall=1L)
		min_bmi <- format(signif(min(metadata$bmi, na.rm=T), 3), nsmall=1L)
		max_bmi <- format(signif(max(metadata$bmi, na.rm=T), 3), nsmall=1L)
		bmi <- paste0(mean_bmi, " (", min_bmi, "-", max_bmi, ")")
	}

	per_female <- format(round(100 * mean(tolower(metadata$sex) == 'f'), 1), nsmall=1L)

	age <- "NA"
	if(sum(is.na(metadata$age)) != nrow(metadata)){
		mean_age <- format(signif(mean(metadata$age, na.rm=T), 3), nsmall=1L)
		min_age <- format(signif(min(metadata$age, na.rm=T), 3), nsmall=1L)
		max_age <- format(signif(max(metadata$age, na.rm=T), 3), nsmall=1L)
		age <- paste0(mean_age, " (", min_age, "-", max_age, ")")
	}

	per_white <- format(round(100*mean(metadata$white), 1), nsmall=1L)

	list(study = make_study_label(study),
				n_subjects = as.character(n_subjects),
				per_obese = per_obese,
				bmi_summary = bmi,
				per_female = per_female,
				age_summary = age,
				per_white = per_white,
				method = technology[study],
				region = region[study],
				beta_p = format_p(beta_summary[study,'p_value']))
}

beta_summary <- read.table("data/process/beta_tests.summary",
															header=T, row.names = 1)
datasets <- sort(rownames(beta_summary))

study_summary <- t(sapply(datasets, get_study_summary))

#colnames(study_summary) <- LETTERS[1:8]
colnames(study_summary) <- c("Study (Ref.)", "\\bigcell{c}{Subjects \\\\ (N)}",
													"\\bigcell{c}{Obese \\\\ (\\%)}",
													"\\bigcell{c}{Average BMI \\\\ (Min-Max)}",
													"\\bigcell{c}{Female \\\\ (\\%)}",
													"\\bigcell{c}{Average Age \\\\ (Min-Max)}",
													"\\bigcell{c}{Non-Hispanic \\\\ White (\\%)}",
													"\\bigcell{c}{Sequencing \\\\ Method}",
													"\\bigcell{c}{16S rRNA \\\\ Gene Region}",
													"\\bigcell{c}{AMOVA \\\\ (P-value)}")

colnames(study_summary) <- paste0("\\textbf{", colnames(study_summary), "}")

composite <- xtable(study_summary, label="")

align(composite) <- "llccccccccc"

print(composite,
		include.rownames=FALSE,
		include.colnames=TRUE,
		hline.after = c(-1,0, nrow(composite)),
		comment=FALSE,
		size="scriptsize",
		table.placement="!ht"
		)
```
