---
title: "Looking for a Signal in the Noise: Revisiting Obesity and the Bacterial Microbiome"
output:
  html_document:
    keep_md: yes
  word_document:
    reference_docx: manuscript_template.docx
bibliography: metaAnalysis.obesity.microbiome.bib
---

Marc A Sze<sup>1</sup> and Patrick D Schloss<sup>1</sup>


<sup>1</sup> Department of Microbiology and Immunology, University of Michigan, Ann Arbor, MI, USA  
<br>
<br>
<br>
<br>
<br>
<br>
Corresponding author: Patrick D Schloss  
*Current Address:*  
1526 MSRB I  
Ann Arbor, MI  
USA 48109-5620  
Email: pschloss@umich.edu  
Phone: 734-936-2951    
<br>
<br>
**Contributions:** MAS (planned, designed, and performed experiments, completed data analysis, wrote first draft), PDS (planned and designed the study, data interpretation, and edited subsequent drafts).  
<br>
**Running Title:** Bacterial Microbiome and Obesity
<br>





*******
##### **Abstract:**
**Rationale:** The hypothesis that changes in the microbiome and obesity are associated has been widely accepted.  However, two recent reports that re-examined existing data suggest that acceptance of this hypothesis is unwarranted.  These studies, however, studied the original studies in isolation from each other. We hypothesized that by statistically pooling the individual results, the statistical power to identify a significant association would increase.  In this meta analysis, we tested the significance of the association between the microbiome and obesity by increasing the number of data sets examined and investigating the effect of pooling the results across the individual studies. 

**Methods:** We identified 196 studies that tested the microbiome-obesity association hypothesis; however when we limited the studies to those that did microbiome sequencing, had publicly available sequences, and were either willing to share the BMI information for every individual or were already published, a total of 9 data sets were identified. We applied three analysis piplines to the data from each study. First, we tested the association between the community diversity, Bacteroides to Firmicutes ratio (B/F), and relative risk (RR) with the subjects' body mass index (BMI) and obesity category.  We also used random forest to identify variables for classification of obese and non-obese indivdiuals.  Second, we pooled the relative risk (RR) ratio of being obese based on characteristics of the microbiome for each studies.  Finally, data normalization was used to investigate all data sets together for relationships between the microbiome and obesity.    

**Results:** There were no significant differences (P > 0.05) between the relative abundance of Bacteroidetes, Firmicutes, or thier ratio in non-obese and obese individuals. The Shannon diversity, OTU richness, relative risk, or oveall community dissimilarity was significant in at least one study (P<0.05).  Overall, the random forest results for Area Under the Curve (AUC) was 0.7553 +/- 0.0888 and 7/9 studies contained an OTU that could be classified to the Ruminococcaceae family.  When we pooled the results across studies, there was a signifcant difference in RR of obesity based on Shannon diversity (RR = 1.19, CI = 1.04, 1.36, (P-value = 0.0139)) and B/F ratio (RR = 0.86, CI = 0.75, 0.99, (P-value = 0.0296)).  

**Conclusion:** We show that by pooling the results of many indivdiual studies we can test the generalizability of the bacterial microbiome.  Our meta-analysis supports the  hypothesis that there is a difference between non-obese and obese individuals in Shannon diversity and B/F ratio, based on the currently avialable bacterial microbiome data included in this meta-analysis.  
  		   





*******
##### **Introduction:**  
Obesity is a growing worldwide health concern with approximately 20% of the youth (aged 2-19) in the United States classified as either overweight or obese [@ogden_prevalence_2014].  This number increases to approximately 35% in adults (aged 20 or older) and these statisitics have seen little change since 2003 [@ogden_prevalence_2014].  Traditionally the body mass index (BMI) has been used as the traditional method of classifying individuals as non-obese or obese [@lichtash_body_2013].  However, other methods do exist, such as body adiposity index (BAI) but do not always correlate with important molecular measures traditionally associated with obesity [@lichtash_body_2013].  Recently, there has been a lot of interest in the bacteiral microbiome and it's potential ability to modulate obesity [@brahe_can_2016; @dror_microbiota_2016].  If this modulation potential is true then the bacterial microbiome could have an enormous impact and role to play in the future treatment of obesity and in helping to stem the rising obesity epidemic.    

There has been a continuous stream of studies that report that there is a link between the bacterial microbiome and obesity.  The first research article to solidify the bacterial microbiomes' role in obesity was the landmark study by Turnbaugh et al. [-@turnbaugh_obesity-associated_2006] which provided direct evidence that by modulating the microbiome one could have an effect on the obesity phenotype of mice.  This was quickly followed up by human studies that found observable differences in the bacterial microbiome between non-obese and obese individuals [@ley_microbial_2006; @turnbaugh_core_2009].  Since the publication of these studies there have been a steady stream of publications on the bacterial microbiome with associated BMI data but without clear explicit mention of any significant correlations between the bacterial community and obesity [@ross_16s_2015; @zupancic_analysis_2012; @nam_comparative_2011; arumugam_enterotypes_2011; @goodrich_human_2014; @yatsunenko_human_2012; @wu_linking_2011; @escobar_gut_2014].  However, two recent publications [@walters_meta-analyses_2014; @finucane_taxonomic_2014] re-analyzed some of the previously publishd data and came to the shocking conclusion that the human bacterial microbiome did not correlate with obesity.  One common critique of these previous studies was that they never looked at the data pooled together and that this could result in a significant result where each individual study alone would not.          

Although there have been numerous RT-PCR and qPCR studies directed at examining the bacterial microbiome and obesity [@guo_development_2008; @schwiertz_microbiota_2010].  The main method of choice to obtain the most complete picture of the overall microbial community is through 16S sequencing of a specific hypervariable region on the 16S rRNA gene.  The sequencers used to complete this task have varied depending on research group with the most popular being the Roche 454 and Illumina MiSeq.  A priori we decided that for this report we would limit ourselves to analyzing studies were 16S sequencing data is avaliable along with BMI information.  We also analyzed every study with a uniform approach utilizing the mothur software package for sequence processing and clean up and R statistical software for analysis.  

The purpose of this study is to perform an extensive meta-analysis of the bacterial microbiome and obesity by analyzing and applying a more sytematic approach than has been used by previous studies in this area.  Additionaly, we will include both seperate and pooled analysis to assess the bacterial microbiome and obesity. By including these two components we will build on the previous two re-analysis studies and provide a stronger assesment of the direction of effect or lack there of on the bacterial microbiome and obesity.  In particular, our goal is to provide additional clarity on the role of both bacterial diversity and the Bacteroidetes/Firmicutes ratio impact on obesity.


*******
##### **Methods:**  
*Literature Review and Study Inclusion*  

We attempted to stick to the original PRISMA guidelines for performing a meta-analysis as published by Moher et al [-@moher_preferred_2010]. A detailed description of the literature seach and the exact formula we used for the initial PubMed search can be found in the online supplement.  In brief, a total of 196 studies made it through the initial identification of studies.  Screening removed a total of 184 studies leaving a total of 12 eligible studies for this meta-analysis.  Upon a full reading of these specific manuscripts 2 were exculded and the reasons can be found in the online supplement.  A total of 10 studies were ultimately deemed eligible for qualitative synthesis [Figure 1].  One final study was removed from the quantitative analysis since no individual in the study was found to be able to be classified in the obese category based on the classical cutoffs. A total of 3 studies were from records identified by database searching [@turnbaugh_core_2009; @zupancic_analysis_2012; @escobar_gut_2014] and the other 6 were from records identified by other sources [@ross_16s_2015; @arumugam_enterotypes_2011; @goodrich_human_2014; @wu_linking_2011; @conlan_species-level_2012].  One of these studies (Baxter) has not been published at the time of this writing.  

*Subject Demographics and Study Outcomes*  

The overall demographics for individuals used in this meta-analysis split by their respective study can be found in table 1.  A summary of the study type, study population, whether the data set is published, and the outcomes reported can be found in table 2.  We found a large variation in reported outcomes with two studies finding bacterial community associations with obesity and three studies finding some association between either Firmicutes, Bacteroidetes, or the Bacteroidetes/Firmicutes ratio and obesity.  However, a total of three manuscripts found no correlations between the bacterial microbiome and obesity.  This is in addition to the two re-analysis studies published in 2014.  The cutoffs we used in this study for the BMI groups were lean (BMI less than or equal to 24), overweight (BMI was greater than 24 but less than 30), and obese (BMI greater than or equal to 30).  For analysis involving obese and non-obese groups the cutoffs we used were a BMI less than 30 (lean/non-obese) and a BMI of equal to or greater than 30 (obese).
  
*Sequence Analysis Pipeline*  

All data was publically available and the sequence data was downloaded from the NCBI Sequence Read Archive, MG-RAST, or the European Nucleotide Archive.  A total of 6/9 [@turnbaugh_core_2009; @ross_16s_2015; @zupancic_analysis_2012; @wu_linking_2011; @escobar_gut_2014; @conlan_species-level_2012] studies were processed by staying as close as possible to the mothur 454 standard operating procedure, which can be found at http://www.mothur.org/wiki/454_SOP.  A total of 2/9 [@goodrich_human_2014] of the studies were processed by staying as close as possible to the mothur MiSeq standard operating procedure found at http://www.mothur.org/wiki/MiSeq_SOP.  The final data set [@arumugam_enterotypes_2011] that was used took advantage of the Metaphlan2 package [@truong_metaphlan2_2015] to generate the necessary data for this analysis.  A comparison of the similarity of Metaphlan2 to 16S sequencing using the Human Microbiome Project (HMP) data set can be found in the online supplement [Figure S1-S3]. For the studies that utilized 16S rRNA gene sequencing three studies used primers targeting the V1-V3 region [@ross_16s_2015; @escobar_gut_2014; @zupancic_analysis_2012], two studies used the V4 region [@goodrich_human_2014], and one study used the V1-V2 region [@wu_linking_2011].  Where there were two choices of data sets to use we used the V3-V5 data of the HMP [@conlan_species-level_2012] and the V2 data from the Turnbaugh study [-@turnbaugh_core_2009].  A detailed walkthrough of the entire analysis pipeline can be found at https://github.com/SchlossLab/Sze_ObesityMicrobiomeMeta_PeerJ_2016/.

*Data Analysis*  

We performed the analysis using R version 3.2.2 and R studio version 0.99.491.  We split the overall analysis into three general categories.  The first category involved following the same method employed by Finucane et al. and Walters et al. [-@finucane_taxonomic_2014; -@walters_meta-analyses_2014] in which each study was re-analyzed seperately for associations with BMI.  The main associations that we explored for BMI in each seperate data set were the Bacteroidetes phyla, Firmicutes phyla, Bacteroidetes/Firmicutes (B/F) ratio, Shannon diversity, OTU richness, NMDS visualization of Bray-Curtis distance matrix PERMANOVA, and relative risk (RR).  A two-way T-test was performed for comparison for non-obese versus obese.  Finally, Random forest analysis to identify the best variables for classification of obese and non-obese was performed to look if there were similarities between data sets that we may have missed.  The R package AUCRF (v1.1) was used for this.  


For the second category of analysis we pooled all the data from each seperate study together to asses whether the combined direction of all the studies increased an individual's RR for being obese.  We investigated both Shannon diversity and the B/F ratio in detail for this part.  In order to generate the RR we used the median value of each study and the number of those that were non-obese or obese in each group (above or below the median respectively) was recorded and a Fisher exact-test was performed.  The relative risk from the the Fisher exact-test was obtained using the epiR (v0.9-6.9) package in R.  We obtained the pooled RR and confidence intervals (CI) from the metafor (v1.9-8) package in R.  Finally, we also assessed the potential bias in the two pooled analysis by using a funnel plot.  


The third and final category involved using  Zscore normalizations to pool all the data together to investigate the correlation between either BMI and Shannon diversity or BMI and the B/F ratio.  We used a two-way t-test on the Zscore values to investigate if there was a difference between non-obese and obese individuals.  In order to asses differences between the BMI categories (if data was available) we used ANOVA with a tukey post hoc test.  Where appropriate we applied a Bonferroni correction for multiple comparisons.  In all other cases a P-value under 0.05 was considered to be significant.


*******
##### Results:

*Seperate Analysis:*

We used the categorical variable of obese or non-obese to analyze each data set seperately for significant correlations to Bacteroidetes, Firmicutes, Bacteroidetes/Firmicute (B/F) ratio, Shannon diversity, OTU richness, evenness, and the Bray-Curtis distance matrix which was a similar approach taken by two previous reviews [@walters_meta-analyses_2014; @finucane_taxonomic_2014].  Using a P-value cutoff for significance as less than 0.05 we found a total of 1 significant result for Shannon diversity, 2 significant results for OTU richness, and 1 signficant result for the Bray-Curtis distance matrix.  There were no significant results for Bacteroidetes, Firmicutes, or the B/F ratio.  The full summary of the exact P-values for each respective study and variable tested can be found in table 3.

Next we investigated how well the OTUs in each study were able to classify obese and non-obese individuals using the random forest algorithm.  We found that the average Out Of Bag (OOB) Area Under the Curve (AUC) was 0.7553 +/- 0.0855.  We found that the total range for the total number of variables used for the classification ranged from 3 (Ross and Escobar) to 40 (Zupancic).  The total number of trees and node size was standardized to 1000 and 20 respectively for each data set.  One aspect we found surprising was that there was very little overlap in the variables used by the random forest algorithm for the classification of obese and non-obese individuals by study.  The three most common variables was for an OTU that could be classified to the family Ruminococcaceae (7/9 studies), an OTU that could be classified to the family Lachnospiraceae (5/9 studies), and the Shannon diversity metric (4/9 studies).   


*Classical Pooled Meta-analysis:*

We next moved on to the pooled meta-analysis.  For the Shannon diversity RR, all studies but Wu et al had those with a lower Shannon diversity then the median as being higher risk for obesity [Figure 2].  This was confirmed with the pooled analysis showing an increased RR of 1.18 (CI 1.03, 1.36) for obesity in the lower then median Shannon diversity group (P-value = 0.0187).  When investigated the B/F ratio RR most studies showed that a low B/F ratio tended to have less obese individuals than those with a high B/F ratio [Figure 3].  A notable exception to this was the HMP data set and the Baxter data set [Figure 3].  The pooled RR was 0.86 (CI 0.75, 0.99) suggesting that a low B/F ratio does indeed have less obese individuals (P-value = 0.0376).

We also used a funnel plot to asses the bias of the 9 data sets used in this meta-analysis.  From the graph we can observe that there does not seem to be any significant  bias [Figure 4].  Both funnel plots have almost equal number of studies scattered on either side of the estimated true value.

The final forest plot that we used for this analysis investigated how the different OTUs, classified to Ruminococcaceae, that were important for the classification of obese and non-obese based on the OOB AUC in the the seperate study analysis looked when pooled together to assess the RR [Figure S4].  There was no significant increase in RR based on the Ruminococcaceae OTU (RR = 0.96 (CI 0.60, 1.52), P-value = 0.8575).  Thus it would seem that although Ruminococcaceae is important for classification the directionality is study dependent [Figure S4].    

*ZScore Normalization Pooled Analysis:*

Using the third approach with Zscore normalization we found that there was a significant difference in the Shannon diversity between the obese and non-obese group (P-value = 0.00493) [Figure 5A].  Further there was also a significant difference in this normalized Shannon diversity value when the data was split between the three BMI groups (P-value = 0.03826) [Figure 5B].  Using a Tukey post hoc test this difference was driven by the overweight versus the obese group (P-value = 0.03945).  However, in contrast to this result we found that the B/F ratio, using the ZScore normalization, was not significant different between obese and non-obese indivdiuals (P-value = 0.4516) [Figure 5C].  Additionally, we did not observe a difference when the indivdiuals were seperated by BMI groups (P-value = 0.3317) [Figure 5D].


*******
##### Discussion:

Our meta-analysis helps to provide a little more clarity to the on going debate of whether or not there are specific bacterial microbiome variables that are influenced by obesity.  Two previous reviews [@walters_meta-analyses_2014; @finucane_taxonomic_2014] have stated that the data do not support that there is a bacterial microbiome difference between non-obese and obese individuals.  However, neither of these studies made an attempt to pool the existing data together to try and harness the additional power that this would give.  Here we perform an extensive literature review of the existing studies on the bacterial microbiome and obesity and perform a meta-analysis on the studies that remained based on our inclusion and exclusion criteria.  We analyze the data using three different approaches.  The first approach uses the method taken by the previous reviews [@walters_meta-analyses_2014; @finucane_taxonomic_2014], the second approach takes a meta-analysis approach using RR representation by forest plots and funnel plots, the final approach also takes a meta-analysis approach but uses ZScore normalizations and does not use RR.

Overall, the first approach in this study by and large agrees with the prevous two reviews[@walters_meta-analyses_2014; @finucane_taxonomic_2014], in that for the majority of studies, no significant correlation between the bacterial microbiome and obestiy could be observed.  However, we found that the other two approaches, that used the pooled data, was able to identify a correlation between measures of the bacterial microbiome and obesity [Figure 2 & 3].  For Shannon diversity both approaches using the pooled data found a correlation while only one of the approaches for the B/F ratio found a correlation.  Taken together we believe that this analysis provides a strong preliminary finding that there is a small but detectable difference between non-obese and obese individuals with respect to both Shannon diversity and the B/F ratio.  However, it should be noted that our analysis would indicate that the signal for Shannon diversity may be more robust than that of the B/F ratio since the two different pooled analysis approaches found Shannon diversity significant while only one did for the B/F ratio.

We do acknowledge that this study does have a few limitations that are worth mentioning.  First,  it has been documented that in a number of groups BMI is not a good classification of obesity [@who_expert_consultation_appropriate_2004; @rahman_accuracy_2010].  Of particular note, the asian population has a seperate set of guidelines for obesity [@who_expert_consultation_appropriate_2004].  Although the asain population in the studies analyzed are very small [Table 1] it is possible that this may contribute to some of the noise in the data.  However, it is important to keep in mind that for these measures, specifically the RR, there does not seem to be any study bias due to studies having more ethnic diversity [Figure 4].  Additionally, we are also limited in our BMI classifications by what the previous studies published and performed.  In general, not enough studies have been done to investigate the bacterial microbiome in the context of other markers for obesity or on different ethnicities to be able to perform the analysis that we did.  Thus as more specific studies are published that address ethnicity and studies on the microbiome that use better markers of obesity are completed it is entirely possible that the noise associated with these measurements will decrease.         

Second, although we make attempts to normalize the data for the pooled analysis so that we can compare the bacterial microbiome across studies it is possible that the different variable regions sampled, sequencing machines used, and the quality of the output data could have an impact on the overall conclusions reached.  However, what our analysis shows is that despite these differences, we are still able to identify an overall signal from the data.    
Third, this meta-analysis and review of the data to date is still rather preliminary.  Only a total of 1542 individuals were included in the pooled analysis which is relatively small compared to other meta-analysis in non-microbiome fields of study.  It will be necessary to continuously update this analysis, as more studies become available, in order to obtain a more accurate estimate of the true effect that obesity has on the bacterial microbiome within humans.   

Fourth, although a significant result exists for the RR for both Shannon diversity and the B/F ratio, it is small.  This is also the case for the ZScore normalization analysis of the Shannon diversity.  As an example, a lower than median Shannon diversity only has an overall 18% increase of obesity in it versus the higher than median Shannon diversity group.  So even though there is a detectable signal it remains to be seen if this is a biologically significant phenomenon that could have a tangible effect on obesity. 

Despite the many problems that could add more noise to the measurements used, this study provides a reasonable first pass attempt to pool this data together in an effort to find a correlation between the bacterial microbiome and obesity.  In doing so we believe that it provides a tanatalizing result that despite the large differences between studies it is still possible to detect a correlation between the bacterial microbiome and obesity.  However, we emphaszie that it should still be viewed with caution since there are still many aspects of this analysis that could be improved upon.  It truly is only a first attempt at trying to summerize and analyze the pooled data that is available.  We hope that this study helps to add important information and clarity to the on-going dialogue of the bacterial microbiome's role in obesity.



*******
##### Acknowledgements:

The authors would like to thank Nielson Baxter for his help and suggestions during the sequence analysis portion.  We would also like to thank Shawn Whitefield for her help with making sure the manuscript adhered to the guidelines set out for a meta-analysis.  Finally, we would like to thank Rick Bushman, Hongzhe Li, and Pixu Shi in agreeing to release and helping to get a hold of the subject specific metadata from the Wu et al study.  


*******
##### Tables:

**Table 1:Summary Demographics of Individuals used in the Meta-analysis.**     
```{r Table_1, echo=FALSE, message=FALSE, warning=FALSE}
library(knitr)

Study <- c("Baxter", "Ross", "Goodrich", "Escobar", "Zupancic*", "HMP", "Wu", "Arumugam", "Turnbaugh**")

Age <- c("54.3 +/- 9.9", "57.0 +/- 11.2", "61.5 +/- 8.9", "38.1 +/- 11.1", "48.2 +/- 13.2", "26.1 +/- 5.0", "26.4 +/- 9.4", "56.5 +/- 7.7", "21 - 32")

Sex <- c("111|61", "48|15", "505|2", "14|16", "198|112", "127|129", "31|27", "45|40", "Not Available")

Ancestry <- c(87.8, 0, "Not Available", 0, 100, 83, "Not Available", "Not Available", 51.4)

BMI <- c("27.0 +/- 5.3", "31.6 +/- 5.3", "26.3 +/- 4.9", "27.4 +/- 4.5", "29.2 +/- 5.2", "24.1 +/- 3.4", "24.6 +/- 4.8", "27.8 +/- 5.8", "Not Available")

Min <- c(17.5, 22.1, 16.2, 19.5, 16.7, 19, 14.0, 18.6, "Not Available")

Max <- c(46.9, 47.9, 44.8, 37.6, 51.1, 34, 41.3, 40.2, "Not Available")

table1 <- cbind(Study, Age, Sex, Ancestry, BMI, Min, Max)

colnames(table1) <- c("Study", "Age (Mean +/- SD)", "Sex (F|M)", "European Ancestry (%)", "BMI (Mean +/- SD)", "Min BMI","Max BMI")

kable(table1, format = "markdown", align = 'c', padding = 2)


```

*If numerical BMI information was not available then ranges and demographic information was taken from the authors published manuscript for this table.

**Only BMI group information was provided and only an age range was given.


**Table 2:Summary of Important Study Characteristics** 

```{r Table_2, echo=FALSE, message=FALSE, warning=FALSE}

Study <- c("Baxter", "Ross", "Goodrich", "Escobar", "Zupancic", "HMP", "Wu", "Arumugam", "Turnbaugh")

Type <- c("Observational", "Observational", "Observational", "Observational", "Observational", "Observational", "Observational", "Observational", "Observational")

Population <- c("Adults without Cancer from Texas, Ontario, Michigan, Massachusetts", "Hispanic adults with and without type 2 diabetes from Texas
", "Adult twins and mothers from the United Kingdom", "Healthy adults from Columbia", "Amish Adults with and without metabolic syndrome from Pennsylvania", "Healthy Adults from Texas and Missouri", "Healthy Adults from Pennsylvania", "Healthy Adults from Denmark", "Adult twins and mothers from Missouri")

Published <- c("No", "Yes", "Yes", "Yes", "Yes", "Yes", "Yes", "Yes","Yes")

Outcomes <- c("N/A (Unplublished)", "Mentions obesity, no specific correlations made though", "Christensenellaceae higher in low BMI individuals. No other correlations found.", "Firmicutes less abundant at higher BMI", "Bacteroidetes/Firmicutes ratio adjusted BMI correlation", "No significant correlations", "Unweighted UniFrac Correlation with BMI", "No significant  correlations", "Decrease diversity, Bacteroidetes, and Actinobacteria correlated with increased BMI ")

table2 <- cbind(Study, Type, Population, Published, Outcomes)

colnames(table2) <- c("Study", "Study Type", "Population", "Published", "Outcomes Reported")


kable(table2, format = "markdown", align = 'c', padding = 2)
```


**Table 3:Summary of P-values for Measurements of Interest for each Individual Study for Obese versus Normal** 

```{r Data_Baxter, echo=FALSE, message=FALSE, warning=FALSE}
###########################################################################
############ Preparing Data Tables for Analysis ###########################
###########################################################################

# Reading in the necessary Data
setwd("C:/Users/marc/Desktop/obesity2/data1")
demographics <- read.csv("demographics.v2.csv")
microbiome <- read.csv("data1.subsample.otus.csv")
phylogenetic.info <- read.csv("data1.summary.taxonomy.csv")
taxonomy <- read.csv("data1.taxonomy.csv")

# Minor modifications and Rownames adustments of data tables
rownames(demographics) <- demographics[,1]
rownames(microbiome) <- microbiome[,2]
demographics <- demographics[,-1]
microbiome <- microbiome[,-2]
test.samples <- rownames(demographics)
microbiome <- microbiome[,-c(1:2)]

# Create Table for phyla information
phylogenetic.info <- phylogenetic.info[-c(1:12),]
phyla <- which(phylogenetic.info$taxlevel == 2)
phyla.table <- phylogenetic.info[phyla, ]
phyla.table <- phyla.table[,-c(1:2,4:5)]
phyla.names <- as.character(phyla.table[,1])
phyla.table <- phyla.table[,-1]
phyla.table <- as.data.frame(t(phyla.table))
colnames(phyla.table) <- phyla.names
rownames(phyla.table) <- rownames(demographics)


#Add phyla together that are not very abundant and delete them from the table
phyla.table$other <- apply(phyla.table[, c("Acidobacteria", "Deferribacteres", "Deinococcus-Thermus", "Fusobacteria", "Lentisphaerae", "Spirochaetes", "Synergistetes", "Tenericutes", "Cyanobacteria_Chloroplast", "TM7")], 1, sum)
phyla.table <- phyla.table[, -c(1, 4:7, 9:11, 13, 15)]

#Create a relative abundance table for phyla
phyla.total <- apply(phyla.table[, c(1:7)], 1, sum)
phyla.table.rel.abund <- (phyla.table/phyla.total)*100
phyla.table.rel.abund <- phyla.table.rel.abund[, -8]

#Generate Alpha Diversity Measures and alpha diversity table
library(vegan)
H <- diversity(microbiome)
S <- specnumber(microbiome)
J <- H/log(S)
alpha.diversity.shannon <- cbind(H,S,J)
alpha.test <- as.data.frame(alpha.diversity.shannon)

#Create age quartiles

demographics$age.quartile <- with(demographics, cut(Age, breaks=quantile(Age, probs=seq(0,1, by=0.25)), include.lowest=TRUE))

#Create Obese Yes/No groups
demographics$obese[demographics$BMI.classification=="Normal" | demographics$BMI.classification=="Overweight"] <- "No"
demographics$obese[demographics$BMI.classification=="Obese" | demographics$BMI.classification=="Extreme Obesity"] <- "Yes"

#Create Obese.num groups
demographics$obese.num[demographics$obese=="No"] <- 0
demographics$obese.num[demographics$obese=="Yes"] <- 1
obese <- factor(demographics$obese.num)

######################################################################################## First Level Analysis & Alpha Diversity with BMI #############
###########################################################################

#Create a column with obese and extreme obese as single entity
demographics$BMIclass2[demographics$BMI.classification=="Normal"] <- "Normal"
demographics$BMIclass2[demographics$BMI.classification=="Overweight"] <- "Overweight"                     
demographics$BMIclass2[demographics$BMI.classification=="Obese" | demographics$BMI.classification=="Extreme Obesity"] <- "Obese"


##Test BMI versus alpha diversity and phyla

bmi <- demographics$BMI

baxterH <- anova(lm(H ~ obese)) #P-value=0.01643
baxterS <- anova(lm(S ~ obese)) #P-value=0.01571
baxterJ <- anova(lm(J ~ obese)) #P-value=0.03241

#B and F tests against obesity
bacter <- phyla.table.rel.abund$Bacteroidetes
firm <- phyla.table.rel.abund$Firmicutes
BFratio <- bacter/firm

baxterBacter <- anova(lm(bacter ~ obese)) #P-value=0.3175
baxterFirm <- anova(lm(firm ~ obese)) #P-value=0.6817
baxterBF <- anova(lm(BFratio ~ obese)) #P-value=0.6305

###########################################################################
############ NMDS and PERMANOVA Analysis###################################
###########################################################################

set.seed(3)
baxter2 <- adonis(microbiome ~ obese, permutations=1000)
baxterPERM <- baxter2$aov.tab
#Not significant PERMANOVA = 0.06394, pseudo-F = 1.4246


###########################################################################
############ Relative Risk#################################################
###########################################################################

library(epiR)

#Generate median values and put them into existing alpha.test dataframe

#Shannon diversity
alpha.test <- within(alpha.test, {shannon.cat = ifelse(H <= median(H), "less", "higher")})

##Run the RR for Shannon Diversity
H.cat <- alpha.test$shannon.cat
bmi.cat <- as.character(obese)
test3 <- cbind(H.cat, bmi.cat)
test3 <- test3[order(H.cat), ]
orderedHCat <- test3[, 1]
BaxtotalN <- length(orderedHCat)
BaxHHighTotal <- length(orderedHCat[orderedHCat=="higher"])
BaxHighShannonGroup <- as.data.frame(table(test3[c(1:BaxHHighTotal), 2]))
BaxLowShannonGroup <- as.data.frame(table(test3[c((BaxHHighTotal + 1):BaxtotalN), 2]))
#Group1 (Higher than median), obese = 22 and non-obese = 64
#Group2 (Lower than median), obese = 25 and non-obese = 61

group1 <- c(BaxHighShannonGroup[2, 2], BaxHighShannonGroup[1, 2])
group2 <- c(BaxLowShannonGroup[2, 2], BaxLowShannonGroup[1, 2])
r.test <- rbind(group2, group1)
colnames(r.test) <- c("Obese", "Not.Obese")
rownames(r.test) <- c("group2", "group1")

baxterHEpi <- epi.2by2(r.test, method="cohort.count")
baxterHMassoc <- baxterHEpi$massoc
baxterHRR <- baxterHMassoc$RR.strata.score
baxterHRRsig <- baxterHMassoc$chisq.strata
## Risk Ratio = 1.14
## CI = 0.70, 1.85
## p-value = 0.608

##Run the RR for B/F ratio
Bacter = phyla.table.rel.abund$Bacteroidetes
Firm = phyla.table.rel.abund$Firmicutes
BFRatio = Bacter/Firm
BFRatio <- as.data.frame(BFRatio)

BFRatio <- within(BFRatio, {BFRatio.cat = ifelse(BFRatio <= median(BFRatio), "less", "higher")})

BFRatio.cat <- BFRatio$BFRatio.cat
test4 <- cbind(BFRatio.cat, obese)
test4 <- test4[order(BFRatio.cat), ]
orderedBFCat <- test4[, 1]
BaxBFHighTotal <- length(orderedBFCat[orderedBFCat=="higher"])
BaxHighBFGroup <- as.data.frame(table(test4[c(1:BaxBFHighTotal), 2]))
BaxLowBFGroup <- as.data.frame(table(test4[c((BaxBFHighTotal + 1):BaxtotalN), 2]))
#Group1 (Higher than median), obese = 23 and non-obese = 63
#Group2 (Lower than median), obese = 24 and non-obese = 62

group1 <- c(BaxHighBFGroup[2, 2], BaxHighBFGroup[1, 2])
group2 <- c(BaxLowBFGroup[2, 2], BaxLowBFGroup[1, 2])
r.test <- rbind(group2, group1)
colnames(r.test) <- c("Obese", "Not.Obese")
rownames(r.test) <- c("group2", "group1")

baxterBFEpi <- epi.2by2(r.test, method="cohort.count")
baxterBFMassoc <- baxterBFEpi$massoc
baxterBFRR <- baxterBFMassoc$RR.strata.score
baxterBFRRsig <- baxterBFMassoc$chisq.strata
## Risk Ratio = 1.04
## CI = 0.64, 1.70
## p-value = 0.864

###########################################################################
############ Classification using AUCRF ###################################
###########################################################################

library(AUCRF)

#generate test set
  # get rid of values that only have 0 and something else
testset <- Filter(function(x)(length(unique(x))>2), microbiome)
  # get rid of those with 0 and only 4 other values
testset <- Filter(function(x)(length(unique(x))>5), microbiome)

#Need to add phyla and alpha diversity measures to the dataset

testset <- cbind(testset, H, S, J, phyla.table.rel.abund)
testset <- cbind(obese, testset)
         
#Try AUCRF with default measures provided in readme
#set.seed(3)
#baxterAUCfit <- AUCRF(obese ~ ., data=testset, ntree=1000, nodesize=20)

```

```{r Data_Ross, echo=FALSE, message=FALSE, warning=FALSE}
###########################################################################
############ Preparing Data Tables for Analysis ###########################
###########################################################################

setwd("C:/users/marc/Desktop/obesity2/hispanic/")

hispanic.microb <- read.table("hispanic.subsample.shared", header=T)

metadata <- read.csv("s40168-015-0072-y-s1.csv")
sample.match <- read.csv("Hispanic_dataset.csv")

#Create a microbiome data table with sample names from metadata
test <- cbind(as.character(sample.match$Run_s), as.character(sample.match$Library_Name_s))
test <- test[order(test[, 1]), ]

test2 <- cbind(test, hispanic.microb)
rownames(test2) <- test2[, 2]
his.microb.edit <- test2[, -c(1:5)]
edit.metadata <- metadata[, -c(2:5)]
rownames(edit.metadata) <- edit.metadata[, 1]
edit.metadata <- edit.metadata[, -1]

#Create a metadata file in the order of the microbiome data
order1 <- rownames(his.microb.edit)
edit.metadata2 <- edit.metadata[order1, ]

#Get alpha diversity of the samples
H <- diversity(his.microb.edit)
S <- specnumber(his.microb.edit)
J <- H/log(S)
alpha.diversity.shannon <- cbind(H,S,J)
alpha.test <- as.data.frame(alpha.diversity.shannon)

#Get phyla information
#Edited out non phyla information first with sed in linux
#combined new labels with previous taxonomy file with excel
phylogenetic.info <- read.csv("phyla.data.csv")
rownames(phylogenetic.info) <- phylogenetic.info[,1]
phylogenetic.info <- phylogenetic.info[,-c(1)]
phyla.names <- as.character(phylogenetic.info$Taxonomy)
keep <- colnames(his.microb.edit)
phyla.good <- phylogenetic.info[keep, ]
phyla.names <- as.character(phyla.good[,2])
phyla.table <- his.microb.edit
colnames(phyla.table) <- phyla.names
  #add all the same columns up and then return the sum
testing <- t(rowsum(t(phyla.table), group = rownames(t(phyla.table))))
phyla.table <- as.data.frame(testing)
rm(testing)
  #combine phyla that are not that abundant
phyla.table$other <- apply(phyla.table[, c("Deinococcus-Thermus", "Elusimicrobia", "Fusobacteria", "Lentisphaerae", "Synergistetes", "TM7")], 1, sum)
phyla.table <- phyla.table[, -c(3:4, 6:7, 9:10)]

#Create a relative abundance table for phyla
phyla.total <- apply(phyla.table[, c(1:7)], 1, sum)
phyla.table.rel.abund <- (phyla.table/phyla.total)*100
phyla.table.rel.abund <- as.data.frame(phyla.table.rel.abund[, -8])


#Create BMI groups
edit.metadata2$BMI.class[edit.metadata2$BMI<=24] <- "Normal"
edit.metadata2$BMI.class[edit.metadata2$BMI>24 & edit.metadata2$BMI<30] <- "Overweight"
edit.metadata2$BMI.class[edit.metadata2$BMI>=30 & edit.metadata2$BMI<40] <- "Obese"
edit.metadata2$BMI.class[edit.metadata2$BMI>=40] <- "Extreme Obesity"


#Create Obese Yes/No groups
edit.metadata2$obese[edit.metadata2$BMI.class=="Normal" | edit.metadata2$BMI.class=="Overweight"] <- "No"
edit.metadata2$obese[edit.metadata2$BMI.class=="Obese" | edit.metadata2$BMI.class=="Extreme Obesity"] <- "Yes"

#Create BMI class 2 groups
edit.metadata2$BMI.class2[edit.metadata2$BMI<=24] <- "Normal"
edit.metadata2$BMI.class2[edit.metadata2$BMI>24 & edit.metadata2$BMI<30] <- "Overweight"
edit.metadata2$BMI.class2[edit.metadata2$BMI>=30] <- "Obese"

#Get paitent demographics to be tested

bmi <- edit.metadata2$BMI
obese <- factor(edit.metadata2$obese)

######################################################################################## First Level Analysis & Alpha Diversity with BMI #############
###########################################################################

rossH <- anova(lm(H ~ obese)) #P-value=0.1657
rossS <- anova(lm(S ~ obese)) #P-value=0.1841
rossJ <- anova(lm(J ~ obese)) #P-value=0.2538

#B and F tests against obesity
bacter <- phyla.table.rel.abund$Bacteroidetes
firm <- phyla.table.rel.abund$Firmicutes
BFratio <- bacter/firm

rossBacter <- anova(lm(bacter ~ obese)) #P-value=0.2092
rossFirm <- anova(lm(firm ~ obese)) #P-value=0.2569
rossBF <- anova(lm(BFratio ~ obese)) #P-value=0.2508

###########################################################################
############ NMDS and PERMANOVA Analysis###################################
###########################################################################

set.seed(3)
ross2 <- adonis(his.microb.edit ~ obese, permutations=1000)
rossPERM <- ross2$aov.tab
#PERMANOVA=0.7153, pseudo-F=0.73249

###########################################################################
############ Relative Risk#################################################
###########################################################################

library(epiR)

#Generate median values and put them into existing alpha.test dataframe
#Shannon diversity

alpha.test <- within(alpha.test, {shannon.cat = ifelse(H <= median(H), "less", "higher")})

##Shannon Diversity
H.cat <- alpha.test$shannon.cat
bmi.cat <- as.character(obese)
test3 <- cbind(H.cat, bmi.cat)
test3 <- test3[order(H.cat), ]
orderedHCat <- test3[, 1]
RosstotalN <- length(orderedHCat)
RossHHighTotal <- length(orderedHCat[orderedHCat=="higher"])
RossHighShannonGroup <- as.data.frame(table(test3[c(1:RossHHighTotal), 2]))
RossLowShannonGroup <- as.data.frame(table(test3[c((RossHHighTotal + 1):RosstotalN), 2]))
group1 <- c(RossHighShannonGroup[2, 2], RossHighShannonGroup[1, 2])
group2 <- c(RossLowShannonGroup[2, 2], RossLowShannonGroup[1, 2])
#Group1 (Higher than median), obese = 18 and non-obese = 13
#Group2 (Lower than median), obese = 20 and non-obese = 12

r.test <- rbind(group2, group1)
colnames(r.test) <- c("Obese", "Not.Obese")
rownames(r.test) <- c("group1", "group2")

rossHEpi <- epi.2by2(r.test, method="cohort.count")
rossHMassoc <- rossHEpi$massoc
rossHRR <- rossHMassoc$RR.strata.score
rossHRRsig <- rossHMassoc$chisq.strata
## Risk Ratio = 1.08
## CI = 0.72, 1.61
## p-value = 0.719


##Run the RR for B/F ratio
Bacter = phyla.table.rel.abund$Bacteroidetes
Firm = phyla.table.rel.abund$Firmicutes
BFRatio = Bacter/Firm
BFRatio <- as.data.frame(BFRatio)

BFRatio <- within(BFRatio, {BFRatio.cat = ifelse(BFRatio <= median(BFRatio), "less", "higher")})

BFRatio.cat <- BFRatio$BFRatio.cat
test4 <- cbind(BFRatio.cat, obese)
test4 <- test4[order(BFRatio.cat), ]
orderedBFCat <- test4[, 1]
RossBFHighTotal <- length(orderedBFCat[orderedBFCat=="higher"])
RossHighBFGroup <- as.data.frame(table(test4[c(1:RossBFHighTotal), 2]))
RossLowBFGroup <- as.data.frame(table(test4[c((RossBFHighTotal + 1):RosstotalN), 2]))
group1 <- c(RossHighBFGroup[2, 2], RossHighBFGroup[1, 2])
group2 <- c(RossLowBFGroup[2, 2], RossLowBFGroup[1, 2])
#Group1 (Higher than median), obese = 22 and non-obese = 9
#Group2 (Lower than median), obese = 16 and non-obese = 16


r.test <- rbind(group2, group1)
colnames(r.test) <- c("Obese", "Not.Obese")
rownames(r.test) <- c("group2", "group1")

rossBFEpi <- epi.2by2(r.test, method="cohort.count")
rossBFMassoc <- rossBFEpi$massoc
rossBFRR <- rossBFMassoc$RR.strata.score
rossBFRRsig <- rossBFMassoc$chisq.strata
## Risk Ratio = 0.70
## CI = 0.47, 1.07
## p-value = 0.089

###########################################################################
############ Classification using AUCRF ###################################
###########################################################################

#Create Obese.num group
edit.metadata2$obese.num[edit.metadata2$obese=="No"] <- 0
edit.metadata2$obese.num[edit.metadata2$obese=="Yes"] <- 1
obese <- factor(edit.metadata2$obese.num)

#generate test set
  # get rid of those with 0 and only 4 other values
testset <- Filter(function(x)(length(unique(x))>5), his.microb.edit)
testset <- cbind(obese, testset)
colnames(testset)[1] <- "obese"
testset <- cbind(testset, H, S, J, phyla.table.rel.abund)

#Try AUCRF with default measures provided in readme
#set.seed(3)
#rossAUCFit <- AUCRF(obese ~ ., data=testset, ntree=1000, nodesize=20)
# list of 3 Measures, AUCopt = 0.7526

```

```{r Goodrich_data, echo=FALSE, message=FALSE, warning=FALSE}
###########################################################################
############ Preparing Data Tables for Analysis ###########################
###########################################################################


setwd("C:/users/marc/Desktop/obesity2/twinsUK")

#Read in and match metadata to microbiome data
metadata <- read.csv("TwinsUKStudy_metadata.csv")
rownames(metadata) <- metadata[, 1]
shared.data <- read.table("combined.tx.1.subsample.shared", header=T)
rownames(shared.data) <- shared.data[, 2]
microbiome <- shared.data[, -c(1:3)]
keep  <- rownames(metadata)

microbiome<- microbiome[keep, ]

rm(good.metadata, shared.data, keep)


#generate alpha diversity measures with vegan
library(vegan)
H <- diversity(microbiome)
S <- specnumber(microbiome)
J <- H/log(S)
alpha.diversity <- as.data.frame(cbind(H, S, J))
alpha.test <- alpha.diversity

#Get phyla information
#Edited out non phyla information first with sed in linux
#combined new labels with previous taxonomy file with excel
phylogenetic.info <- read.table("phyla.txt", header=T)
rownames(phylogenetic.info) <- phylogenetic.info[,1]
phylogenetic.info <- phylogenetic.info[,-c(1)]
phyla.names <- as.character(phylogenetic.info$Taxonomy)
keep <- colnames(microbiome)
phyla.good <- phylogenetic.info[keep, ]
phyla.names <- as.character(phyla.good[,2])
phyla.table <- microbiome
colnames(phyla.table) <- phyla.names
#add all the same columns up and then return the sum
testing <- t(rowsum(t(phyla.table), group = rownames(t(phyla.table))))
phyla.table <- as.data.frame(testing)
rm(testing)
#combine phyla that are not that abundant
phyla.table$other <- apply(phyla.table[, c("Acidobacteria", "Chloroflexi", "Deferribacteres", "Elusimicrobia", "Fusobacteria", "Gemmatimonadetes", "Lentisphaerae", "OD1", "Planctomycetes", "Spirochaetes", "SR1", "Synergistetes", "Tenericutes")], 1, sum)
phyla.table <- phyla.table[, -c(1, 4:6, 8:12, 14:17)]

#Create a relative abundance table for phyla
phyla.total <- apply(phyla.table[, c(1:7)], 1, sum)
phyla.table.rel.abund <- (phyla.table/phyla.total)*100

#Create BMI groups
metadata$BMI.class[metadata$body_mass_index_s<=24] <- "Normal"
metadata$BMI.class[metadata$body_mass_index_s>24 & metadata$body_mass_index_s<30] <- "Overweight"
metadata$BMI.class[metadata$body_mass_index_s>=30 & metadata$body_mass_index_s<40] <- "Obese"
metadata$BMI.class[metadata$body_mass_index_s>=40] <- "Extreme Obesity"

#Create Obese Yes/No groups
metadata$obese[metadata$BMI.class=="Normal" | metadata$BMI.class=="Overweight"] <- "No"
metadata$obese[metadata$BMI.class=="Obese" | metadata$BMI.class=="Extreme Obesity"] <- "Yes"

#Create a column with obese and extreme obese as single entity
metadata$BMIclass2[metadata$BMI.class=="Normal"] <- "Normal"
metadata$BMIclass2[metadata$BMI.class=="Overweight"] <- "Overweight"                     
metadata$BMIclass2[metadata$BMI.class=="Obese" | metadata$BMI.class=="Extreme Obesity"] <- "Obese"

#Get paitent demographics to be tested
bmi <- metadata$body_mass_index_s
obese <- factor(metadata$obese)



###########################################################################
######### First Level Analysis & Alpha Diversity with BMI #################
###########################################################################

##Test BMI versus alpha diversity and phyla

goodrichH <- anova(lm(H ~ obese)) #P-value=0.6671
goodrichS <- anova(lm(S ~ obese)) #P-value=0.9073
goodrichJ <- anova(lm(J ~ obese)) #P-value=0.6319

#B and F tests against obesity
bacter <- phyla.table.rel.abund$Bacteroidetes
firm <- phyla.table.rel.abund$Firmicutes
BFratio <- bacter/firm

goodrichBacter <- anova(lm(bacter ~ obese)) #P-value=0.6661
goodrichFirm <- anova(lm(firm ~ obese)) #P-value=0.9688
goodrichBF <- anova(lm(BFratio ~ obese)) #P-value=0.8709

###########################################################################
############ NMDS and PERMANOVA Analysis###################################
###########################################################################

set.seed(3)
goodrich2 <- adonis(microbiome ~ obese, permutations=1000) 
goodrichPERM <- goodrich2$aov.tab
#PERMANOVA=0.004995, pseudo-F=2.9308

###########################################################################
############ Relative Risk#################################################
###########################################################################

#Generate median values and put them into existing alpha.test dataframe
#Shannon diversity

alpha.test <- within(alpha.test, {shannon.cat = ifelse(H <= median(H), "less", "higher")})


##Shannon Diversity
H.cat <- alpha.test$shannon.cat
bmi.cat <- as.character(obese)
test3 <- cbind(H.cat, bmi.cat)
test3 <- test3[order(H.cat), ]
orderedHCat <- test3[, 1]
GoodtotalN <- length(orderedHCat)
GoodHHighTotal <- length(orderedHCat[orderedHCat=="higher"])
GoodHighShannonGroup <- as.data.frame(table(test3[c(1:GoodHHighTotal), 2]))
GoodLowShannonGroup <- as.data.frame(table(test3[c((GoodHHighTotal + 1):GoodtotalN), 2]))
group1 <- c(GoodHighShannonGroup[2, 2], GoodHighShannonGroup[1, 2])
group2 <- c(GoodLowShannonGroup[2, 2], GoodLowShannonGroup[1, 2])
#Group1 (Higher than median), obese = 47 and non-obese = 206
#Group2 (Lower than median), obese = 56 and non-obese = 198

r.test <- rbind(group2, group1)
colnames(r.test) <- c("Obese", "Not.Obese")
rownames(r.test) <- c("group2", "group1")

goodrichHEpi <- epi.2by2(r.test, method="cohort.count")
goodrichHMassoc <- goodrichHEpi$massoc
goodrichHRR <- goodrichHMassoc$RR.strata.score
goodrichHRRsig <- goodrichHMassoc$chisq.strata
## Risk Ratio = 1.19
## CI = 0.84, 1.68
## p-value = 0.332

##Run the RR for B/F ratio
Bacter = phyla.table.rel.abund$Bacteroidetes
Firm = phyla.table.rel.abund$Firmicutes
BFRatio = Bacter/Firm
BFRatio <- as.data.frame(BFRatio)

BFRatio <- within(BFRatio, {BFRatio.cat = ifelse(BFRatio <= median(BFRatio), "less", "higher")})

BFRatio.cat <- BFRatio$BFRatio.cat
test4 <- cbind(BFRatio.cat, obese)
test4 <- test4[order(BFRatio.cat), ]
orderedBFCat <- test4[, 1]
GoodBFHighTotal <- length(orderedBFCat[orderedBFCat=="higher"])
GoodHighBFGroup <- as.data.frame(table(test4[c(1:GoodBFHighTotal), 2]))
GoodLowBFGroup <- as.data.frame(table(test4[c((GoodBFHighTotal + 1):GoodtotalN), 2]))
group1 <- c(GoodHighBFGroup[2, 2], GoodHighBFGroup[1, 2])
group2 <- c(GoodLowBFGroup[2, 2], GoodLowBFGroup[1, 2])
#Group1 (Higher than median), obese = 52 and non-obese = 201
#Group2 (Lower than median), obese = 51 and non-obese = 203

r.test <- rbind(group2, group1)
colnames(r.test) <- c("Obese", "Not.Obese")
rownames(r.test) <- c("group2", "group1")

goodrichBFEpi <- epi.2by2(r.test, method="cohort.count")
goodrichBFMassoc <- goodrichBFEpi$massoc
goodrichBFRR <- goodrichBFMassoc$RR.strata.score
goodrichBFRRsig <- goodrichBFMassoc$chisq.strata
## Risk Ratio = 0.98
## CI = 0.69, 1.38
## p-value = 0.894

###########################################################################
############ Classification using AUCRF####################################
###########################################################################

#Create Obese.num group
metadata$obese.num[metadata$obese=="No"] <- 0
metadata$obese.num[metadata$obese=="Yes"] <- 1
obese <- factor(metadata$obese.num)

#generate test set
# get rid of those with 0 and only 4 other values
testset <- Filter(function(x)(length(unique(x))>5), microbiome)
testset <- cbind(obese, testset)
colnames(testset)[1] <- "obese"
testset <- cbind(testset, H, S, J, phyla.table.rel.abund)

#Try AUCRF with default measures provided in readme
#set.seed(3)
#goodrichAUCFit <- AUCRF(obese ~ ., data=testset, ntree=1000, nodesize=20)
# list of 8 Measures, AUCopt = 0.6767159

```

```{r escobar_data, echo=FALSE, message=FALSE, warning=FALSE}
###########################################################################
############ Preparing Data Tables for Analysis ###########################
###########################################################################

setwd("C:/users/marc/Desktop/obesity2/columbian/")

columbian.microb <- read.table("columbian.0.03.subsample.shared", header=T)
metadata <- read.csv("columbian_dataset.csv")

#Organize the microbiome shared data
rownames(columbian.microb) <- columbian.microb[, 2]
columbian.microb <- columbian.microb[, -c(1:3)]

#Organize the metadata
rownames(metadata) <- metadata[, 5]

#Get only data that we are interested in
edit.metadata <- metadata[, -c(1:7, 9, 13:15, 16:52)]

#Sort metadata into the same order as the microbiome data
order1 <- rownames(columbian.microb)
edit.metadata2 <- edit.metadata[order1, ]


#Get alpha diversity of the samples
H <- diversity(columbian.microb)
S <- specnumber(columbian.microb)
J <- H/log(S)
alpha.diversity.shannon <- cbind(H,S,J)
alpha.test <- as.data.frame(alpha.diversity.shannon)

#Get phyla information
#Edited out non phyla information first with sed in linux
#combined new labels with previous taxonomy file with excel
phylogenetic.info <- read.csv("phyla.data.csv")
rownames(phylogenetic.info) <- phylogenetic.info[,1]
phylogenetic.info <- phylogenetic.info[,-c(1)]
phyla.names <- as.character(phylogenetic.info$Taxonomy)
keep <- colnames(columbian.microb)
phyla.good <- phylogenetic.info[keep, ]
phyla.names <- as.character(phyla.good[,2])
phyla.table <- columbian.microb
colnames(phyla.table) <- phyla.names
#add all the same columns up and then return the sum
testing <- t(rowsum(t(phyla.table), group = rownames(t(phyla.table))))
phyla.table <- as.data.frame(testing)
rm(testing)
#combine phyla that are not that abundant
phyla.table$other <- apply(phyla.table[, c("Fusobacteria", "Lentisphaerae", "Spirochaetes", "Synergistetes", "TM7")], 1, sum)
phyla.table <- phyla.table[, -c(4:5, 7:9)]

#Create a relative abundance table for phyla
phyla.total <- apply(phyla.table[, c(1:7)], 1, sum)
phyla.table.rel.abund <- (phyla.table/phyla.total)*100
phyla.table.rel.abund <- phyla.table.rel.abund[, -8]


#Create Obese Yes/No groups
edit.metadata2$obese[edit.metadata2$description_s=="Adequate weight male stool sample" | edit.metadata2$description_s=="Overweight male stool sample" | edit.metadata2$description_s=="Adequate weight female stool sample" | edit.metadata2$description_s=="Overweight female stool sample"] <- "No"
edit.metadata2$obese[edit.metadata2$description_s=="Obese male stool sample" | edit.metadata2$description_s=="Obese female stool sample"] <- "Yes"

#Create BMI groups
edit.metadata2$BMI.class[edit.metadata2$body_mass_index_s<=24] <- "Normal"
edit.metadata2$BMI.class[edit.metadata2$body_mass_index_s>24 & edit.metadata2$body_mass_index_s<30] <- "Overweight"
edit.metadata2$BMI.class[edit.metadata2$body_mass_index_s>=30 & edit.metadata2$body_mass_index_s<40] <- "Obese"
edit.metadata2$BMI.class[edit.metadata2$body_mass_index_s>=40] <- "Extreme Obesity"

#Create BMI groups2
edit.metadata2$BMI.class2[edit.metadata2$body_mass_index_s<=24] <- "Normal"
edit.metadata2$BMI.class2[edit.metadata2$body_mass_index_s>24 & edit.metadata2$body_mass_index_s<30] <- "Overweight"
edit.metadata2$BMI.class2[edit.metadata2$body_mass_index_s>=30] <- "Obese"

#Get paitent demographics data to be tested 
bmi <- edit.metadata2$body_mass_index_s
obese <- factor(edit.metadata2$obese)

######################################################################################## First Level Analysis & Alpha Diversity with BMI #############
###########################################################################

##Test BMI versus alpha diversity and phyla

escobarH <- anova(lm(H ~ obese)) #P-value=0.8636
escobarS <- anova(lm(S ~ obese)) #P-value=0.1947
escobarJ <- anova(lm(J ~ obese)) #P-value=0.4762

#B and F tests against obesity
bacter <- phyla.table.rel.abund$Bacteroidetes
firm <- phyla.table.rel.abund$Firmicutes
BFratio <- bacter/firm

escobarBacter <- anova(lm(bacter ~ obese)) #P-value=0.09253
escobarFirm <- anova(lm(firm ~ obese)) #P-value=0.3602
escobarBF <- anova(lm(BFratio ~ obese)) #P-value=0.4663

###########################################################################
############ NMDS and PERMANOVA Analysis###################################
###########################################################################

set.seed(3)
escobar2 <- adonis(columbian.microb ~ obese, permutations=1000) 
escobarPERM <- escobar2$aov.tab
#PERMANOVA=0.08891, pseudo-F=1.3797

###########################################################################
############ Relative Risk#################################################
###########################################################################

#Generate median values and put them into existing alpha.test dataframe
#Shannon diversity

alpha.test <- within(alpha.test, {shannon.cat = ifelse(H <= median(H), "less", "higher")})

##Shannon Diversity
H.cat <- alpha.test$shannon.cat
bmi.cat <- as.character(obese)
test3 <- cbind(H.cat, bmi.cat)
test3 <- test3[order(H.cat), ]
orderedHCat <- test3[, 1]
EscototalN <- length(orderedHCat)
EscoHHighTotal <- length(orderedHCat[orderedHCat=="higher"])
EscoHighShannonGroup <- as.data.frame(table(test3[c(1:EscoHHighTotal), 2]))
EscoLowShannonGroup <- as.data.frame(table(test3[c((EscoHHighTotal + 1):EscototalN), 2]))
group1 <- c(EscoHighShannonGroup[2, 2], EscoHighShannonGroup[1, 2])
group2 <- c(EscoLowShannonGroup[2, 2], EscoLowShannonGroup[1, 2])
#Group1 (Higher than median), obese = 4 and non-obese = 11
#Group2 (Lower than median), obese = 6 and non-obese = 9

r.test <- rbind(group2, group1)
colnames(r.test) <- c("Obese", "Not.Obese")
rownames(r.test) <- c("group1", "group2")

escobarHEpi <- epi.2by2(r.test, method="cohort.count")
escobarHMassoc <- escobarHEpi$massoc
escobarHRR <- escobarHMassoc$RR.strata.score
escobarHRRsig <- escobarHMassoc$chisq.strata
## Risk Ratio = 1.50
## CI = 0.53, 4.26
## p-value = 0.439

##Run the RR for B/F ratio
Bacter = phyla.table.rel.abund$Bacteroidetes
Firm = phyla.table.rel.abund$Firmicutes
BFRatio = Bacter/Firm
BFRatio <- as.data.frame(BFRatio)

BFRatio <- within(BFRatio, {BFRatio.cat = ifelse(BFRatio <= median(BFRatio), "less", "higher")})

BFRatio.cat <- BFRatio$BFRatio.cat
test4 <- cbind(BFRatio.cat, obese)
test4 <- test4[order(BFRatio.cat), ]
orderedBFCat <- test4[, 1]
EscoBFHighTotal <- length(orderedBFCat[orderedBFCat=="higher"])
EscoHighBFGroup <- as.data.frame(table(test4[c(1:EscoBFHighTotal), 2]))
EscoLowBFGroup <- as.data.frame(table(test4[c((EscoBFHighTotal + 1):EscototalN), 2]))
group1 <- c(EscoHighBFGroup[2, 2], EscoHighBFGroup[1, 2])
group2 <- c(EscoLowBFGroup[2, 2], EscoLowBFGroup[1, 2])
#Group1 (Higher than median), obese = 7 and non-obese = 8
#Group2 (Lower than median), obese = 3 and non-obese = 12

r.test <- rbind(group2, group1)
colnames(r.test) <- c("Obese", "Not.Obese")
rownames(r.test) <- c("group2", "group1")

escobarBFEpi <- epi.2by2(r.test, method="cohort.count")
escobarBFMassoc <- escobarBFEpi$massoc
escobarBFRR <- escobarBFMassoc$RR.strata.score
escobarBFRRsig <- escobarBFMassoc$chisq.strata
## Risk Ratio = 0.43
## CI = 0.14, 1.35
## p-value = 0.121

###########################################################################
############ Classification using AUCRF ###################################
###########################################################################

#Create Obese.num group
edit.metadata2$obese.num[edit.metadata2$obese=="No"] <- 0
edit.metadata2$obese.num[edit.metadata2$obese=="Yes"] <- 1
obese <- factor(edit.metadata2$obese.num)

#generate test set
  # get rid of those with 0 and only 4 other values
testset <- Filter(function(x)(length(unique(x))>5), columbian.microb)
testset <- cbind(obese, testset)
colnames(testset)[1] <- "obese"
testset <- cbind(testset, H, S, J, phyla.table.rel.abund)

#Try AUCRF with default measures provided in readme
#set.seed(3)
#escobarAUCFit <- AUCRF(obese ~ ., data=testset, ntree=1000, nodesize=20)
# list of 3 Measures, AUCopt = 0.95

```

```{r Zupancic_data, echo=FALSE, message=FALSE, warning=FALSE}
###########################################################################
############ Preparing Data Tables for Analysis ###########################
###########################################################################


setwd("C:/users/marc/Desktop/obesity2/Amish/updatedGOOD")

#Read in and match metadata to microbiome data
metadata <- read.csv("amish_obesity_table2.csv")
metadata2 <- read.csv("amish.metadata.csv")
test <- metadata[!duplicated(metadata$submitted_sample_id_s), ]
rownames(test) <- test[, 4]
test2 <- metadata2[!duplicated(metadata2$SUBJID), ]
rownames(test2) <- test2[, 1]
shared.data <- read.table("combined.0.03.subsample.shared", header=T)
rownames(shared.data) <- shared.data[, 2]
shared.data <- shared.data[, -c(1:3)]
BactFamily <- read.csv("BacterialFamilyData.csv")
rownames(BactFamily) <- BactFamily[, 1]
BactFamily <- BactFamily[, -1]
keep  <- rownames(shared.data)

test3 <- test[keep, ]
good.metadata <- test3
test4 <- test2[keep, ]
good.metadata2 <- test4

family <- BactFamily[keep, ]
microbiome <- shared.data[keep, ]
metadata <- cbind(good.metadata, good.metadata2)

rm(good.metadata, shared.data, keep, test, test2, test3, test4, BactFamily, good.metadata2, metadata2)

test <- metadata[!duplicated(metadata$submitted_subject_id_s), ]
keep  <- rownames(test)
metadata <- test
family <- family[keep, ]
microbiome <- microbiome[keep, ]

rm(test, keep)

metadata <- metadata[complete.cases(metadata), ]
keep <- rownames(metadata)
family <- family[keep, ]
microbiome <- microbiome[keep, ]

rm(keep)

#generate alpha diversity measures with vegan
library(vegan)
H <- diversity(microbiome)
S <- specnumber(microbiome)
J <- H/log(S)
select.alpha.diversity <- as.data.frame(cbind(H, S, J))
s1.alpha.diversity <- as.data.frame(select.alpha.diversity)
alpha.test <- s1.alpha.diversity

#Get phyla information
#Edited out non phyla information first with sed in linux
#combined new labels with previous taxonomy file with excel
phylogenetic.info <- read.csv("phyla.data.csv")
rownames(phylogenetic.info) <- phylogenetic.info[,1]
phylogenetic.info <- phylogenetic.info[,-c(1)]
phyla.names <- as.character(phylogenetic.info$Taxonomy)
keep <- colnames(microbiome)
phyla.good <- phylogenetic.info[keep, ]
phyla.names <- as.character(phyla.good[,2])
phyla.table <- microbiome
colnames(phyla.table) <- phyla.names
#add all the same columns up and then return the sum
testing <- t(rowsum(t(phyla.table), group = rownames(t(phyla.table))))
phyla.table <- as.data.frame(testing)
rm(testing)
#combine phyla that are not that abundant
phyla.table$other <- apply(phyla.table[, c("Fusobacteria", "Spirochaetes")], 1, sum)
phyla.table <- phyla.table[, -c(4, 6)]

#Create a relative abundance table for phyla
phyla.total <- apply(phyla.table[, c(1:6)], 1, sum)
phyla.table.rel.abund <- (phyla.table/phyla.total)*100

#Create BMI groups
metadata$BMI.class[metadata$BMI<=24] <- "Normal"
metadata$BMI.class[metadata$BMI>24 & metadata$BMI<30] <- "Overweight"
metadata$BMI.class[metadata$BMI>=30 & metadata$BMI<40] <- "Obese"
metadata$BMI.class[metadata$BMI>=40] <- "Extreme Obesity"


#Create Obese Yes/No groups
metadata$obese[metadata$BMI.class=="Normal" | metadata$BMI.class=="Overweight"] <- "No"
metadata$obese[metadata$BMI.class=="Obese" | metadata$BMI.class=="Extreme Obesity"] <- "Yes"

#Create BMI class 2 groups
metadata$BMI.class2[metadata$BMI<=24] <- "Normal"
metadata$BMI.class2[metadata$BMI>24 & metadata$BMI<30] <- "Overweight"
metadata$BMI.class2[metadata$BMI>=30] <- "Obese"

#create groups to be used
bmi <- metadata$BMI
obese <- factor(metadata$obese)


####################################################################################### First Level Analysis & Alpha Diversity with BMI #############
###########################################################################
###########################################################################

##Test BMI versus alpha diversity and phyla

zupancicH <- anova(lm(H ~ obese)) #P-value=0.2671
zupancicS <- anova(lm(S ~ obese)) #P-value=0.05423
zupancicJ <- anova(lm(J ~ obese)) #P-value=0.6229

#B and F tests against obesity
bacter <- phyla.table.rel.abund$Bacteroidetes
firm <- phyla.table.rel.abund$Firmicutes
BFratio <- bacter/firm

zupancicBacter <- anova(lm(bacter ~ obese)) #P-value=0.8592
zupancicFirm <- anova(lm(firm ~ obese)) #P-value=0.2564
zupancicBF <- anova(lm(BFratio ~ obese)) #P-value=0.989

####################################################################################### NMDS and PERMANOVA Analysis###################################
###########################################################################

set.seed(3)
zupancic2 <- adonis(microbiome ~ obese, permutations=1000) 
zupancicPERM <- zupancic2$aov.tab
#PERMANOVA=0.6863, pseudo-F=0.92888

###########################################################################
############ Relative Risk#################################################
###########################################################################

#Generate median values and put them into existing alpha.test dataframe
#Shannon diversity

alpha.test <- within(alpha.test, {shannon.cat = ifelse(H <= median(H), "less", "higher")})


##Shannon Diversity
H.cat <- alpha.test$shannon.cat
bmi.cat <- as.character(obese)
test3 <- cbind(H.cat, bmi.cat)
test3 <- test3[order(H.cat), ]
orderedHCat <- test3[, 1]
ZupatotalN <- length(orderedHCat)
ZupaHHighTotal <- length(orderedHCat[orderedHCat=="higher"])
ZupaHighShannonGroup <- as.data.frame(table(test3[c(1:ZupaHHighTotal), 2]))
ZupaLowShannonGroup <- as.data.frame(table(test3[c((ZupaHHighTotal + 1):ZupatotalN), 2]))
group1 <- c(ZupaHighShannonGroup[2, 2], ZupaHighShannonGroup[1, 2])
group2 <- c(ZupaLowShannonGroup[2, 2], ZupaLowShannonGroup[1, 2])
#Group1 (Higher than median), obese = 31 and non-obese = 72
#Group2 (Lower than median), obese = 47 and non-obese = 57

r.test <- rbind(group2, group1)
colnames(r.test) <- c("Obese", "Not.Obese")
rownames(r.test) <- c("group2", "group1")

zupancicHEpi <- epi.2by2(r.test, method="cohort.count")
zupancicHMassoc <- zupancicHEpi$massoc
zupancicHRR <- zupancicHMassoc$RR.strata.score
zupancicHRRsig <- zupancicHMassoc$chisq.strata
## Risk Ratio = 1.49
## CI = 1.01, 2.19
## p-value = 0.043


##Run the RR for B/F ratio
Bacter = phyla.table.rel.abund$Bacteroidetes
Firm = phyla.table.rel.abund$Firmicutes
BFRatio = Bacter/Firm
BFRatio <- as.data.frame(BFRatio)

BFRatio <- within(BFRatio, {BFRatio.cat = ifelse(BFRatio <= median(BFRatio), "less", "higher")})

BFRatio.cat <- BFRatio$BFRatio.cat
test4 <- cbind(BFRatio.cat, obese)
test4 <- test4[order(BFRatio.cat), ]
orderedBFCat <- test4[, 1]
ZupaBFHighTotal <- length(orderedBFCat[orderedBFCat=="higher"])
ZupaHighBFGroup <- as.data.frame(table(test4[c(1:ZupaBFHighTotal), 2]))
ZupaLowBFGroup <- as.data.frame(table(test4[c((ZupaBFHighTotal + 1):ZupatotalN), 2]))
group1 <- c(ZupaHighBFGroup[2, 2], ZupaHighBFGroup[1, 2])
group2 <- c(ZupaLowBFGroup[2, 2], ZupaLowBFGroup[1, 2])
#Group1 (Higher than median), obese = 41 and non-obese = 62
#Group2 (Lower than median), obese = 37 and non-obese = 67

r.test <- rbind(group2, group1)
colnames(r.test) <- c("Obese", "Not.Obese")
rownames(r.test) <- c("group2", "group1")

zupancicBFEpi <- epi.2by2(r.test, method="cohort.count")
zupancicBFMassoc <- zupancicBFEpi$massoc
zupancicBFRR <- zupancicBFMassoc$RR.strata.score
zupancicBFRRsig <- zupancicBFMassoc$chisq.strata
## Risk Ratio = 0.94
## CI = 0.64, 1.36
## p-value = 0.727

###########################################################################
############ Classification using AUCRF ###################################
###########################################################################

#Create Obese.num group
metadata$obese.num[metadata$obese=="No"] <- 0
metadata$obese.num[metadata$obese=="Yes"] <- 1
obese <- factor(metadata$obese.num)

#generate test set
# get rid of those with 0 and only 4 other values
testset <- Filter(function(x)(length(unique(x))>5), microbiome)
testset <- cbind(obese, testset)
colnames(testset)[1] <- "obese"
testset <- cbind(testset, H, S, J, phyla.table.rel.abund)

#Try AUCRF with default measures provided in readme
#set.seed(3)
#zupancicAUCFit <- AUCRF(obese ~ ., data=testset, ntree=1000, nodesize=20)
#zupAUC <- fit$`OOB-AUCopt`
# list of 40 Measures, AUCopt = 0.6596899

```


```{r HMP_data, echo=FALSE, message=FALSE, warning=FALSE}
###########################################################################
############ Preparing Data Tables for Analysis ###########################
###########################################################################

##Read in relevant data

setwd("C:/users/marc/Desktop/obesity2/HMP.analysis")
microbiome <- read.table("Stool.an.0.03.subsample.shared", header=T)
meta.cat <- read.table("categorical.metadata", header=T)
meta.cont <- read.table("continuous.metadata", header=T)

#Only interested in first visit microbiome data
#need to create microbiome data set for only that

test <- microbiome[grep("\\.01\\.", microbiome$Group), ] #selecting by .01. 

microb.rownames <- gsub("([0-9]+).*", "\\1", test$Group) #extracting only numeric before first "."
rownames(test) <- microb.rownames

#Get rid of information not used in downstram analysis
test <- test[, -c(1:3)]

#Subset data to the microbiome total n
select.meta.cat <- meta.cat[microb.rownames, ]
select.meta.cont <- meta.cont[microb.rownames, ]

#generate alpha diversity measures with vegan
library(vegan)
H <- diversity(test)
S <- specnumber(test)
J <- H/log(S)
select.alpha.diversity <- as.data.frame(cbind(H, S, J))
alpha.test <- select.alpha.diversity

#Get phyla information
#Edited out non phyla information first with sed in linux
#combined new labels with previous taxonomy file with excel
phylogenetic.info <- read.csv("phyla.data.csv")
rownames(phylogenetic.info) <- phylogenetic.info[,1]
phylogenetic.info <- phylogenetic.info[,-c(1)]
phyla.names <- as.character(phylogenetic.info$Taxonomy)
keep <- colnames(test)
phyla.good <- phylogenetic.info[keep, ]
phyla.names <- as.character(phyla.good[,2])
phyla.table <- test
colnames(phyla.table) <- phyla.names
#add all the same columns up and then return the sum
testing <- t(rowsum(t(phyla.table), group = rownames(t(phyla.table))))
phyla.table <- as.data.frame(testing)
rm(testing)

#combine phyla that are not that abundant
phyla.table$other <- apply(phyla.table[, c("Acidobacteria", "Deinococcus-Thermus", "Fusobacteria", "Lentisphaerae", "Spirochaetes", "Synergistetes", "Tenericutes", "TM7")], 1, sum)
phyla.table <- phyla.table[, -c(1, 4, 6, 7, 9, 10:12)]

#Create a relative abundance table for phyla
phyla.total <- apply(phyla.table[, c(1:7)], 1, sum)
phyla.table.rel.abund <- (phyla.table/phyla.total)*100
phyla.table.rel.abund <- phyla.table.rel.abund[, -8]


#Create Obese Yes/No groups
select.meta.cat$obese[select.meta.cat$BMI_C=="normal" | select.meta.cat$BMI_C=="overweight"] <- "No"
select.meta.cat$obese[select.meta.cat$BMI_C=="obese" | select.meta.cat$BMI_C=="extreme obesity"] <- "Yes"

#Create BMI groups2
select.meta.cat$BMI.class2[select.meta.cont$DTPBMI<=24] <- "Normal"
select.meta.cat$BMI.class2[select.meta.cont$DTPBMI>24 & select.meta.cont$DTPBMI<30] <- "Overweight"
select.meta.cat$BMI.class2[select.meta.cont$DTPBMI>=30] <- "Obese"

#Get paitent demographics data to be tested 
bmi <- select.meta.cont$DTPBMI
obese <- factor(select.meta.cat$obese)

######################################################################################## First Level Analysis & Alpha Diversity with BMI #############
###########################################################################

##Test BMI versus alpha diversity and phyla

HMPH <- anova(lm(H ~ obese)) #P-value=0.585
HMPS <- anova(lm(S ~ obese)) #P-value=0.845
HMPJ <- anova(lm(J ~ obese)) #P-value=0.3559


#B and F tests against obesity
bacter <- phyla.table.rel.abund$Bacteroidetes
firm <- phyla.table.rel.abund$Firmicutes
BFratio <- bacter/firm

HMPBacter <- anova(lm(bacter ~ obese)) #P-value=0.3518
HMPFirm <- anova(lm(firm ~ obese)) #P-value=0.636
HMPBF <- anova(lm(BFratio ~ obese)) #P-value=0.5688

###########################################################################
############ NMDS and PERMANOVA Analysis###################################
###########################################################################

set.seed(3)
HMP2 <- adonis(test ~ obese, permutations=1000) 
HMPPERM <- HMP2$aov.tab
#PERMANOVA=0.8112, pseudo-F=0.7024


######################################################################################## Relative Risk ###############################################
###########################################################################

#Generate median values and put them into existing alpha.test dataframe
#Shannon diversity

alpha.test <- within(alpha.test, {shannon.cat = ifelse(H <= median(H), "less", "higher")})


##Shannon Diversity
H.cat <- alpha.test$shannon.cat
bmi.cat <- as.character(obese)
test3 <- cbind(H.cat, bmi.cat)
test3 <- test3[order(H.cat), ]
orderedHCat <- test3[, 1]
HMPtotalN <- length(orderedHCat)
HMPHHighTotal <- length(orderedHCat[orderedHCat=="higher"])
HMPHighShannonGroup <- as.data.frame(table(test3[c(1:HMPHHighTotal), 2]))
HMPLowShannonGroup <- as.data.frame(table(test3[c((HMPHHighTotal + 1):HMPtotalN), 2]))
group1 <- c(HMPHighShannonGroup[2, 2], HMPHighShannonGroup[1, 2])
group2 <- c(HMPLowShannonGroup[2, 2], HMPLowShannonGroup[1, 2])
#Group1 (Higher than median), obese = 10 and non-obese = 118
#Group2 (Lower than median), obese = 16 and non-obese = 112

r.test <- rbind(group2, group1)
colnames(r.test) <- c("Obese", "Not.Obese")
rownames(r.test) <- c("group1", "group2")

HMPHEpi <- epi.2by2(r.test, method="cohort.count")
HMPHMassoc <- HMPHEpi$massoc
HMPHRR <- HMPHMassoc$RR.strata.score
HMPHRRsig <- HMPHMassoc$chisq.strata
## Risk Ratio = 1.60
## CI = 0.75, 3.39
## p-value = 0.214

##Run the RR for B/F ratio
Bacter = phyla.table.rel.abund$Bacteroidetes
Firm = phyla.table.rel.abund$Firmicutes
BFRatio = Bacter/Firm
BFRatio <- as.data.frame(BFRatio)

BFRatio <- within(BFRatio, {BFRatio.cat = ifelse(BFRatio <= median(BFRatio), "less", "higher")})

BFRatio.cat <- BFRatio$BFRatio.cat
test4 <- cbind(BFRatio.cat, obese)
test4 <- test4[order(BFRatio.cat), ]
orderedBFCat <- test4[, 1]
HMPBFHighTotal <- length(orderedBFCat[orderedBFCat=="higher"])
HMPHighBFGroup <- as.data.frame(table(test4[c(1:HMPBFHighTotal), 2]))
HMPLowBFGroup <- as.data.frame(table(test4[c((HMPBFHighTotal + 1):HMPtotalN), 2]))
group1 <- c(HMPHighBFGroup[2, 2], HMPHighBFGroup[1, 2])
group2 <- c(HMPLowBFGroup[2, 2], HMPLowBFGroup[1, 2])
#Group1 (Higher than median), obese = 13 and non-obese = 115
#Group2 (Lower than median), obese = 13 and non-obese = 115

r.test <- rbind(group2, group1)
colnames(r.test) <- c("Obese", "Not.Obese")
rownames(r.test) <- c("group2", "group1")


HMPBFEpi <- epi.2by2(r.test, method="cohort.count")
HMPBFMassoc <- HMPBFEpi$massoc
HMPBFRR <- HMPBFMassoc$RR.strata.score
HMPBFRRsig <- HMPBFMassoc$chisq.strata
## Risk Ratio = 1.00
## CI = 0.48, 2.07
## p-value = 1.00

###########################################################################
############ Classification using AUCRF ###################################
###########################################################################

#Create Obese.num group
select.meta.cat$obese.num[select.meta.cat$obese=="No"] <- 0
select.meta.cat$obese.num[select.meta.cat$obese=="Yes"] <- 1
obese <- factor(select.meta.cat$obese.num)

#generate test set
  # get rid of those with 0 and only 4 other values
testset <- Filter(function(x)(length(unique(x))>5), test)
testset <- cbind(obese, testset)
colnames(testset)[1] <- "obese"
testset <- cbind(testset, H, S, J, phyla.table.rel.abund)

#Try AUCRF with default measures provided in readme
#set.seed(3)
#HMPAUCFit <- AUCRF(obese ~ ., data=testset, ntree=1000, nodesize=20)
# list of 6 Measures, AUCopt = 0.7031773

```

```{r Wu_data, echo=FALSE, message=FALSE, warning=FALSE}
###########################################################################
############ Preparing Data Tables for Analysis ###########################
###########################################################################


setwd("C:/users/marc/Desktop/obesity2/COMBO/")

microbiome <- read.table("combined.good.subsample.shared", header=T)
rownames(microbiome) <- microbiome$Group
microbiome <- microbiome[, -c(1:3)]
seqData <- read.csv("COMBO_data_table.csv", header=T)
seqData <- seqData[-1, ]
rownames(seqData) <- seqData$Run_s
metadata <- read.table("BMI_and_Counts.txt", header=T)
metadata <- metadata[, c(1, 2)]

#get seqData set in line with microbiome data
namesToKeep <- rownames(microbiome)
test <- seqData[namesToKeep, ]
seqData <- test
rm(test)

#change all rownames to match sample IDs
rownames(seqData) <- seqData$submitted_subject_id_s
rownames(microbiome) <- rownames(seqData)
rownames(metadata) <- metadata[, 1]

#Match the metadata now with the microbiome data
namesToKeep <- rownames(microbiome)
test <- metadata[namesToKeep, ]
metadata <- test
rm(test)
rm(seqData, keep, namesToKeep)

#Get alpha diversity of the samples
library(vegan)

H <- diversity(microbiome)
S <- specnumber(microbiome)
J <- H/log(S)
alpha.diversity.shannon <- cbind(H,S,J)
alpha.test <- as.data.frame(alpha.diversity.shannon)

#Get phyla information
#Edited out non phyla information first with sed in linux
#combined new labels with previous taxonomy file with excel
phylogenetic.info <- read.table("combined.good.phyla.txt", header=T)
rownames(phylogenetic.info) <- phylogenetic.info[,1]
phylogenetic.info <- phylogenetic.info[,-c(1)]
phyla.names <- as.character(phylogenetic.info$Taxonomy)
keep <- colnames(microbiome)
phyla.good <- phylogenetic.info[keep, ]
phyla.names <- as.character(phyla.good[,2])
phyla.table <- microbiome
colnames(phyla.table) <- phyla.names
#add all the same columns up and then return the sum
testing <- t(rowsum(t(phyla.table), group = rownames(t(phyla.table))))
phyla.table <- as.data.frame(testing)
rm(testing)
#combine phyla that are not that abundant
phyla.table$other <- apply(phyla.table[, c("Fusobacteria", "Synergistetes", "TM7")], 1, sum)
phyla.table <- phyla.table[, -c(4, 6:7)]

#Create a relative abundance table for phyla
phyla.total <- apply(phyla.table[, c(1:6)], 1, sum)
phyla.table.rel.abund <- (phyla.table/phyla.total)*100
rm(keep, phyla.names, phyla.total, phylogenetic.info, phyla.table, phyla.good)

#Create BMI groups
metadata$BMI.class[metadata$BMI<=24] <- "Normal"
metadata$BMI.class[metadata$BMI>24 & metadata$BMI<30] <- "Overweight"
metadata$BMI.class[metadata$BMI>=30 & metadata$BMI<40] <- "Obese"
metadata$BMI.class[metadata$BMI>=40] <- "Extreme Obesity"

#Create Obese Yes/No groups
metadata$obese[metadata$BMI.class=="Normal" | metadata$BMI.class=="Overweight"] <- "No"
metadata$obese[metadata$BMI.class=="Obese" | metadata$BMI.class=="Extreme Obesity"] <- "Yes"

#Create a column with obese and extreme obese as single entity
metadata$BMIclass2[metadata$BMI.class=="Normal"] <- "Normal"
metadata$BMIclass2[metadata$BMI.class=="Overweight"] <- "Overweight"                     
metadata$BMIclass2[metadata$BMI.class=="Obese" | metadata$BMI.class=="Extreme Obesity"] <- "Obese"

#Get paitent demographics to be tested
bmi <- metadata$BMI
obese <- factor(metadata$obese)

###########################################################################
######### First Level Analysis & Alpha Diversity with BMI #################
###########################################################################

##Test BMI versus alpha diversity and phyla

WuH <- anova(lm(H ~ obese)) #P-value=0.7493
WuS <- anova(lm(S ~ obese)) #P-value=0.7511
WuJ <- anova(lm(J ~ obese)) #P-value=0.5557

#B and F tests against obesity
bacter <- phyla.table.rel.abund$Bacteroidetes
firm <- phyla.table.rel.abund$Firmicutes
BFratio <- bacter/firm

WuBacter <- anova(lm(bacter ~ obese)) #P-value=0.9173
WuFirm <- anova(lm(firm ~ obese)) #P-value=0.5765
WuBF <- anova(lm(BFratio ~ obese)) #P-value=0.6735

###########################################################################
############ NMDS and PERMANOVA Analysis###################################
###########################################################################

set.seed(3)
Wu2 <- adonis(microbiome ~ obese, permutations=1000)
WuPERM <- Wu2$aov.tab
#PERMANOVA=0.8921, pseudo-F=0.66065


###########################################################################
############ Relative Risk#################################################
###########################################################################

#Generate median values and put them into existing alpha.test dataframe
#Shannon diversity

alpha.test <- within(alpha.test, {shannon.cat = ifelse(H <= median(H), "less", "higher")})


##Shannon Diversity
H.cat <- alpha.test$shannon.cat
bmi.cat <- as.character(obese)
test3 <- cbind(H.cat, bmi.cat)
test3 <- test3[order(H.cat), ]
orderedHCat <- test3[, 1]
WutotalN <- length(orderedHCat)
WuHHighTotal <- length(orderedHCat[orderedHCat=="higher"])
WuHighShannonGroup <- as.data.frame(table(test3[c(1:WuHHighTotal), 2]))
WuLowShannonGroup <- as.data.frame(table(test3[c((WuHHighTotal + 1):WutotalN), 2]))
group1 <- c(WuHighShannonGroup[2, 2], WuHighShannonGroup[1, 2])
group2 <- c(WuLowShannonGroup[2, 2], WuLowShannonGroup[1, 2])
#Group1 (Higher than median), obese = 3 and non-obese = 26
#Group2 (Lower than median), obese = 2 and non-obese = 27

r.test <- rbind(group2, group1)
colnames(r.test) <- c("Obese", "Not.Obese")
rownames(r.test) <- c("group2", "group1")

WuHEpi <- epi.2by2(r.test, method="cohort.count")
WuHMassoc <- WuHEpi$massoc
WuHRR <- WuHMassoc$RR.strata.score
WuHRRsig <- WuHMassoc$chisq.strata
## Risk Ratio = 0.67
## CI = 0.12, 3.70
## p-value = 0.64

##Run the RR for B/F ratio
Bacter = phyla.table.rel.abund$Bacteroidetes
Firm = phyla.table.rel.abund$Firmicutes
BFRatio = Bacter/Firm
BFRatio <- as.data.frame(BFRatio)

BFRatio <- within(BFRatio, {BFRatio.cat = ifelse(BFRatio <= median(BFRatio), "less", "higher")})

BFRatio.cat <- BFRatio$BFRatio.cat
test4 <- cbind(BFRatio.cat, obese)
test4 <- test4[order(BFRatio.cat), ]
orderedBFCat <- test4[, 1]
WuBFHighTotal <- length(orderedBFCat[orderedBFCat=="higher"])
WuHighBFGroup <- as.data.frame(table(test4[c(1:WuBFHighTotal), 2]))
WuLowBFGroup <- as.data.frame(table(test4[c((WuBFHighTotal + 1):WutotalN), 2]))
group1 <- c(WuHighBFGroup[2, 2], WuHighBFGroup[1, 2])
group2 <- c(WuLowBFGroup[2, 2], WuLowBFGroup[1, 2])
#Group1 (Higher than median), obese = 3 and non-obese = 26
#Group2 (Lower than median), obese = 2 and non-obese = 27

r.test <- rbind(group2, group1)
colnames(r.test) <- c("Obese", "Not.Obese")
rownames(r.test) <- c("group2", "group1")

WuBFEpi <- epi.2by2(r.test, method="cohort.count")
WuBFMassoc <- WuBFEpi$massoc
WuBFRR <- WuBFMassoc$RR.strata.score
WuBFRRsig <- WuBFMassoc$chisq.strata
## Risk Ratio = 0.67
## CI = 0.12, 3.70
## p-value = 0.64

###########################################################################
############ Classification using AUCRF####################################
###########################################################################

#Create Obese.num group
metadata$obese.num[metadata$obese=="No"] <- 0
metadata$obese.num[metadata$obese=="Yes"] <- 1
obese <- factor(metadata$obese.num)

#generate test set
# get rid of those with 0 and only 4 other values
testset <- Filter(function(x)(length(unique(x))>5), microbiome)
testset <- cbind(obese, testset)
colnames(testset)[1] <- "obese"
testset <- cbind(testset, H, S, J, phyla.table.rel.abund)

#Try AUCRF with default measures provided in readme
#set.seed(3)
#WuAUCFit <- AUCRF(obese ~ ., data=testset, ntree=1000, nodesize=20)
# list of 15 Measures, AUCopt = 0.7075472

```

```{r Arumugam_data, echo=FALSE, message=FALSE, warning=FALSE}
###########################################################################
############ Preparing Data Tables for Analysis ###########################
###########################################################################

setwd("C:/users/marc/Desktop/obesity2/MetaHit/")

microb <- read.csv("finalized_metaHit_shared.csv")
rownames(microb) <- microb[, 1]
microb <- microb[,-1]

metadata <- read.csv("MetaHit_metadata.csv")
rownames(metadata) <- metadata[, 1]
metadata <- metadata[, -1]

phyla <- read.csv("phyla.csv")
rownames(phyla) <- phyla[, 1]
phyla <- phyla[, -1]

# Generate Overall Relative Abundance - normalize to a value of 100
# Needs to be done since not all measures have 100 bacteria from metaphlan2
overall <- rowSums(microb)
microb.norm <- (microb / overall) * 100

phyla.table <- read.csv("phyla.csv", header = T)
rownames(phyla.table) <- phyla.table[, 1]
phyla.table <- phyla.table[, -1]
phyla.table <- as.data.frame(t(phyla.table))

#Create a relative abundance table for phyla
phyla.total <- apply(phyla.table[, c(1:10)], 1, sum)
phyla.table.rel.abund <- (phyla.table/phyla.total)*100
rm(phyla.table, phyla.info)

#Get alpha diversity of the samples
library(vegan)

H <- diversity(microb.norm)
S <- specnumber(microb.norm)
J <- H/log(S)
alpha.diversity.shannon <- cbind(H,S,J)
alpha.test <- as.data.frame(alpha.diversity.shannon)
# Seems to look okay....

#Create BMI groups
metadata$BMI.class[metadata$BMI<=24] <- "Normal"
metadata$BMI.class[metadata$BMI>24 & metadata$BMI<30] <- "Overweight"
metadata$BMI.class[metadata$BMI>=30 & metadata$BMI<40] <- "Obese"
metadata$BMI.class[metadata$BMI>=40] <- "Extreme Obesity"

#Create Obese Yes/No groups
metadata$obese[metadata$BMI.class=="Normal" | metadata$BMI.class=="Overweight"] <- "No"
metadata$obese[metadata$BMI.class=="Obese" | metadata$BMI.class=="Extreme Obesity"] <- "Yes"


#Create Obese.num group
metadata$obese.num[metadata$obese=="No"] <- 0
metadata$obese.num[metadata$obese=="Yes"] <- 1


#Create a column with obese and extreme obese as single entity
metadata$BMIclass2[metadata$BMI<=24] <- "Normal"
metadata$BMIclass2[metadata$BMI>24 & metadata$BMI<30] <- "Overweight"      
metadata$BMIclass2[metadata$BMI>=30] <- "Obese"


#Get paitent demographics to be tested
bmi <- metadata$BMI
obese <- factor(metadata$obese)

####################################################################################### First Level Analysis & Alpha Diversity with BMI #############
###########################################################################
###########################################################################

##Test BMI versus alpha diversity and phyla

ArumugamH <- anova(lm(H ~ obese)) #P-value=0.2174
ArumugamS <- anova(lm(S ~ obese)) #P-value=0.9878
ArumugamJ <- anova(lm(J ~ obese)) #P-value=0.1874

#B and F tests against obesity
bacter <- phyla.table.rel.abund$Bacteroidetes
firm <- phyla.table.rel.abund$Firmicutes
BFratio <- bacter/firm

ArumugamBacter <- anova(lm(bacter ~ obese)) #P-value=0.1911
ArumugamFirm <- anova(lm(firm ~ obese)) #P-value=0.2925
ArumugamBF <- anova(lm(BFratio ~ obese)) #P-value=0.05254

####################################################################################### NMDS and PERMANOVA Analysis###################################
###########################################################################

set.seed(3)
Arumugam2 <- adonis(microb.norm ~ obese, permutations=1000) 
ArumugamPERM <- Arumugam2$aov.tab
#PERMANOVA=0.05495, pseudo-F=1.6047


###########################################################################
############ Relative Risk#################################################
###########################################################################

#Generate median values and put them into existing alpha.test dataframe
#Shannon diversity


alpha.test <- within(alpha.test, {shannon.cat = ifelse(H <= median(H), "less", "higher")})

##Shannon Diversity
H.cat <- alpha.test$shannon.cat
bmi.cat <- as.character(obese)
test3 <- cbind(H.cat, bmi.cat)
test3 <- test3[order(H.cat), ]
orderedHCat <- test3[, 1]
ArutotalN <- length(orderedHCat)
AruHHighTotal <- length(orderedHCat[orderedHCat=="higher"])
AruHighShannonGroup <- as.data.frame(table(test3[c(1:AruHHighTotal), 2]))
AruLowShannonGroup <- as.data.frame(table(test3[c((AruHHighTotal + 1):ArutotalN), 2]))
group1 <- c(AruHighShannonGroup[2, 2], AruHighShannonGroup[1, 2])
group2 <- c(AruLowShannonGroup[2, 2], AruLowShannonGroup[1, 2])
#Group1 (Higher than median), obese = 17 and non-obese = 25
#Group2 (Lower than median), obese = 20 and non-obese = 23

r.test <- rbind(group2, group1)
colnames(r.test) <- c("Obese", "Not.Obese")
rownames(r.test) <- c("group2", "group1")

ArumugamHEpi <- epi.2by2(r.test, method="cohort.count")
ArumugamHMassoc <- ArumugamHEpi$massoc
ArumugamHRR <- ArumugamHMassoc$RR.strata.score
ArumugamHRRsig <- ArumugamHMassoc$chisq.strata
## Risk Ratio = 1.15
## CI = 0.71, 1.87
## p-value = 0.575


##Run the RR for B/F ratio
Bacter = phyla.table.rel.abund$Bacteroidetes
Firm = phyla.table.rel.abund$Firmicutes
BFRatio = Bacter/Firm
BFRatio <- as.data.frame(BFRatio)

BFRatio <- within(BFRatio, {BFRatio.cat = ifelse(BFRatio <= median(BFRatio), "less", "higher")})

BFRatio.cat <- BFRatio$BFRatio.cat
test4 <- cbind(BFRatio.cat, obese)
test4 <- test4[order(BFRatio.cat), ]
orderedBFCat <- test4[, 1]
AruBFHighTotal <- length(orderedBFCat[orderedBFCat=="higher"])
AruHighBFGroup <- as.data.frame(table(test4[c(1:AruBFHighTotal), 2]))
AruLowBFGroup <- as.data.frame(table(test4[c((AruBFHighTotal + 1):ArutotalN), 2]))
group1 <- c(AruHighBFGroup[2, 2], AruHighBFGroup[1, 2])
group2 <- c(AruLowBFGroup[2, 2], AruLowBFGroup[1, 2])
#Group1 (Higher than median), obese = 20 and non-obese = 22
#Group2 (Lower than median), obese = 17 and non-obese = 26

r.test <- rbind(group2, group1)
colnames(r.test) <- c("Obese", "Not.Obese")
rownames(r.test) <- c("group2", "group1")

ArumugamBFEpi <- epi.2by2(r.test, method="cohort.count")
ArumugamBFMassoc <- ArumugamBFEpi$massoc
ArumugamBFRR <- ArumugamBFMassoc$RR.strata.score
ArumugamBFRRsig <- ArumugamBFMassoc$chisq.strata
## Risk Ratio = 0.83
## CI = 0.51, 1.35
## p-value = 0.452

###########################################################################
############ Classification using AUCRF ###################################
###########################################################################

library(AUCRF)

#Create Obese.num group
metadata$obese.num[metadata$obese=="No"] <- 0
metadata$obese.num[metadata$obese=="Yes"] <- 1
obese <- factor(metadata$obese.num)

#generate test set
# get rid of those with 0 and only 4 other values
testset <- Filter(function(x)(length(unique(x))>5), microb.norm)
testset <- cbind(obese, testset)
colnames(testset)[1] <- "obese"
testset <- cbind(testset, H, S, J, phyla.table.rel.abund)

#Try AUCRF with default measures provided in readme
#set.seed(3)
#ArumugamAUCFit <- AUCRF(obese ~ ., data=testset, ntree=1000, nodesize=20)
# list of 19 Measures, AUCopt = 0.7649212

```

```{r Turnbaugh_data, echo=FALSE, message=FALSE, warning=FALSE}
###########################################################################
############ Preparing Data Tables for Analysis ###########################
###########################################################################

setwd("C:/users/marc/Desktop/obesity2/turnbaugh.twins")
shared.data <- read.table("test.unique.good.filter.unique.precluster.pick.pick.an.shared", header=T)
rownames(shared.data) <- shared.data[, 2]
shared.data <- shared.data[, -c(1:3)]

subsample.data <- read.table("test.unique.good.filter.unique.precluster.pick.pick.an.0.03.subsample.shared", header=T)
rownames(subsample.data) <- subsample.data[, 2]
subsample.data <- subsample.data[, -c(1:3)]


metadata <- read.csv("turnbaugh.metadata.csv")
rownames(metadata) <- metadata[, 4]
metadata <- metadata[, -4]

keep1 <- which(metadata$Sample == 1) # n =146

#use the first sampling
#subset data for only the first sampling
s1.metadata <- metadata[keep1, ]
s1.subsample.data <- subsample.data[keep1, ]


#generate alpha diversity measures with vegan
library(vegan)
H <- diversity(s1.subsample.data)
S <- specnumber(s1.subsample.data)
J <- H/log(S)
select.alpha.diversity <- as.data.frame(cbind(H, S, J))
s1.alpha.diversity <- as.data.frame(select.alpha.diversity)
alpha.test <- s1.alpha.diversity

#Generate phyla table data

phyla.info <- read.csv("phyla.csv")
phyla.table <- shared.data
colnames(phyla.table) <- phyla.info[, 2]


#add all the same columns up and then return the sum
testing <- t(rowsum(t(phyla.table), group = rownames(t(phyla.table))))
phyla.table <- as.data.frame(testing)
rm(testing)

#combine phyla that are not that abundant
phyla.table$other <- apply(phyla.table[, c("Fusobacteria", "Lentisphaerae", "Spirochaetes", "Synergistetes", "TM7")], 1, sum)
phyla.table <- phyla.table[, -c(4:5, 7:9)]

#Create a relative abundance table for phyla
phyla.total <- apply(phyla.table[, c(1:7)], 1, sum)
phyla.table.rel.abund <- (phyla.table/phyla.total)*100
s1.phyla.rel.abund <- phyla.table.rel.abund[rownames(shared.data), ]
s1.phyla.rel.abund <- s1.phyla.rel.abund[keep1, ]

#Create Obese Yes/No groups
s1.metadata$obese[s1.metadata$BMI.category=="Lean" | s1.metadata$BMI.category=="Overweight"] <- "No"
s1.metadata$obese[s1.metadata$BMI.category=="Obese"] <- "Yes"

#Create Obese.num group
s1.metadata$obese.num[s1.metadata$obese=="No"] <- 0
s1.metadata$obese.num[s1.metadata$obese=="Yes"] <- 1

#create groups to be used
obese <- factor(s1.metadata$obese)
bmi <- s1.metadata$BMI.category

####################################################################################### First Level Analysis & Alpha Diversity with BMI #############
###########################################################################
###########################################################################

##Test BMI versus alpha diversity and phyla

turnbaughH <- anova(lm(H ~ obese)) #P-value=0.06976
turnbaughS <- anova(lm(S ~ obese)) #P-value=0.02435
turnbaughJ <- anova(lm(J ~ obese)) #P-value=0.1253

#B and F tests against obesity
bacter <- s1.phyla.rel.abund$Bacteroidetes
firm <- s1.phyla.rel.abund$Firmicutes
BFratio <- bacter/firm

turnbaughBacter <- anova(lm(bacter ~ obese)) #P-value=0.5771
turnbaughFirm <- anova(lm(firm ~ obese)) #P-value=0.4073
turnbaughBF <- anova(lm(BFratio ~ obese)) #P-value=0.2216

####################################################################################### NMDS and PERMANOVA Analysis###################################
###########################################################################

set.seed(3)
turnbaugh2 <- adonis(s1.subsample.data ~ obese, permutations=1000) 
turnbaughPERM <- turnbaugh2$aov.tab
#PERMANOVA=0.09491, pseudo-F=1.2114

###########################################################################
############ Relative Risk#################################################
###########################################################################

#Generate median values and put them into existing alpha.test dataframe
#Shannon diversity

alpha.test <- within(alpha.test, {shannon.cat = ifelse(H <= median(H), "less", "higher")})

##Shannon Diversity
H.cat <- alpha.test$shannon.cat
bmi.cat <- as.character(obese)
test3 <- cbind(H.cat, bmi.cat)
test3 <- test3[order(H.cat), ]
orderedHCat <- test3[, 1]
TurntotalN <- length(orderedHCat)
TurnHHighTotal <- length(orderedHCat[orderedHCat=="higher"])
TurnHighShannonGroup <- as.data.frame(table(test3[c(1:TurnHHighTotal), 2]))
TurnLowShannonGroup <- as.data.frame(table(test3[c((TurnHHighTotal + 1):TurntotalN), 2]))
group1 <- c(TurnHighShannonGroup[2, 2], TurnHighShannonGroup[1, 2])
group2 <- c(TurnLowShannonGroup[2, 2], TurnLowShannonGroup[1, 2])
#Group1 (Higher than median), obese = 47 and non-obese = 26
#Group2 (Lower than median), obese = 52 and non-obese = 21

r.test <- rbind(group2, group1)
colnames(r.test) <- c("Obese", "Not.Obese")
rownames(r.test) <- c("group2", "group1")

turnbaughHEpi <- epi.2by2(r.test, method="cohort.count")
turnbaughHMassoc <- turnbaughHEpi$massoc
turnbaughHRR <- turnbaughHMassoc$RR.strata.score
turnbaughHRRsig <- turnbaughHMassoc$chisq.strata
## Risk Ratio = 1.11
## CI = 0.88, 1.38
## p-value = 0.376


##Run the RR for B/F ratio
Bacter = s1.phyla.rel.abund$Bacteroidetes
Firm = s1.phyla.rel.abund$Firmicutes
BFRatio = Bacter/Firm
BFRatio <- as.data.frame(BFRatio)

BFRatio <- within(BFRatio, {BFRatio.cat = ifelse(BFRatio <= median(BFRatio), "less", "higher")})

BFRatio.cat <- BFRatio$BFRatio.cat
test4 <- cbind(BFRatio.cat, obese)
test4 <- test4[order(BFRatio.cat), ]
orderedBFCat <- test4[, 1]
TurnBFHighTotal <- length(orderedBFCat[orderedBFCat=="higher"])
TurnHighBFGroup <- as.data.frame(table(test4[c(1:TurnBFHighTotal), 2]))
TurnLowBFGroup <- as.data.frame(table(test4[c((TurnBFHighTotal + 1):TurntotalN), 2]))
group1 <- c(TurnHighBFGroup[2, 2], TurnHighBFGroup[1, 2])
group2 <- c(TurnLowBFGroup[2, 2], TurnLowBFGroup[1, 2])
#Group1 (Higher than median), obese = 54 and non-obese = 19
#Group2 (Lower than median), obese = 45 and non-obese = 28

r.test <- rbind(group2, group1)
colnames(r.test) <- c("Obese", "Not.Obese")
rownames(r.test) <- c("group2", "group1")

turnbaughBFEpi <- epi.2by2(r.test, method="cohort.count")
turnbaughBFMassoc <- turnbaughBFEpi$massoc
turnbaughBFRR <- turnbaughBFMassoc$RR.strata.score
turnbaughBFRRsig <- turnbaughBFMassoc$chisq.strata
## Risk Ratio = 0.83
## CI = 0.66, 1.05
## p-value = 0.111

###########################################################################
############ Classification using AUCRF ###################################
###########################################################################

#Create Obese.num group
#s1.metadata$obese.num[s1.metadata$obese=="No"] <- 0
#s1.metadata$obese.num[s1.metadata$obese=="Yes"] <- 1
#obese <- factor(s1.metadata$obese.num)

#generate test set
# get rid of those with 0 and only 4 other values
#testset <- Filter(function(x)(length(unique(x))>5), s1.subsample.data)
#testset <- cbind(obese, testset)
#colnames(testset)[1] <- "obese"
#testset <- cbind(testset, H, S, J, s1.phyla.rel.abund)

#Try AUCRF with default measures provided in readme
#set.seed(3)
#TurnbaughAUCFit <- AUCRF(obese ~ ., data=testset, ntree=1000, nodesize=20)
# list of 11 Measures, AUCopt = 0.7764883


```


```{r Table_3, echo=FALSE, message=FALSE, warning=FALSE}

Study <- c("Baxter", "Ross", "Goodrich", "Escobar", "Zupancic", "HMP", "Wu", "Arumugam", "Turnbaugh")

Bacteroidetes <- signif(c(baxterBacter[1,5], rossBacter[1,5], goodrichBacter[1,5], escobarBacter[1,5], zupancicBacter[1,5], HMPBacter[1,5], WuBacter[1,5], ArumugamBacter[1,5], turnbaughBacter[1,5]), 3)

Firmicutes <- signif(c(baxterFirm[1,5], rossFirm[1,5], goodrichFirm[1,5], escobarFirm[1,5], zupancicFirm[1,5], HMPFirm[1,5], WuFirm[1,5], ArumugamFirm[1,5], turnbaughFirm[1,5]), 3)

BFRatio <- signif(c(baxterBF[1,5], rossBF[1,5], goodrichBF[1,5], escobarBF[1,5], zupancicBF[1,5], HMPBF[1,5], WuBF[1,5], ArumugamBF[1,5], turnbaughBF[1,5]), 3)

Shannon <- signif(c(baxterH[1,5], rossH[1,5], goodrichH[1,5], escobarH[1,5], zupancicH[1,5], HMPH[1,5], WuH[1,5], ArumugamH[1,5], turnbaughH[1,5]), 3)

OTURich <- signif(c(baxterS[1,5], rossS[1,5], goodrichS[1,5], escobarS[1,5], zupancicS[1,5], HMPS[1,5], WuS[1,5], ArumugamS[1,5], turnbaughS[1,5]),3)

Evenness <- signif(c(baxterJ[1,5], rossJ[1,5], goodrichJ[1,5], escobarJ[1,5], zupancicJ[1,5], HMPJ[1,5], WuJ[1,5], ArumugamJ[1,5], turnbaughJ[1,5]), 3)

BrayC <- signif(c(baxterPERM[1,6], rossPERM[1,6], goodrichPERM[1,6], escobarPERM[1,6], zupancicPERM[1,6], HMPPERM[1,6], WuPERM[1,6], ArumugamPERM[1,6], turnbaughPERM[1,6]), 3)

table3 <- cbind(Study, Bacteroidetes, Firmicutes, BFRatio, Shannon, OTURich, Evenness, BrayC)

colnames(table3) <- c("Study", "Bacteroidetes", "Firmicutes", "B/F Ratio", "Shannon Diversity", "OTU Richness", "Evenness", "Bray Curtis")


kable(table3, format = "markdown", align = 'c', padding = 2)
```




*******
##### Figures:


![Figure 1](Sze_ObesityMicrobiomeMeta_PeerJ_2016_files/figure-html/PrismaFlowGraph.png)\


**Figure 1: PRISMA Flow Diagram of Total Records Searched.**  This was adapted from:  Moher D, Liberati A, Tetzlaff J, Altman DG, The PRISMA Group (2009). Preferred Reporting Items for Systematic Reviews and Meta-Analyses: The PRISMA Statement. PLoS Med 6(7): e1000097. doi:10.1371/journal.pmed1000097. 


```{r Figure_2, echo=FALSE, message=FALSE, warning=FALSE}

library(meta)
library(metafor)
library(ggplot2)
library(plyr)
library(reshape2)
library(RColorBrewer)
library(scales)
library(pheatmap)
library(gridExtra)

StudyNo <- c(1, 2, 3, 4, 5, 6, 7, 8, 9)
Study <- c("Baxter", "HMP", "Ross", "Escobar", "Arumugam", "Turnbaugh", "Zupancic", "Wu", "Goodrich")
Year <- c("2015", "2011", "2015", "2014", "2010", "2009", "2012", "2012", "2014")
Total <- c("n=172", "n=256", "n=63", "n=30", "n=85", "n=146", "n=199", "n=58", "n=507")
tpos <- c(BaxLowShannonGroup[2, 2], HMPLowShannonGroup[2, 2], RossLowShannonGroup[2, 2], EscoLowShannonGroup[2, 2], AruLowShannonGroup[2, 2], TurnLowShannonGroup[2, 2], ZupaLowShannonGroup[2, 2], WuLowShannonGroup[2, 2], GoodLowShannonGroup[2, 2]) #obese in low diversity group
tneg <- c(BaxLowShannonGroup[1, 2], HMPLowShannonGroup[1, 2], RossLowShannonGroup[1, 2], EscoLowShannonGroup[1, 2], AruLowShannonGroup[1, 2], TurnLowShannonGroup[1, 2], ZupaLowShannonGroup[1, 2], WuLowShannonGroup[1, 2], GoodLowShannonGroup[1, 2]) #nonobese in low diversity group
cpos <- c(BaxHighShannonGroup[2, 2], HMPHighShannonGroup[2, 2], RossHighShannonGroup[2, 2], EscoHighShannonGroup[2, 2], AruHighShannonGroup[2, 2], TurnHighShannonGroup[2, 2], ZupaHighShannonGroup[2, 2], WuHighShannonGroup[2, 2], GoodHighShannonGroup[2, 2]) #obese in high diversity group
cneg <- c(BaxHighShannonGroup[1, 2], HMPHighShannonGroup[1, 2], RossHighShannonGroup[1, 2], EscoHighShannonGroup[1, 2], AruHighShannonGroup[1, 2], TurnHighShannonGroup[1, 2], ZupaHighShannonGroup[1, 2], WuHighShannonGroup[1, 2], GoodHighShannonGroup[1, 2]) #nonobese in high diversity group
RR <- signif(c(baxterHRR[1,1], HMPHRR[1,1], rossHRR[1,1], escobarHRR[1,1], ArumugamHRR[1,1], turnbaughHRR[1,1], zupancicHRR[1,1], WuHRR[1,1], goodrichHRR[1,1]), 2)
low <- signif(c(baxterHRR[1,2], HMPHRR[1,2], rossHRR[1,2], escobarHRR[1,2], ArumugamHRR[1,2], turnbaughHRR[1,2], zupancicHRR[1,2], WuHRR[1,2], goodrichHRR[1,2]), 2)
high <- signif(c(baxterHRR[1,3], HMPHRR[1,3], rossHRR[1,3], escobarHRR[1,3], ArumugamHRR[1,3], turnbaughHRR[1,3], zupancicHRR[1,3], WuHRR[1,3], goodrichHRR[1,3]), 2)

numeric.data <-as.data.frame(cbind(tpos, tneg, cpos, cneg))
raw.overall <- as.data.frame(cbind(StudyNo, Study,Year, Total, numeric.data))

dat <- escalc(measure = "RR", ai = tpos, bi = tneg, ci = cpos, di = cneg, data=raw.overall, append=TRUE)
k <- length(dat$StudyNo)
dat.fm <- data.frame(study=factor(rep(1:k, each = 4)))
dat.fm$grp <- factor(rep(c("T", "T", "C", "C"), k), levels = c("T", "C"))
dat.fm$out <- factor(rep(c("+", "-", "+", "-"), k), levels = c("+", "-"))
dat.fm$freq <- with(raw.overall, c(rbind(tpos, tneg, cpos, cneg)))

#Used to generate the RE Model estimate based on all the studies
res <- rma(ai=tpos, bi=tneg, ci=cpos, di=cneg, data=raw.overall, measure="RR", slab=paste(Study, Year, Total, sep=", "), method="REML")


#To create a graph visualize the data
forest(res, at=log(c(0.3, 1, 5)), atransf=exp, xlab="RR of Obesity in high versus low Shannon diversity based on Median", cex=0.8)
par(font=2) #Switch to bold

text(5.6, 10.3, "Relative Risk [95% CI]", pos=2, cex=0.8)
text(-6.5, 10.3, "Study, Year, and n", pos=4, cex=0.8) 

```

**Figure 2: Meta Analysis of the Relative Risk of Obesity Based on Shannon Diversity.**  Groups were divided for each study on high and low Shannon Diversity groups based on the median for that study.  The overall pooled relative risk was 1.18 for the low diversity group (P-value = 0.0187).


```{r Figure_3, echo=FALSE, message=FALSE, warning=FALSE}

StudyNo <- c(1, 2, 3, 4, 5, 6, 7, 8, 9)
Study <- c("Baxter", "HMP", "Ross", "Escobar", "Arumugam", "Turnbaugh", "Zupancic", "Wu", "Goodrich")
Year <- c("2015", "2011", "2015", "2014", "2010", "2009", "2012", "2012", "2014")
Total <- c("n=172", "n=256", "n=63", "n=30", "n=85", "n=146", "n=199", "n=58", "n=507")
tpos <- c(BaxLowBFGroup[2, 2], HMPLowBFGroup[2, 2], RossLowBFGroup[2, 2], EscoLowBFGroup[2, 2], AruLowBFGroup[2, 2], TurnLowBFGroup[2, 2], ZupaLowBFGroup[2, 2], WuLowBFGroup[2, 2], GoodLowBFGroup[2, 2]) #obese in low BF group
tneg <- c(BaxLowBFGroup[1, 2], HMPLowBFGroup[1, 2], RossLowBFGroup[1, 2], EscoLowBFGroup[1, 2], AruLowBFGroup[1, 2], TurnLowBFGroup[1, 2], ZupaLowBFGroup[1, 2], WuLowBFGroup[1, 2], GoodLowBFGroup[1, 2]) #nonobese in low BF group
cpos <- c(BaxHighBFGroup[2, 2], HMPHighBFGroup[2, 2], RossHighBFGroup[2, 2], EscoHighBFGroup[2, 2], AruHighBFGroup[2, 2], TurnHighBFGroup[2, 2], ZupaHighBFGroup[2, 2], WuHighBFGroup[2, 2], GoodHighBFGroup[2, 2]) #obese in high BF group
cneg <- c(BaxHighBFGroup[1, 2], HMPHighBFGroup[1, 2], RossHighBFGroup[1, 2], EscoHighBFGroup[1, 2], AruHighBFGroup[1, 2], TurnHighBFGroup[1, 2], ZupaHighBFGroup[1, 2], WuHighBFGroup[1, 2], GoodHighBFGroup[1, 2]) #nonobese in high BF group
RR <- signif(c(baxterBFRR[1,1], HMPBFRR[1,1], rossBFRR[1,1], escobarBFRR[1,1], ArumugamBFRR[1,1], turnbaughBFRR[1,1], zupancicBFRR[1,1], WuBFRR[1,1], goodrichBFRR[1,1]), 2)
low <- signif(c(baxterBFRR[1,2], HMPBFRR[1,2], rossBFRR[1,2], escobarBFRR[1,2], ArumugamBFRR[1,2], turnbaughBFRR[1,2], zupancicBFRR[1,2], WuBFRR[1,2], goodrichBFRR[1,2]), 2)
high <- signif(c(baxterBFRR[1,3], HMPBFRR[1,3], rossBFRR[1,3], escobarBFRR[1,3], ArumugamBFRR[1,3], turnbaughBFRR[1,3], zupancicBFRR[1,3], WuBFRR[1,3], goodrichBFRR[1,3]), 2)

numeric.data <-as.data.frame(cbind(tpos, tneg, cpos, cneg))
raw.overall <- as.data.frame(cbind(StudyNo, Study,Year, Total, numeric.data))

dat <- escalc(measure = "RR", ai = tpos, bi = tneg, ci = cpos, di = cneg, data=raw.overall, append=TRUE)
k <- length(dat$StudyNo)
dat.fm <- data.frame(study=factor(rep(1:k, each = 4)))
dat.fm$grp <- factor(rep(c("T", "T", "C", "C"), k), levels = c("T", "C"))
dat.fm$out <- factor(rep(c("+", "-", "+", "-"), k), levels = c("+", "-"))
dat.fm$freq <- with(raw.overall, c(rbind(tpos, tneg, cpos, cneg)))

#Used to generate the RE Model estimate based on all the studies
res2 <- rma(ai=tpos, bi=tneg, ci=cpos, di=cneg, data=raw.overall, measure="RR", slab=paste(Study, Year, Total, sep=", "), method="REML")


#To create a graph visualize the data
forest(res2, at=log(c(0.1, 1, 3)), atransf=exp, xlab="RR of Obesity in high versus low B/F ratio based on Median", cex=0.8)
par(font=2) #Switch to bold

text(5.6, 10.3, "Relative Risk [95% CI]", pos=2, cex=0.8)
text(-6.0, 10.3, "Study, Year, and n", pos=4, cex=0.8)
```

**Figure 3: Meta Analysis of the Relative Risk of Obesity Based on Bacteroidetes/Firmicutes Ratio.**  Groups were divided for each study on high and low B/F ratio groups based on the median for that specific study.  The overall pooled relative risk was 0.86 for the low diversity group (P-value = 0.0376).


```{r Figure_4, echo=FALSE}
par(mfrow=c(1,2))
funnel(res, main="A")
funnel(res2, main="B")
```

**Figure 4: Funnel Plot of the Shannon Diversity and Bacteroidetes/Firmicutes Ratio Relative Risk.**  **A)** Overall there does not seem to be any bias associated with the studies selected for the Shannon Diversity analysis with studies falling on either side of the predicted value and those with smaller total n falling further away from this. **B)** For the Bacteroidetes/Firmicutes ratio there does not seem to be any associated bias.  The overall pattern is similar to that observed for the Shannon Diversity analysis.


```{r Data_Load_Fig4, echo=FALSE, message=FALSE, warning=FALSE}


setwd("C:/Users/marc/Desktop/obesity2/BMI.normalization")

#Load in BMI data
data1 <- read.csv("combined.H.corr.data.csv")
#Load in Turnbaugh data
turn.data <- read.csv("Turnbaugh.normalized.csv")

#Turnbaugh STD = 0.4818751
turn.ZScore.H <- turn.data$H.corr/0.4818751

#Amish STD = 0.5932651
keepamish <- which(data1$dataset == "Amish")
amishdata <- data1[keepamish, ]
amish.ZScore.H <- amishdata$H.corr/0.5932651

#MetaHit-Den STD = 0.4038558
keepmetahit <- which(data1$dataset == "MetaHit")
metahitdata <- data1[keepmetahit, ]
metahit.ZScore.H <- metahitdata$H.corr/0.4038558

#Korean STD = 0.5400615
keepKorean <- which(data1$dataset == "Korean")
Koreandata <- data1[keepKorean, ]
Korean.ZScore.H <- Koreandata$H.corr/0.5400615

#HMP STD = 0.6774514
keepHMP <- which(data1$dataset == "HMP")
hmpdata <- data1[keepHMP, ]
HMP.ZScore.H <- hmpdata$H.corr/0.6774514

#CCHC STD = 0.5784522
keepCCHC <- which(data1$dataset == "CCHC")
CCHCdata <- data1[keepCCHC, ]
CCHC.ZScore.H <- CCHCdata$H.corr/0.5784522

#GLNE07 STD = 0.4663871
keepGLNE <- which(data1$dataset == "GLNE")
GLNEdata <- data1[keepGLNE, ]
GLNE.ZScore.H <- GLNEdata$H.corr/0.4663871

#Escobar STD = 0.5378935
keepescobar <- which(data1$dataset == "Escobar")
escobardata <- data1[keepescobar, ]
Escobar.ZScore.H <- escobardata$H.corr/0.5378935

#COMBO STD = 0.6870763
keepCOMBO <- which(data1$dataset == "COMBO")
COMBOdata <- data1[keepCOMBO, ]
COMBO.ZScore.H <- COMBOdata$H.corr/0.6870763

#COMBO STD = 0.3489582
keeptwinsUK <- which(data1$dataset == "twinsUK")
twinsUKdata <- data1[keeptwinsUK, ]
twinsUK.ZScore.H <- twinsUKdata$H.corr/0.3489582


#data1 order Escobar, GLNE, CCHC, HMP, Korean, MetaHit, COMBO, twinsUK
HZScore <- c(Escobar.ZScore.H, GLNE.ZScore.H, CCHC.ZScore.H, HMP.ZScore.H, Korean.ZScore.H, metahit.ZScore.H, COMBO.ZScore.H, twinsUK.ZScore.H, amish.ZScore.H)
data1 <- cbind(HZScore, data1)

turn.data <- cbind(turn.ZScore.H, turn.data)
rm(Escobar.ZScore.H, GLNE.ZScore.H, CCHC.ZScore.H, HMP.ZScore.H, Korean.ZScore.H, metahit.ZScore.H, amish.ZScore.H, turn.ZScore.H, COMBO.ZScore.H, twinsUK.ZScore.H)
rm(keepmetahit, keepCCHC, keepescobar, keepGLNE, keepHMP, keepKorean)

#Create BMI groups
data1$BMI.class[data1$bmi<=24] <- "Lean"
data1$BMI.class[data1$bmi>24 & data1$bmi<30] <- "Overweight"
data1$BMI.class[data1$bmi>=30] <- "Obese"

#Create Obese Yes/No groups
data1$obese[data1$BMI.class=="Lean" | data1$BMI.class=="Overweight"] <- "No"
data1$obese[data1$BMI.class=="Obese"] <- "Yes"

#Create group analysis table to harness full data set
#For ZScore normalization
d1.obese <- data1$obese
t.obese <- as.character(turn.data$obese)

d1.bmi.class <- data1$BMI.class
t.bmi.class <- as.character(turn.data$BMI.category)

d1.H.Zscore <- data1$HZScore
t.H.Zscore <- turn.data$turn.ZScore.H

#Combine everything together
comb1.obese <- c(d1.obese, t.obese)
comb1.bmi.class <- c(d1.bmi.class, t.bmi.class)
comb1.H.Zscore <- as.data.frame(c(d1.H.Zscore, t.H.Zscore))

group.data.test1 <- cbind(comb1.H.Zscore, comb1.obese, comb1.bmi.class)
colnames(group.data.test1) <- c("H.Zscore", "obese", "BMI.class")

setwd("C:/Users/marc/Desktop/obesity2/BF.normalization")

#Let's change the order so it makes sense
group.data.test1$BMI.class <- factor(group.data.test1$BMI.class, levels = c("Lean", "Overweight", "Obese"))

group.data.test <- group.data.test1

#Load in data
data2 <- read.csv("combined.BF.corr.csv")

#Create BMI groups
data2$BMI.class[data2$bmi<=24] <- "Lean"
data2$BMI.class[data2$bmi>24 & data2$bmi<30] <- "Overweight"
data2$BMI.class[data2$bmi>=30] <- "Obese"

#Create Obese Yes/No groups
data2$obese[data2$BMI.class=="Lean" | data2$BMI.class=="Overweight"] <- "No"
data2$obese[data2$BMI.class=="Obese"] <- "Yes"

##Run Zscore normalized Bacter and Firm overall analysis
#data1 order Escobar, GLNE, CCHC, HMP, Korean, MetaHit
d1.zbf.corr <- data2$ZBF
t.zbf.corr <- turn.data$ZBF

#Bacteroidetes/Firmicutes Datasets
comb.zbf.corr <- as.data.frame(c(d1.zbf.corr, t.zbf.corr))
group.data.test <- cbind(comb.zbf.corr, group.data.test)
colnames(group.data.test)[1] <- "zbf"


```



```{r Figure_5, echo=FALSE, message=FALSE, warning=FALSE}
# Section A
p1 <- ggplot(group.data.test, aes(obese, H.Zscore)) 
p2 <- p1 + geom_boxplot(aes(fill = obese)) + scale_fill_brewer(palette="Set1")
p3 <- p2 + theme(axis.text=element_text(size=8), axis.title=element_text(size=10, face="bold"))
p4 <- p3 + xlab("Obese Category") + ylab("ZScore Normalized H") 
p5 <- p4 + guides(fill=guide_legend(title="Obese")) + theme(legend.title = element_text(size=8, face="bold"))
p6 <- p5 + theme(axis.line=element_line(colour="black"), 
                 panel.grid.major = element_blank(), 
                 panel.grid.minor = element_blank(), 
                 panel.border = element_blank(), 
                 panel.background = element_blank())
p7 <- p6 + ggtitle('A') + theme(plot.title=element_text(hjust=0, face="bold", size=20))

# Section B
q1 <- ggplot(group.data.test, aes(BMI.class, H.Zscore)) 
q2 <- q1 + geom_boxplot(aes(fill = BMI.class)) + scale_fill_brewer(palette="Set1")
q3 <- q2 + theme(axis.text=element_text(size=8), axis.title=element_text(size=10, face="bold"))
q4 <- q3 + xlab("BMI Classification") + ylab("ZScore Normalized H") 
q5 <- q4 + guides(fill=guide_legend(title="BMI Class")) + theme(legend.title = element_text(size=8, face="bold"))
q6 <- q5 + theme(axis.line=element_line(colour="black"), 
                 panel.grid.major = element_blank(), 
                 panel.grid.minor = element_blank(), 
                 panel.border = element_blank(), 
                 panel.background = element_blank())
q7 <- q6 + ggtitle('B') + theme(plot.title=element_text(hjust=0, face="bold", size=20))

# Section C
bf1 <- ggplot(group.data.test, aes(obese, zbf)) 
bf2 <- bf1 + geom_boxplot(aes(fill = obese)) + scale_fill_brewer(palette="Set1")
bf3 <- bf2 + theme(axis.text=element_text(size=8), axis.title=element_text(size=10, face="bold"))
bf4 <- bf3 + xlab("Obese Category") + ylab("ZScore Normalized B/F Ratio") 
bf5 <- bf4 + guides(fill=guide_legend(title="Obese")) + theme(legend.title = element_text(size=8, face="bold"))
bf6 <- bf5 + theme(axis.line=element_line(colour="black"), 
                   panel.grid.major = element_blank(), 
                   panel.grid.minor = element_blank(), 
                   panel.border = element_blank(), 
                   panel.background = element_blank())
bf7 <- bf6 + ggtitle('C') + theme(plot.title=element_text(hjust=0, face="bold", size=20))

# Section D
qbf1 <- ggplot(group.data.test, aes(BMI.class, zbf)) 
qbf2 <- qbf1 + geom_boxplot(aes(fill = BMI.class)) + scale_fill_brewer(palette="Set1")
qbf3 <- qbf2 + theme(axis.text=element_text(size=8), axis.title=element_text(size=10, face="bold"))
qbf4 <- qbf3 + xlab("BMI Classification") + ylab("ZScore Normalized B/F Ratio") 
qbf5 <- qbf4 + guides(fill=guide_legend(title="BMI Class")) + theme(legend.title = element_text(size=8, face="bold"))
qbf6 <- qbf5 + theme(axis.line=element_line(colour="black"), 
                     panel.grid.major = element_blank(), 
                     panel.grid.minor = element_blank(), 
                     panel.border = element_blank(), 
                     panel.background = element_blank())
qbf7 <- qbf6 + ggtitle('D') + theme(plot.title=element_text(hjust=0, face="bold", size=20))

grid.arrange(p7, q7, bf7, qbf7)
```


**Figure 5: Pooled Analysis of Shannon Diversity and B/F ratio by either Obesity Status or BMI Groups.**  **A)** ZScore Normalized Shannon Diversity and Obesity.  There was a significant difference between non-obese (n=1099) and obese groups (n=435) (P-value = 0.005992).  **B)** ZScore Normalized Shannon Diversity by BMI Group.  Total n for lean, overweight, and obese groups are 575, 435, 524 respectively.  Using an ANOVA with tukey post-hoc testing there was a significant diference between the overweight and obese group (P-value = 0.018) but no difference between lean and obese group (P-value = 0.081).  **C)** ZScore Normalized B/F Ratio and Obesity.  There was no significant difference between the non-obese (n=1099) and obese (n=435) group based on B/F ratio (P-value = 0.456).  **D)** ZScore Normalized B/F Ratio by BMI Group.  Total n for lean, overweight, and obese groups are 575, 435, 524 respectively.  There was no significant difference between any of the three groups based on ANOVA with Tukey post-hoc testing (P-value = 0.398).  



*******
##### References:














