---
title: "Bacterial Microbiome and Obesity Meta-analysis Sequence Processing Pipeline"
author: "Marc Sze"
date: "January 26, 2016"
output: 
  html_document:
    keep_md: yes
bibliography: ../metaAnalysis.obesity.microbiome.bib
---
#### **Preamble**  
Each study is brought up one at a time.  This recount starts from time of download all the way to generation of the final shared file used for the meta-analysis in the study.  You will also need the latest version of mothur, SRA Toolkit, and Python 3+ installed on your system.  There are also a few reference files that you will need.  The first is the [SILVA reference](http://www.mothur.org/wiki/Silva_reference_files) database and the second is the [RDP database](http://www.mothur.org/wiki/RDP_reference_files) of which we used release 102 and version 9 respectively.  In general, we tried to keep as close as possible to the Schloss SOP for processsing sequence reads for either 454 or the MiSeq.  Most of the differences between the SOP as written and what was performed involved the use of flowgrams.  Most of the data in the SRA were stored in the Fastq format and this step was not possible.  The batch files can be found in the data/process directory under the respective study directory.
<br>
<br>

##### **Baxter et al. 2016:**
This data set consisted of controls without colorectal cancer (CRC) and those with CRC.  The sequence reads of only the controls who did not have CRC were obtained directly from the first author since the data was generated within the Schloss lab.  However, the sequences can be found on the SRA under the accession number SRP062005. Only the raw 16S sequence reads for normal controls were processed  from the Sequence Read Archive (SRA) and subsequently used in the analysis.  So here we go....
<br>
<br>
I'm sure a smarter person could seperate the specific files that are normal only but I'm lazy and we had the storage space so I downloaded the entire data set.  
<br>
```{r, engine='bash', eval=FALSE}
wget -r -np -nd -k ftp://ftp-trace.ncbi.nih.gov/sra/sra-instant/reads/ByStudy/sra/SRP/SRP062/SRP062005/

fastq-dump \*.sra
```
<br><br>
The metadata can be found and downloaded by searching SRP062005 at the following site: http://www.ncbi.nlm.nih.gov/Traces/study/
<br><br>
The simple python program (Python 3.0) used to generate the .file file needed to make contigs can be found in this repository titled: *MSze.makefile.good.trial2.py* and the resulting file is titled *mothur.data.file*.  All you need to make this work is to have only the necessary fastq files in the respective folder.
<br><br>
Before getting started enter mothur and curate the silva.fasta for the v4 region with the following commands:

```{r, engine='bash', eval=FALSE}
pcr.seqs(fasta=silva.bacteria.fasta, start=11894, end=25319, keepdots=F)
system(mv silva.bacteria.pcr.fasta silva.v4.fasta)
```

Next we utilized the batch mode process and can come back the next day to see everything completed.  Our command was the following:

```{r, engine='bash', eval=FALSE}
mnt/EXT/Schloss-data/bin/mothur Baxter.batch
```


##### **Ross et al. 2015:**
This data set consisted of normal individuals from the Cameron Count Hispanic Cohort (CCHC).  The paper for this data set can be found [here](http://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-015-0072-y).  The project accession number in the SRA is SRP053023.  We can download the entire data set again with the code mentioned above but with one small substitution at the ending. 

```{r, engine='bash', eval=FALSE}
wget -r -np -nd -k ftp://ftp-trace.ncbi.nih.gov/sra/sra-instant/reads/ByStudy/sra/SRP/SRP053/SRP053023/
```

Now since there was no sff file provided in this data set even though it is a 454 data set (I found this out when I tried to run the command -> sff-dump \*.sra).  We have to use the fastq function of the SRA Tool kit.
<br>
```{r, engine='bash', eval=FALSE}
fastq-dump \*.sra
```
<br>
Because of this difference there are a few changes in this processing versus the standardized mothur 454 protocol.  First, instead of the sff.info() command we have to use the fastq.info() command.  Second, and perhaps more importantly, because we don't have the sff file we have to use the quality scores.  We used the author specified defaults of *qwindowaverage=25* and *qwindowsize=50*.  It should also be noted that we had *maxambig=5* and *maxhomop=8*.  Since the data come seperated already we do not need an oligo file and all we have to do is loop the mothur *trim.seqs* command. You can run a bash or shell program to get all of this done with either *#!/bin/sh* or *#!/bin/bash*.  So in a text editor:    

```{r, engine='sh', eval=FALSE}
#!/bin/sh  
for sample in $(ls *sra)  
do  
  fastq-dump $sample  
done

for sample2 in $(ls *fastq)  
do  
  mnt/EXT/Schloss-data/bin/mothur "#fastq.info(fastq=$sample2)"  
  rm *logfile  
done

for TRIM in $(ls ERR*.fasta)  
do  
  QFILE=${TRIM//fasta/qual}  
  mnt/EXT/Schloss-data/bin/mothur "#trim.seqs(fasta=$TRIM, qfile=$QFILE, maxambig=5, maxhomop=8, qwindowaverage=25, qwindowsize=50, processors=8)"  
done  

rm *.fastq
```

The next lines of code are used to combine the fasta file, create a names file with the mothur unique command, and create a combined group file for subsequent processing.

```{r, engine='bash', eval=FALSE}
cat \*trim.fasta > combined.trim.fasta  
mnt/EXT/Schloss-data/bin/mothur "#unique.seqs(fasta=combined.trim.fasta)"  

grep -e '^>' combined.trim.fasta > test.fasta  
sed 's/>//' test.fasta > test2.fasta  
sed 's/\./_/g' test2.fasta > test3.fasta  
sed 's/_.*//' test3.fasta > test4.fasta  
paste -d" " test2.fasta test4.fasta > combined.groups  

exit 0
```

After this little program is run we follow the mothur 454 SOP pretty much as  written and can use the following command to complete the processing using the put together *Ross.batch* file and running mothur.  Moving forward for the screen.seqs command we use a minlength cutoff of 200 in addition to what is written in the 454 SOP for all data sets.   

```{r, engine='bash', eval=FALSE}
mnt/EXT/Schloss-data/bin/mothur Ross.batch
```

That completes the processing for this data set.  Simple so far I hope.  On to the next data set.
<br><br>

##### **Goodrich et al. 2014:**  
This is a second larger twin data set and the paper can be found [here](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4255478/).  The individuals that participated in the study were part of the TwinsUK registry and is somewhat heavily skewed toward women.  The raw sequence data can be found on the SRA under ERP006339.  So using wget we can download the data with the following code:

```{r, engine='bash', eval=FALSE}
wget -r -np -nd -k ftp://ftp-trace.ncbi.nih.gov/sra/sra-instant/reads/ByStudy/sra/ERP/ERP006/ERP006339/

```

I started off with analyzing the data separately while sticking the the default mothur SOP for MiSeq processing.....

```{r, engine='sh', eval=FALSE}
#!/bin/sh  
# Unpack sra files  
# mkdir sra.complete  
for start in $(ls *sra)  
do  
  fastq-dump $start  
  mv $start sra.complete  
done

#Unpack fastq files  
mkdir fastq.complete  
mkdir logfile.complete  
for info in $(ls *fastq)  
do  
	mnt/EXT/Schloss-data/bin/mothur "#fastq.info(fastq=$info)"  
	mv $info fastq.complete  
	mv *.logfile logfile.complete  
done

#Trim the fasta file that came from the fastq  
mkdir fasta.trimming.complete  
for next in $(ls *fasta)  
do  
	QualFile=${next//fasta/qual}
	/mnt/EXT/Schloss-data/bin/mothur "#trim.seqs(fasta=$next, qfile=$QualFile, maxambig=0, maxhomop=8, qwindowaverage=35, qwindowsize=50)"  
	mv $next fasta.trimming.complete  
	mv $QualFile fasta.trimming.complete
	mv *.logfile logfile.complete  
done

#Create a groups file for each sample
for sample2 in $(ls *trim.fasta)
do
	groupfile=${sample2//fasta/groups}
	grep -e '^>' $sample2 > test.fasta  
	sed 's/>//' test.fasta > test2.fasta  
	sed 's/\..*$//' test2.fasta > test3.fasta  
	paste -d"\t" test2.fasta test3.fasta > ${groupfile}

done

#Make the combined fasta file  
mkdir combined.complete  
cat *trim.fasta > combined.fasta  
mv *trim.fasta combined.complete
cat *trim.groups > combined.groups
mv *trim.groups combined.complete
```

From here you can run a batch file with mothur to get the final output files.

```{r, engine='bash', eval=FALSE}
/share/scratch/schloss/mothur/mothur Goodrich.batch
```


<br><br>

##### **Escobar et al. 2014:**
This data set consisted of normal individuals from Medellin, Colombia South America.  The paper for this data set can be found [here](http://bmcmicrobiol.biomedcentral.com/articles/10.1186/s12866-014-0311-6).  The project accession number in the ENA is ERP003466.  We can download the entire data set again with the code mentioned above but with a few substitutions versus the previous two studies. 

```{r, engine='bash', eval=FALSE}
wget -r -np -nd -k ftp://ftp-trace.ncbi.nih.gov/sra/sra-instant/reads/ByStudy/sra/ERP/ERP003/ERP003466/
```

Now just like the previous data set there was no sff file provided in this data set even though it is a 454 data set (I found this out, again, when I tried to run the command -> sff-dump \*.sra).  We have to use the fastq function of the SRA Tool kit.

```{r, engine='bash', eval=FALSE}
fastq-dump *.sra
```

Because of this difference there are a few changes in this processing versus the standardized mothur 454 protocol.  First, instead of the sff.info() command we have to use the fastq.info() command.  Second, and perhaps more importantly, because we don't have the sff file we have to use the quality scores.  Unlike the Ross, et al study we used the mothur specified defaults of *qwindowaverage=35* and *qwindowsize=50*.  Since the data come seperated already we do not need an oligo file and all we have to do is loop the mothur *trim.seqs* command. You can run a bash or shell program to get all of this done with either *#!/bin/sh* or *#!/bin/bash*.  So in a text editor:    

```{r, engine='sh', eval=FALSE}
#!/bin/sh  
for sample in $(ls *sra)  
do  
  fastq-dump $sample  
done

for sample2 in $(ls *fastq)  
do  
  mnt/EXT/Schloss-data/bin/mothur "#fastq.info(fastq=$sample2)"  
  rm *logfile  
done
<br><br>
for TRIM in $(ls SRR*.fasta)  
do  
  QFILE=\${TRIM//fasta/qual}  
mnt/EXT/Schloss-data/bin/mothur "#trim.seqs(fasta=$TRIM, qfile=$QFILE, maxambig=0, maxhomop=8, qwindowaverage=35, qwindowsize=50, processors=8)"  
done  

rm *.fastq
```

The next lines of code are used to combine the fasta file, create a names file with the mothur unique command, and create a combined group file for subsequent processing.

```{r, engine='bash', eval=FALSE}
cat *trim.fasta > combined.trim.fasta  

mnt/EXT/Schloss-data/bin/mothur "#unique.seqs(fasta=combined.trim.fasta)"  

grep -e '^>' combined.trim.fasta > test.fasta  
sed 's/>//' test.fasta > test2.fasta  
sed 's/\./_/g' test2.fasta > test3.fasta  
sed 's/_.*//' test3.fasta > test4.fasta  
paste -d" " test2.fasta test4.fasta > combined.groups  
exit 0
```

After this little program is run we follow the mothur 454 SOP as written and can use the following command to complete the processing using the put together *Escobar.batch* file and running mothur.  Again like the Ross et al data set one final difference from the standardized mothur approach that was made was during the screen.seqs command after alignment a hard cutoff *minlength=200* and *maxlength=250* was used.

```{r, engine='bash', eval=FALSE}
mnt/EXT/Schloss-data/bin/mothur Escobar.batch
```

That completes the processing for this data set.  So far so good, no major problems.  On to the next data set.
<br><br>

##### **Zupancic et al. 2012:**
This data set consisted of normal individuals from an Amish community living in Lancaster County, Pennsylvania.  The paper for this data set can be found [here](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.00430526).  The project accession number in the SRA is SRP002465.  We can download the entire data set again with the code mentioned above but with a few substitutions.  Here we go......
```{r, engine='bash', eval=FALSE}
wget -r -np -nd -k ftp://ftp-trace.ncbi.nih.gov/sra/sra-instant/reads/ByStudy/sra/SRP/SRP002/SRP002465/
```

Now this data set included the sff files so we can use sffinfo function and start from the very start of the mothur 454 pipleine.  However, to keep things consistent we will keep with the fastq approach that we have been using from the previous data sets.  The below program was used and you can use either *#!/bin/sh* or *#!/bin/bash* and text editor to create it and then run it.  

```{r, engine='bash', eval=FALSE}
#!/bin/bash

# Make a directory to store the reference files
# Remember to move these back into the necessary directory later
# When you get to the batch file point,
mkdir reference.files
mv silva.bacteria.fasta reference.files
mv trainset9_032012.pds.tax reference.files
mv trainset9_032012.pds.fasta reference.files

# Read in the sra and make a fastq instead of a sff.
mkdir completed.redo.sra
mkdir logfile.completed

for sample in $(ls *sra)
do
	fastq-dump $sample
	mv $sample completed.redo.sra
	mv *logfile logfile.completed 
	
done

# Get the quality file and fasta information from the fastq
mkdir completed.fastq

for sample2 in $(ls *fastq)
do
	mnt/EXT/Schloss-data/bin/mothur "#fastq.info(fastq=$sample2)"	
	mv $sample2 completed.fastq
	mv *logfile logfile.completed
	
done

# Trim the fastas according to quality 
mkdir first.trim.completed
for Trimming in $(ls *fasta)
do
	Quality=${Trimming//fasta/qual}
	mnt/EXT/Schloss-data/bin/mothur "#trim.seqs(fasta=$Trimming, qfile=$Quality, maxambig=0, maxhomop=8, qwindowaverage=35, qwindowsize=50)"
	mv $Trimming first.trim.completed
	mv $Quality first.trim.completed
	mv *logfile logfile.completed
done

mkdir scrap.fasta.completed
mkdir scrap.qual.completed
mkdir trim.qual.completed
mv *scrap.qual scrap.qual.completed
mv * scrap.fasta.completed
mv *trim.qual trim.qual.completed
```

Now I realized moving through this data set that they had SRR's for an individual sample so I decided to combine all these SRR's together based on their sample ID.  I used a python script that would make this possible.  It takes to columns seperated by a tab and looks for common names from one column and inserts a user defined command at the beginning as well as any special symbols like /* or .fasta to the end of each column as needed.  If you want to use the program it is found in my personal repository [here](https://github.com/marcsze/pythonPrograms) called RepeatEdit.py.  At any rate if you just want to copy and paste the command (it is long) you can see it below.  The output of the RepeatEdit.py program will basically spit this out for you.

```{r, engine='bash', eval=FALSE}
#!/bin/bash

cat SRR050640* SRR050641* > p0_s2.fasta
cat SRR050550* SRR050551* SRR050806* SRR050807* > p0_s1.fasta
cat SRR050558* SRR050559* SRR050814* SRR050815* SRR050732* SRR050733* > p10_s1.fasta
cat SRR050808* SRR050809* SRR050552* SRR050553* > p100_s1.fasta
cat SRR050724* SRR050725* > p103_s1.fasta
cat SRR050452* SRR050453* SRR050650* SRR050651* > p115_s1.fasta
cat SRR050456* SRR050457* > p118_s1.fasta
cat SRR050460* SRR050461* > p120_s2.fasta
cat SRR050556* SRR050557* SRR050812* SRR050813* > p108_s1.fasta
cat SRR050560* SRR050561* SRR050816* SRR050817* > p111_s1.fasta
cat SRR050642* SRR050643* > p106_s2.fasta
cat SRR050644* SRR050645* > p112_s1.fasta
cat SRR050648* SRR050649* > p114_s1.fasta
cat SRR050652* SRR050653* > p119_s1.fasta
cat SRR050728* SRR050729* > p105_s1.fasta
cat SRR050730* SRR050731* > p105_s2.fasta
cat SRR050554* SRR050555* SRR050810* SRR050811* > p106_s1.fasta
cat SRR050458* SRR050459* > p120_s1.fasta
cat SRR050464* SRR050465* > p124_s1.fasta
cat SRR050466* SRR050467* > p125_s1.fasta
cat SRR050470* SRR050471* > p129_s1.fasta
cat SRR050472* SRR050473* > p130_s1.fasta
cat SRR050476* SRR050477* > p133_s1.fasta
cat SRR050478* SRR050479* > p134_s1.fasta
cat SRR050482* SRR050483* > p136_s1.fasta
cat SRR050484* SRR050485* > p137_s1.fasta
cat SRR050488* SRR050489* > p140_s1.fasta
cat SRR050490* SRR050491* > p142_s1.fasta
cat SRR050494* SRR050495* > p148_s1.fasta
cat SRR050496* SRR050497* > p149_s1.fasta
cat SRR050500* SRR050501* > p151_s1.fasta
cat SRR050502* SRR050503* > p152_s1.fasta
cat SRR050522* SRR050523* > p36_s1.fasta
cat SRR050524* SRR050525* > p39_s2.fasta
cat SRR050528* SRR050529* > p42_s1.fasta
cat SRR050530* SRR050531* SRR050682* SRR050683* SRR095127* SRR095128* > p44_s2.fasta
cat SRR050564* SRR050565* SRR050820* SRR050821* > p14_s1.fasta
cat SRR050568* SRR050569* SRR050824* SRR050825* > p17_s1.fasta
cat SRR050572* SRR050573* SRR050828* SRR050829* > p20_s1.fasta
cat SRR050576* SRR050577* SRR050832* SRR050833* > p3_s1.fasta
cat SRR050580* SRR050581* SRR050764* SRR050765* SRR050836* SRR050837* > p48_s1.fasta
cat SRR050584* SRR050585* SRR050586* SRR050587* SRR050840* SRR050841* > p4_s1.fasta
cat SRR050660* SRR050661* > p16_s2.fasta
cat SRR050662* SRR050663* > p18_s2.fasta
cat SRR050736* SRR050737* > p13_s1.fasta
cat SRR050738* SRR050739* > p13_s2.fasta
cat SRR050538* SRR050539* > p6_s2.fasta
cat SRR050540* SRR050541* > p72_s2.fasta
cat SRR050582* SRR050583* SRR050766* SRR050767* SRR050838* SRR050839* > p49_s1.fasta
cat SRR050588* SRR050589* SRR050842* SRR050843* > p50_s1.fasta
cat SRR050592* SRR050593* SRR050846* SRR050847* > p54_s1.fasta
cat SRR050594* SRR050595* SRR050848* SRR050849* > p55_s1.fasta
cat SRR050598* SRR050599* SRR050852* SRR050853* > p57_s1.fasta
cat SRR050600* SRR050601* SRR050854* SRR050855* > p61_s1.fasta
cat SRR050604* SRR050605* SRR050858* SRR050859* > p64_s1.fasta
cat SRR050606* SRR050607* SRR050860* SRR050861* > p6_s1.fasta
cat SRR050610* SRR050611* SRR050864* SRR050865* > p79_s1.fasta
cat SRR050612* SRR050613* SRR050866* SRR050867* > p7_s1.fasta
cat SRR050616* SRR050617* SRR050870* SRR050871* > p82_s1.fasta
cat SRR050626* SRR050627* SRR050882* SRR050883* > p8_s1.fasta
cat SRR050666* SRR050667* > p24_s1.fasta
cat SRR050668* SRR050669* > p25_s1.fasta
cat SRR050672* SRR050673* > p27_s1.fasta
cat SRR050674* SRR050675* > p37_s1.fasta
cat SRR050678* SRR050679* > p39_s1.fasta
cat SRR050680* SRR050681* > p43_s1.fasta
cat SRR050602* SRR050603* SRR050686* SRR050687* SRR050856* SRR050857* > p63_s1.fasta
cat SRR050690* SRR050691* SRR050782* SRR050783* > p69_s1.fasta
cat SRR050694* SRR050695* > p71_s2.fasta
cat SRR050696* SRR050697* SRR050788* SRR050789* > p72_s1.fasta
cat SRR050700* SRR050701* > p74_s2.fasta
cat SRR050702* SRR050703* SRR050796* SRR050797* > p77_s1.fasta
cat SRR050706* SRR050707* > p81_s2.fasta
cat SRR050708* SRR050709* > p82_s2.fasta
cat SRR050740* SRR050741* > p20_s2.fasta
cat SRR050742* SRR050743* > p22_s1.fasta
cat SRR050746* SRR050747* > p29_s1.fasta
cat SRR050748* SRR050749* > p29_s2.fasta
cat SRR050752* SRR050753* > p33_s1.fasta
cat SRR050754* SRR050755* > p38_s1.fasta
cat SRR050758* SRR050759* > p41_s1.fasta
cat SRR050578* SRR050579* SRR050760* SRR050761* SRR050834* SRR050835* > p46_s1.fasta
cat SRR050772* SRR050773* > p62_s1.fasta
cat SRR050774* SRR050775* > p62_s2.fasta
cat SRR050778* SRR050779* > p64_s2.fasta
cat SRR050688* SRR050689* SRR050780* SRR050781* > p68_s1.fasta
cat SRR050698* SRR050699* SRR050790* SRR050791* > p73_s1.fasta
cat SRR050794* SRR050795* > p75_s1.fasta
cat SRR050570* SRR050571* SRR050826* SRR050827* > p18_s1.fasta
cat SRR050504* SRR050505* > p153_s1.fasta
cat SRR050506* SRR050507* > p154_s1.fasta
cat SRR050510* SRR050511* > p156_s1.fasta
cat SRR050512* SRR050513* > p158_s1.fasta
cat SRR050518* SRR050519* > p161_s1.fasta
cat SRR050534* SRR050535* > p48_s2.fasta
cat SRR050546* SRR050547* > p87_s2.fasta
cat SRR050548* SRR050549* > p90_s2.fasta
cat SRR050620* SRR050621* SRR050876* SRR050877* > p86_s1.fasta
cat SRR050622* SRR050623* SRR050800* SRR050801* SRR050878* SRR050879* > p87_s1.fasta
cat SRR050628* SRR050629* SRR050884* SRR050885* > p90_s1.fasta
cat SRR050630* SRR050631* SRR050886* SRR050887* > p91_s1.fasta
cat SRR050634* SRR050635* SRR050890* SRR050891* > p95_s1.fasta
cat SRR050636* SRR050637* SRR050892* SRR050893* > p97_s1.fasta
cat SRR050544* SRR050545* SRR050710* SRR050711* > p85_s2.fasta
cat SRR050716* SRR050717* > p91_s2.fasta
cat SRR050720* SRR050721* > p98_s1.fasta
cat SRR050722* SRR050723* > p98_s2.fasta
cat SRR050804* SRR050805* > p94_s2.fasta
cat SRR050608* SRR050609* SRR050862* SRR050863* > p78_s1.fasta
cat SRR050872* SRR050873* > p83_s1.fasta
cat SRR050624* SRR050625* SRR050880* SRR050881* > p88_s1.fasta
cat SRR050638* SRR050639* SRR050894* SRR050895* > p99_s1.fasta
cat SRR095005* SRR095006* SRR095053* SRR095054* SRR095125* SRR095126* > 177.fasta
cat SRR095007* SRR095008* SRR095055* SRR095056* SRR095101* SRR095102* > 186.fasta
cat SRR095011* SRR095012* SRR095059* SRR095060* > 211.fasta
cat SRR095013* SRR095014* SRR095061* SRR095062* > 213.fasta
cat SRR095017* SRR095018* SRR095065* SRR095066* > 224.fasta
cat SRR095021* SRR095022* SRR095069* SRR095070* SRR095105* SRR095106* > 238.fasta
cat SRR095001* SRR095002* SRR095047* SRR095048* > 102.fasta
cat SRR095009* SRR095010* SRR095057* SRR095058* > 208.fasta
cat SRR095095* SRR095096* > 168.fasta
cat SRR095099* SRR095100* > 179.fasta
cat SRR095107* SRR095108* > 259.fasta
cat SRR095109* SRR095110* SRR095147* SRR095148* > 283.fasta
cat SRR095133* SRR095134* > 116.fasta
cat SRR095135* SRR095136* > 126.fasta
cat SRR095139* SRR095140* > 162.fasta
cat SRR095141* SRR095142* > 269.fasta
cat SRR095145* SRR095146* > 275.fasta
cat SRR095149* SRR095150* > 297.fasta
cat SRR095253* SRR095254* > 128.fasta
cat SRR095257* SRR095258* > 157.fasta
cat SRR095261* SRR095262* > 169.fasta
cat SRR095269* SRR095270* > 212.fasta
cat SRR095273* SRR095274* > 226.fasta
cat SRR095275* SRR095276* > 242.fasta
cat SRR095369* SRR095370* SRR095456* SRR095457* > 107.fasta
cat SRR095371* SRR095372* > 113.fasta
cat SRR095375* SRR095376* SRR095424* SRR095425* > 11.fasta
cat SRR095377* SRR095378* SRR095434* SRR095435* > 219.fasta
cat SRR095381* SRR095382* SRR095482* SRR095483* > 12.fasta
cat SRR095373* SRR095374* SRR095387* SRR095388* > 13.fasta
cat SRR095393* SRR095394* SRR095610* SRR095611* > 150.fasta
cat SRR095395* SRR095396* SRR095492* SRR095493* > 156.fasta
cat SRR095401* SRR095402* SRR095458* SRR095459* > 287.fasta
cat SRR095412* SRR095413* SRR095440* SRR095441* > 192.fasta
cat SRR095414* SRR095415* SRR095430* SRR095431* > 217.fasta
cat SRR095418* SRR095419* SRR095438* SRR095439* > 1.fasta
cat SRR095420* SRR095421* SRR095598* SRR095599* > 201.fasta
cat SRR095428* SRR095429* SRR095452* SRR095453* > 216.fasta
cat SRR095436* SRR095437* SRR095502* SRR095503* > 230.fasta
cat SRR095379* SRR095380* SRR095460* SRR095461* > 289.fasta
cat SRR095442* SRR095443* SRR095466* SRR095467* > 2.fasta
cat SRR095426* SRR095427* SRR095532* SRR095533* > 214.fasta
cat SRR095391* SRR095392* SRR095620* SRR095621* > 14.fasta
cat SRR095031* SRR095032* SRR095079* SRR095080* > 392.fasta
cat SRR095033* SRR095034* SRR095081* SRR095082* SRR095121* SRR095122* > 397.fasta
cat SRR095037* SRR095038* SRR095085* SRR095086* SRR095123* SRR095124* > 453.fasta
cat SRR095027* SRR095028* SRR095075* SRR095076* > 327.fasta
cat SRR095111* SRR095112* > 313.fasta
cat SRR095115* SRR095116* > 323.fasta
cat SRR095119* SRR095120* > 388.fasta
cat SRR095155* SRR095156* > 333.fasta
cat SRR095159* SRR095160* > 358.fasta
cat SRR095161* SRR095162* > 381.fasta
cat SRR095167* SRR095168* > 459.fasta
cat SRR095169* SRR095170* > 46.fasta
cat SRR095173* SRR095174* > 481.fasta
cat SRR095175* SRR095176* SRR095177* SRR095178* SRR095179* SRR095180* SRR095181* SRR095182* SRR095183* SRR095184* > 482.fasta
cat SRR095185* SRR095186* SRR095187* SRR095188* SRR095189* SRR095190* SRR095191* SRR095192* SRR095193* SRR095194* SRR095195* SRR095196* SRR095197* SRR095198* SRR095199* SRR095200* > 49.fasta
cat SRR095283* SRR095284* > 354.fasta
cat SRR095291* SRR095292* > 406.fasta
cat SRR095293* SRR095294* > 416.fasta
cat SRR095299* SRR095300* > 444.fasta
cat SRR095301* SRR095302* > 44.fasta
cat SRR095307* SRR095308* SRR095309* SRR095310* SRR095311* SRR095312* SRR095313* SRR095314* SRR095315* SRR095316* SRR095317* SRR095318* SRR095319* SRR095320* > 488.fasta
cat SRR095321* SRR095322* SRR095323* SRR095324* SRR095325* SRR095326* SRR095327* SRR095328* SRR095329* SRR095330* SRR095331* SRR095332* SRR095333* SRR095334* SRR095335* SRR095336* SRR095337* SRR095338* SRR095339* SRR095340* SRR095341* SRR095342* SRR095343* SRR095344* SRR095345* SRR095346* SRR095347* SRR095348* SRR095349* SRR095350* SRR095351* SRR095352* SRR095353* SRR095354* SRR095355* SRR095356* SRR095357* SRR095358* SRR095359* SRR095360* SRR095361* SRR095362* > 50.fasta
cat SRR095403* SRR095404* SRR095572* SRR095573* SRR095574* SRR095575* SRR095576* SRR095577* > 48.fasta
cat SRR095405* SRR095406* SRR095560* SRR095561* > 473.fasta
cat SRR095432* SRR095433* SRR095540* SRR095541* > 445.fasta
cat SRR095446* SRR095447* SRR095514* SRR095515* > 395.fasta
cat SRR095450* SRR095451* SRR095550* SRR095551* > 463.fasta
cat SRR095454* SRR095455* SRR095518* SRR095519* > 401.fasta
cat SRR095470* SRR095471* SRR095500* SRR095501* > 367.fasta
cat SRR095448* SRR095449* SRR095472* SRR095473* > 308.fasta
cat SRR095476* SRR095477* SRR095478* SRR095479* > 316.fasta
cat SRR095486* SRR095487* > 346.fasta
cat SRR095490* SRR095491* SRR095568* SRR095569* > 349.fasta
cat SRR095494* SRR095495* SRR095548* SRR095549* > 352.fasta
cat SRR095480* SRR095481* SRR095504* SRR095505* > 373.fasta
cat SRR095383* SRR095384* SRR095508* SRR095509* > 376.fasta
cat SRR095522* SRR095523* > 410.fasta
cat SRR095524* SRR095525* > 411.fasta
cat SRR095528* SRR095529* SRR095542* SRR095543* > 424.fasta
cat SRR095530* SRR095531* SRR095536* SRR095537* > 427.fasta
cat SRR095534* SRR095535* SRR095538* SRR095539* > 429.fasta
cat SRR095546* SRR095547* SRR095590* SRR095591* > 455.fasta
cat SRR095516* SRR095517* SRR095556* SRR095557* > 398.fasta
cat SRR095564* SRR095565* SRR095633* SRR095634* > 475.fasta
cat SRR095474* SRR095475* SRR095570* SRR095571* > 483.fasta
cat SRR095562* SRR095563* SRR095600* SRR095601* > 474.fasta
cat SRR050450* SRR050451* SRR095255* SRR095256* > p113_s1.fasta
cat SRR050516* SRR050517* SRR095289* SRR095290* > p160_s1.fasta
cat SRR050574* SRR050575* SRR095297* SRR095298* > p34_s1.fasta
cat SRR095019* SRR095020* SRR095067* SRR095068* > 234.fasta
cat SRR095039* SRR095040* SRR095087* SRR095088* > 59.fasta
cat SRR095041* SRR095042* SRR095089* SRR095090* > 66.fasta
cat SRR095045* SRR095046* SRR095093* SRR095094* > 84.fasta
cat SRR050566* SRR050567* SRR095051* SRR095052* SRR095097* SRR095098* > p15_s1.fasta
cat SRR095113* SRR095114* SRR095151* SRR095152* > p58_s1.fasta
cat SRR095129* SRR095130* > 523.fasta
cat SRR095165* SRR095166* > 423.fasta
cat SRR095201* SRR095202* SRR095203* SRR095204* SRR095205* SRR095206* SRR095207* SRR095208* SRR095209* SRR095210* SRR095211* SRR095212* > 51.fasta
cat SRR095213* SRR095214* SRR095215* SRR095216* SRR095217* SRR095218* SRR095219* SRR095220* SRR095221* SRR095222* SRR095223* SRR095224* SRR095225* SRR095226* SRR095227* SRR095228* SRR095229* SRR095230* SRR095231* SRR095232* SRR095233* SRR095234* SRR095235* SRR095236* SRR095237* SRR095238* SRR095239* SRR095240* SRR095241* SRR095242* SRR095243* SRR095244* SRR095245* SRR095246* > 52.fasta
cat SRR095249* SRR095250* > 87.fasta
cat SRR095265* SRR095266* > 190.fasta
cat SRR095267* SRR095268* > p127_s1.fasta
cat SRR095287* SRR095288* > p138_s1.fasta
cat SRR095303* SRR095304* > p123_s1.fasta
cat SRR095363* SRR095364* > p121_s1.fasta
cat SRR095365* SRR095366* > 68.fasta
cat SRR095397* SRR095398* > 160.fasta
cat SRR095410* SRR095411* SRR095510* SRR095511* > 191.fasta
cat SRR095444* SRR095445* > 24.fasta
cat SRR095468* SRR095469* SRR095631* SRR095632* > 92.fasta
cat SRR095552* SRR095553* SRR095558* SRR095559* > 464.fasta
cat SRR095580* SRR095581* SRR095602* SRR095603* > 53.fasta
cat SRR095582* SRR095583* SRR095584* SRR095585* SRR095586* SRR095587* SRR095594* SRR095595* > 54.fasta
cat SRR095596* SRR095597* SRR095604* SRR095605* > 61.fasta
cat SRR095608* SRR095609* SRR095624* SRR095625* > 83.fasta
cat SRR095612* SRR095613* SRR095630* > 77.fasta
cat SRR095614* SRR095615* SRR095618* SRR095619* > 7.fasta
cat SRR095622* SRR095623* SRR095628* SRR095629* > 81.fasta

mkdir trim.fasta.renamed.complete
mv *trim.fasta trim.fasta.renamed.complete
```

So after this I need to make a group file and then combine the fastas and group files together.
```{r, engine='bash', eval=FALSE}
#!/bin/bash

groupN="102 107 113 116 11 126 128 12 13 14 150 156 157 160 162 168 169 177 179 186 190 191 192 1 201 208 211 212 213 214 216 217 219 224 226 230 234 238 242 24 259 269 275 283 287 289 297 2 308 313 316 323 327 333 346 349 352 354 358 367 373 376 381 388 392 395 397 398 401 406 410 411 416 423 424 427 429 444 445 44 453 455 459 463 464 46 473 474 475 481 482 483 488 48 49 50 51 523 52 53 54 59 61 66 68 77 7 81 83 84 87 92 p0_s1 p0_s2 p100_s1 p103_s1 p105_s1 p105_s2 p106_s1 p106_s2 p108_s1 p10_s1 p111_s1 p112_s1 p113_s1 p114_s1 p115_s1 p118_s1 p119_s1 p120_s1 p120_s2 p121_s1 p123_s1 p124_s1 p125_s1 p127_s1 p129_s1 p130_s1 p133_s1 p134_s1 p136_s1 p137_s1 p138_s1 p13_s1 p13_s2 p140_s1 p142_s1 p148_s1 p149_s1 p14_s1 p151_s1 p152_s1 p153_s1 p154_s1 p156_s1 p158_s1 p15_s1 p160_s1 p161_s1 p16_s2 p17_s1 p18_s1 p18_s2 p20_s1 p20_s2 p22_s1 p24_s1 p25_s1 p27_s1 p29_s1 p29_s2 p33_s1 p34_s1 p36_s1 p37_s1 p38_s1 p39_s1 p39_s2 p3_s1 p41_s1 p42_s1 p43_s1 p44_s2 p46_s1 p48_s1 p48_s2 p49_s1 p4_s1 p50_s1 p54_s1 p55_s1 p57_s1 p58_s1 p61_s1 p62_s1 p62_s2 p63_s1 p64_s1 p64_s2 p68_s1 p69_s1 p6_s1 p6_s2 p71_s2 p72_s1 p72_s2 p73_s1 p74_s2 p75_s1 p77_s1 p78_s1 p79_s1 p7_s1 p81_s2 p82_s1 p82_s2 p83_s1 p85_s2 p86_s1 p87_s1 p87_s2 p88_s1 p8_s1 p90_s1 p90_s2 p91_s1 p91_s2 p94_s2 p95_s1 p97_s1 p98_s1 p98_s2 p99_s1"

for V in ${groupN}
do
	grep -e '^>' ${V}.fasta > test.fasta
	sed 's/>//' test.fasta > test2.fasta
	sed "s/$/\t$V/" test2.fasta > ${V}.groups
	rm test.fasta
	rm test2.fasta
done 

cat *fasta > combined.fasta
cat *groups > combined.groups

```

From here things get pretty simple.  All you now need to do is run mothur with the provided batch file and everything should work....

```{r, engine='bash', eval=FALSE}
mnt/EXT/Schloss-data/bin/mothur Zupancic.batch
```

With that you should now be able to generate the subsample shared file that was used for the downstream analysis.
<br><br>


##### **Human Microbiome Project (HMP) 2011:**  
This data set consisted of healthy individuals that were sampled at multiple body sites and underwent extensive medical work up for metadata collection.  In order to be included in this study you could not have any underlying medical conditions.  The project website can be found [here](http://hmpdacc.org/).  The project accession number in the SRA is SRP002860.  The sequence processing pipeline can be found in an ipython notebook located [here](http://nbviewer.jupyter.org/gist/pschloss/9815766/notebook.ipynb).  The one exception to the workflow provided in the notebook is that you will need to make sure that you do not use a closed reference data base approach for OTU clustering and instead use an open reference approach.  We used the default approach listed rather than repeat the whole analysis again.  
<br><br>


##### **Wu, et al. 2011:**  
This study was published in Science using the COMBO data set and can be found [here](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3368382/).  The investigators were mostly interested in investigating dietary patterns its influence of gur microbial enterotypes. The data for this project can be found on the SRA website under the Accession number SRP002424.  So with a little modification we can use wget again to get the data.

```{r, engine='bash', eval=FALSE}
wget -r -np -nd -k ftp://ftp-trace.ncbi.nih.gov/sra/sra-instant/reads/ByStudy/sra/SRP/SRP002/SRP002424/
```

From here there was a program that was run to loop through a few of the mothur functions on each individual file.  Since they are already seperated by indivdiual we don't need the oligo file.  So here we go......

```{r, engine='bash', eval=FALSE}
#!/bin/bash

# Make a directory to store the reference files
# Remember to move these back into the necessary directory later
# When you get to the batch file point,
mkdir reference.files
mv silva.bacteria.fasta reference.files
mv trainset9_032012.pds.tax reference.files
mv trainset9_032012.pds.fasta reference.files

# Read in the sra and make a fastq instead of a sff.
mkdir completed.redo.sra
mkdir logfile.completed

for sample in $(ls *sra)
do
	fastq-dump $sample
	mv $sample completed.redo.sra
	mv *logfile logfile.completed 
	
done

# Get the quality file and fasta information from the fastq
mkdir completed.fastq

for sample2 in $(ls *fastq)
do
	mnt/EXT/Schloss-data/bin/mothur "#fastq.info(fastq=$sample2)"	
	mv $sample2 completed.fastq
	mv *logfile logfile.completed
	
done

# Combine the qaulity and fasta files together
cat *fasta > combined.fasta
cat *qual > combined.qual

```


After you have done this you can run a batch file with mothur and that should generate the subsample shared file that we used for this anlaysis.  It should be noted that this data set came with the barcodes and they needed to be removed so this differed form the previous data sets in this regard.  However, this is all taken care of in the batch file.

```{r, engine='bash', eval=FALSE}
#!/bin/bash

mnt/EXT/Schloss-data/bin/mothur Wu.batch
```


##### **Turnbaugh, et al. 2008:**  
This is the very famous twin study of the microbiome and obesity.  The other two re-analysis articles [@finucane_taxonomic_2014; @walters_meta-analyses_2014] also use this data set.  The raw data is not avialable on the SRA but you can download a fasta file for either the V2 or V6 16S rRNA gene [here](https://gordonlab.wustl.edu/NatureTwins_2008/TurnbaughNature_11_30_08.html).  We used the V2 region.  For the most part it was relatively straight forward.  The code used and instruction are below:

```{r, engine='bash', eval=FALSE}
grep -e '^>' V2.fasta > test.fasta  
sed 's/>//' test.fasta > test2.fasta    
sed 's/\s.*$//' test2.fasta > test3.fasta  
paste -d" " test2.fasta test3.fasta > V2.groups''
```

Once we have a groups file we can just create a batch file and put it into mothur to run and come back when it is completed.
```{r, engine='bash', eval=FALSE}
mnt/EXT/Schloss-data/bin/mothur Turnbaugh.batch
```

##### **References:** 

























