\documentclass[12pt,]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
  \newcommand{\euro}{€}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage[margin=1.0in]{geometry}
\usepackage{hyperref}
\PassOptionsToPackage{usenames,dvipsnames}{color} % color is loaded by hyperref
\hypersetup{unicode=true,
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{{#1}}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{{#1}}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{{#1}}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{{#1}}}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{{#1}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{{#1}}}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{{#1}}}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{{#1}}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{{#1}}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{{#1}}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{{#1}}}
\newcommand{\ImportTok}[1]{{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{{#1}}}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{{#1}}}}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{{#1}}}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{{#1}}}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{{#1}}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{{#1}}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{{#1}}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{{#1}}}}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{{#1}}}}
\newcommand{\BuiltInTok}[1]{{#1}}
\newcommand{\ExtensionTok}[1]{{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{{#1}}}
\newcommand{\RegionMarkerTok}[1]{{#1}}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{{#1}}}}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{{#1}}}}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{{#1}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{{#1}}}}
\newcommand{\NormalTok}[1]{{#1}}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\newcommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}
  \title{}
  \pretitle{\vspace{\droptitle}}
  \posttitle{}
  \author{}
  \preauthor{}\postauthor{}
  \date{}
  \predate{}\postdate{}


\usepackage{helvet} % Helvetica font
\renewcommand*\familydefault{\sfdefault} % Use the sans serif version of the font
\usepackage[T1]{fontenc}

\usepackage[none]{hyphenat}

\usepackage{setspace}
\doublespacing
\setlength{\parskip}{1em}

\usepackage{lineno}

\usepackage{pdfpages}
\usepackage{comment}

% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

\begin{document}

\subsection{Supplementary Text 1: Looking for a Signal in the Noise:
Revisiting Obesity and the
Microbiome}\label{supplementary-text-1-looking-for-a-signal-in-the-noise-revisiting-obesity-and-the-microbiome}

Marc A Sze and Patrick D Schloss

\subsubsection{In-Depth Overview of Search
Strategy}\label{in-depth-overview-of-search-strategy}

The initial search strategy included looking for all papers that
initially fit under the below NCBI PubMed advanced search criteria. The
terms included in this criteria were that the manuscript had to have
``Bacterial Microbiome'' and ``Obesity, BMI, bmi, obesity'' in their
manuscript criteria, it was not published more than 10 years ago, they
were not review articles, and it contained research on humans only. The
below formula when put into PubMed should recapitulate our initial
search on the website.

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{((}\NormalTok{(((((((Bacterial Microbiome) AND (Obesity or bmi or body mass index or BMI or obesity) AND }\StringTok{"last 10 years"}\NormalTok{[PDat] AND Humans[Mesh])) NOT review[ptyp]) AND }\StringTok{"last 10 years"}\NormalTok{[PDat] AND Humans[Mesh])) AND }\StringTok{"last 10 years"}\NormalTok{[PDat] AND Humans[Mesh])) AND }\StringTok{"last 10 years"}\NormalTok{[PDat] AND Humans[Mesh])}
\end{Highlighting}
\end{Shaded}

This search yielded a total of 187 manuscripts. From two previous other
reviews of obesity and the bacterial microbiome along with knowledge of
two other published papers that investigated obesity but were missed by
the database search we obtained a total of 7 more articles. We also had
access to normal healthy individuals from an unpublished dataset. This
brought our total number of records to 196.

From this total we browsed abstracts for mention of stool or feces
examination, that did not involve children, was not a clinical trial for
probiotics or other diet related treatments, did not only have
participants with inflammatory bowel disease, the articles were in
English, did not only use PCR, qPCR, or RT-PCR only for their analysis,
and sequencing that used only clone libraries. This ultimately excluded
all but a total of 13 studies.

From this total of 11 studies with the full text was reviewed for
whether or not sequencing data was publicly available, BMI information
(either categorical or continuous) was available in a supplement or, if
it was not available, whether authors upon contact were willing to share
this information or direct us to repositories that stored this specific
information. One study was excluded (1) because it contained children
and their sequencing of the 16S rRNA gene involved amplicons of only
100bp in length. They also did not have obesity as part of their results
in the actual published manuscript. The second study was excluded
because it did not use 16S rRNA gene sequencing for their bacterial
microbiome analysis (2).

Once these 2 studies were excluded there was a total of 11 studies in
the qualitative synthesis of the analysis. Because we decided a prioi to
use the standard definition for BMI group classification one study from
this 11 did not have any individuals who were obese by this criteria (3)
and was excluded from the final quantitative synthesis and analysis.

\textbf{\emph{Inclusion Criteria:}}

\begin{itemize}
\tightlist
\item
  Contains mention of Bacterial Microbiome and Obesity
\item
  BMI, bmi, or obesity could be referenced instead of Obesity
\item
  Not published more than 10 years ago
\item
  Research on Humans only
\item
  At least one specific result examining obesity and a bacterial
  microbiome measure
\item
  Participants did not have Inflammatory Bowel Disease or Cancer
\item
  Greater than 100bp single or dual end reads for 16S Sequencing
\item
  DNA obtained from stool or feces
\end{itemize}

\textbf{\emph{Exclusion Criteria:}}

\begin{itemize}
\tightlist
\item
  PCR, qPCR, metagenomic sequencing, or RT-PCR used as main analysis
  Tool
\item
  TRFLP or clone sequencing used to asses the bacterial community
\item
  Utilization of 100bp or less single end reads for sequencing
\item
  Sequencing Data not publicly available for download
\item
  BMI not available and authors do not return correspondence
\item
  Samples were not stool or feces
\item
  Study contained children
\item
  Study was a review
\end{itemize}

\subsubsection*{References}\label{references}
\addcontentsline{toc}{subsubsection}{References}

\hypertarget{refs}{}
\hypertarget{ref-yatsunenkoux5fhumanux5f2012}{}
1. \textbf{Yatsunenko T}, \textbf{Rey FE}, \textbf{Manary MJ},
\textbf{Trehan I}, \textbf{Dominguez-Bello MG}, \textbf{Contreras M},
\textbf{Magris M}, \textbf{Hidalgo G}, \textbf{Baldassano RN},
\textbf{Anokhin AP}, \textbf{Heath AC}, \textbf{Warner B},
\textbf{Reeder J}, \textbf{Kuczynski J}, \textbf{Caporaso JG},
\textbf{Lozupone CA}, \textbf{Lauber C}, \textbf{Clemente JC},
\textbf{Knights D}, \textbf{Knight R}, \textbf{Gordon JI}. 2012. Human
gut microbiome viewed across age and geography. Nature
\textbf{486}:222--227.
doi:\href{https://doi.org/10.1038/nature11053}{10.1038/nature11053}.

\hypertarget{ref-arumugamux5fenterotypesux5f2011}{}
2. \textbf{Arumugam M}, \textbf{Raes J}, \textbf{Pelletier E},
\textbf{Le Paslier D}, \textbf{Yamada T}, \textbf{Mende DR},
\textbf{Fernandes GR}, \textbf{Tap J}, \textbf{Bruls T}, \textbf{Batto
J-M}, \textbf{Bertalan M}, \textbf{Borruel N}, \textbf{Casellas F},
\textbf{Fernandez L}, \textbf{Gautier L}, \textbf{Hansen T},
\textbf{Hattori M}, \textbf{Hayashi T}, \textbf{Kleerebezem M},
\textbf{Kurokawa K}, \textbf{Leclerc M}, \textbf{Levenez F},
\textbf{Manichanh C}, \textbf{Nielsen HB}, \textbf{Nielsen T},
\textbf{Pons N}, \textbf{Poulain J}, \textbf{Qin J},
\textbf{Sicheritz-Ponten T}, \textbf{Tims S}, \textbf{Torrents D},
\textbf{Ugarte E}, \textbf{Zoetendal EG}, \textbf{Wang J},
\textbf{Guarner F}, \textbf{Pedersen O}, \textbf{Vos WM de},
\textbf{Brunak S}, \textbf{Doré J}, \textbf{MetaHIT Consortium},
\textbf{Antolín M}, \textbf{Artiguenave F}, \textbf{Blottiere HM},
\textbf{Almeida M}, \textbf{Brechot C}, \textbf{Cara C},
\textbf{Chervaux C}, \textbf{Cultrone A}, \textbf{Delorme C},
\textbf{Denariaz G}, \textbf{Dervyn R}, \textbf{Foerstner KU},
\textbf{Friss C}, \textbf{Guchte M van de}, \textbf{Guedon E},
\textbf{Haimet F}, \textbf{Huber W}, \textbf{Hylckama-Vlieg J van},
\textbf{Jamet A}, \textbf{Juste C}, \textbf{Kaci G}, \textbf{Knol J},
\textbf{Lakhdari O}, \textbf{Layec S}, \textbf{Le Roux K},
\textbf{Maguin E}, \textbf{Mérieux A}, \textbf{Melo Minardi R},
\textbf{M'rini C}, \textbf{Muller J}, \textbf{Oozeer R},
\textbf{Parkhill J}, \textbf{Renault P}, \textbf{Rescigno M},
\textbf{Sanchez N}, \textbf{Sunagawa S}, \textbf{Torrejon A},
\textbf{Turner K}, \textbf{Vandemeulebrouck G}, \textbf{Varela E},
\textbf{Winogradsky Y}, \textbf{Zeller G}, \textbf{Weissenbach J},
\textbf{Ehrlich SD}, \textbf{Bork P}. 2011. Enterotypes of the human gut
microbiome. Nature \textbf{473}:174--180.
doi:\href{https://doi.org/10.1038/nature09944}{10.1038/nature09944}.

\hypertarget{ref-namux5fcomparativeux5f2011}{}
3. \textbf{Nam Y-D}, \textbf{Jung M-J}, \textbf{Roh SW}, \textbf{Kim
M-S}, \textbf{Bae J-W}. 2011. Comparative analysis of Korean human gut
microbiota by barcoded pyrosequencing. PloS One \textbf{6}:e22109.
doi:\href{https://doi.org/10.1371/journal.pone.0022109}{10.1371/journal.pone.0022109}.

\end{document}
