\documentclass[12pt,]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
    \usepackage{xltxtra,xunicode}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Mapping=tex-text,Scale=MatchLowercase}
  \newcommand{\euro}{â‚¬}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage[margin=1.0in]{geometry}
\ifxetex
  \usepackage[setpagesize=false, % page size defined by xetex
              unicode=false, % unicode breaks when used with xetex
              xetex]{hyperref}
\else
  \usepackage[unicode=true]{hyperref}
\fi
\hypersetup{breaklinks=true,
            bookmarks=true,
            pdfauthor={},
            pdftitle={},
            colorlinks=true,
            citecolor=blue,
            urlcolor=blue,
            linkcolor=magenta,
            pdfborder={0 0 0}}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\newcommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}
  \title{}
  \pretitle{\vspace{\droptitle}}
  \posttitle{}
  \author{}
  \preauthor{}\postauthor{}
  \date{}
  \predate{}\postdate{}

\usepackage{helvet} % Helvetica font
\renewcommand*\familydefault{\sfdefault} % Use the sans serif version of the font
\usepackage[T1]{fontenc}

\usepackage[none]{hyphenat}

\usepackage{setspace}
\doublespacing
\setlength{\parskip}{1em}

\usepackage{lineno}

\usepackage{pdfpages}
\usepackage{comment}

% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

\begin{document}
\maketitle

\section{Looking for a Signal in the Noise: Revisiting Obesity and the
Microbiome}\label{looking-for-a-signal-in-the-noise-revisiting-obesity-and-the-microbiome}

\begin{center}
\vspace{25mm}


Running Title: The Human Microbiome and Obesity

\vspace{10mm}

Marc A Sze and Patrick D Schloss${^\dagger}$

\vspace{10mm}

Contributions: Both authors contributed to the planning, design, execution, interpretation, and writing of the analyses.


\vspace{20mm}

$\dagger$ To whom correspondence should be addressed: pschloss@umich.edu

Department of Microbiology and Immunology, University of Michigan, Ann Arbor, MI
\end{center}

\newpage

\linenumbers

\subsection{Abstract}\label{abstract}

Two recent studies have re-analyzed published data and found that when
datasets are analyzed independently there was limited support for the
widely accepted hypothesis that changes in the microbiome are associated
with obesity. This hypothesis was reconsidered by increasing the number
of datasets and pooling the results across the individual datasets. The
Preferred Reporting Items for Systematic Reviews and Meta-Analyses
(PRISMA) guidelines were applied to identify 10 studies for an updated
and more synthetic analysis. Alpha diversity metrics and the relative
risk of obesity based on those metrics were used to identify a limited
number of significant associations with obesity; however, when the
results of the studies were pooled using a random effects model
significant associations were observed between Shannon diversity, number
of observed OTUs, and Shannon evenness and obesity status. They were not
observed for the ratio of \emph{Bacteroidetes} and \emph{Firmicutes} or
their individual relative abundances. Although these tests yielded small
P-values, the difference between the Shannon diversity index of
non-obese and obese individuals was 2.07\%. A power analysis
demonstrated that only one of the studies had sufficient power to detect
a 5\% difference in diversity. When Random Forest machine learning
models were trained on one dataset and then tested using the other 9
datasets, the median accuracy varied between 33.01 and 64.77\%
(median=56.68\%). Although there was support for a relationship between
the microbial communities found in human feces and obesity status, this
association was relatively weak and its detection is confounded by large
interpersonal variation and insufficient sample sizes.

\subsubsection{Importance}\label{importance}

As interest in the human microbiome grows there is an increasing number
of studies that can be used to test numerous hypotheses across human
populations. The hypothesis that variation in the gut microbiota can
explain or be used to predict obesity status has received considerable
attention and is frequently mentioned as an example for the role of the
microbiome in human health. Here we assess this hypothesis using ten
independent studies and find that although there is an association, it
is smaller than can be detected by most microbiome studies. Furthermore,
we directly tested the ability to predict obesity status based on the
composition of an individual's microbiome and find that the median
classification accuracy is between 33.01 and 64.77\%. This type of
analysis can be used to design future studies and expanded to explore
other hypotheses.

\newpage

\subsection{Introduction}\label{introduction}

Obesity is a growing health concern with approximately 20\% of the youth
(aged 2-19) in the United States classified as either overweight or
obese (1). This number increases to approximately 35\% in adults (aged
20 or older) and these statistics have seen little change since 2003
(1). Traditionally, the body mass index (BMI) has been used to classify
individuals as non-obese or obese (2). Recently, there has been
increased interest in the role of the microbiome in modulating obesity
(3, 4). If the microbiome does affect obesity status, then manipulating
the microbiome could have a significant role in the future treatment of
obesity and in helping to stem the current epidemic.

There have been several studies that report observing a link between the
composition of microbiome and obesity in animal models and in humans.
The first such study used genetically obese mice and observed the ratio
of the relative abundances of \emph{Bacteroidetes} to \emph{Firmicutes}
(B:F) was lower in obese mice than lean mice (5). Translation of this
result to humans by the same researchers did not observe this effect,
but did find that obese individuals had a lower alpha-diversity than
lean individuals (6). They also showed that the relative abundance of
\emph{Bacteroidetes} and \emph{Firmicutes} increased and decreased,
respectively, as obese individuals lost weight while on a fat or
carbohydrate restricted diet (7). Two re-analysis studies by Walters et
al. (8) and Finucane et al. (9) interrogated previously published
microbiome and obesity data and concluded that the previously reported
differences in community diversity and B:F among non-obese and obese
individuals could not be generalized. Regardless of the results using
human populations, studies using animal models where the community was
manipulated with antibiotics or established by colonizing germ-free
animals with varied communities appear to support the association since
these manipulations yielded differences in animal weight (10--13). The
purported association between the differences in the microbiome and
obesity have been widely repeated with little attention given to the
lack of a clear signal in human cohort studies.

The recent publication of additional studies that collected BMI data for
each subject as well as other studies that were not included in the
earlier re-analysis studies offered the opportunity to revisit the
question relating the structure of the human microbiome to obesity. One
critique of the prior re-analysis studies is that the authors did not
aggregate the results across studies to increase the effective sample
size. It is possible that there were small associations within each
study that were not statistically significant because the individual
studies lacked sufficient power. Alternatively, diversity metrics may
mask the appropriate signal and it is necessary to measure the
association at the level of microbial populations. The Walters
re-analysis study demonstrated that Random Forest machine learning
models were capable of predicting obesity status within a single cohort,
but did not attempt to test the models on other cohorts. The purpose of
this study was to perform a meta-analysis of the association between
differences in the microbiome and obesity status by analyzing and
applying a more systematic and synthetic approach than was used
previously.

\subsection{Results}\label{results}

\textbf{\emph{Literature Review and Study Inclusion.}} To perform a
robust meta-analysis and limit inclusion bias, we followed the Preferred
Reporting Items for Systematic Reviews and Meta-Analyses (PRISMA)
guidelines to identify the studies that we analyzed (14). A detailed
description of our selection process and the exact search terms are
provided in the Methods section and in Figure 1. Briefly, we searched
PubMed for original research studies that involved studying obesity and
the human microbiome. The initial search yielded 187 studies. We
identified 10 additional studies that were not designed to explicitly
test for an association between the microbiome and obesity. We then
manually curated the 197 studies to select those studies that included
BMI and 16S rRNA gene sequence data. This yielded 11 eligible studies.
An additional study was removed from our analysis because no individuals
in the study had a BMI over 30. Among the final 10 studies, 3 were
identified from our PubMed search (10, 15, 16), 5 were originally
identified from the 10 studies that did not explicitly investigate
obesity but included BMI data (17--21), and two datasets were used (22,
23) because these publications did not specifically look for any
metabolic or obesity conditions but had control populations and enabled
us to help mitigate against publication biases associated with the
bacterial microbiome and obesity. The ten studies are summarized in
Table 1. For comparison, two of these studies were included in the
Finucane re-analysis study (10, 21) and four of these studies were
included in the Walters re-analysis study (10, 15, 20, 21). The 16S rRNA
gene sequence data from each study was re-analyzed using a similar
approach based on previously described methods for reducing the number
of chimeric sequences and sequencing errors for 454 and Illumina MiSeq
data (24, 25). The sequences were clustered into operational taxonomic
units (OTUs) using the average neighbor approach (26) and into taxonomic
groupings based on their classification using a naive Bayesian
classifier (27).

\textbf{\emph{Alpha diversity analysis.}} We calculated the Shannon
diversity index, observed richness, and Shannon evenness, the relative
abundance of \emph{Bacteroidetes} and \emph{Firmicutes}, and the ratio
of their relative abundance (B:F) for each sample. Once we transformed
each of the six alpha diversity metrics to make them normally
distributed, we used a t-test to identify significant associations
between the alpha diversity metric and whether an individual was obese
for each of the ten studies. The B:F and the relative abundance of
\emph{Firmicutes} were not significantly associated with obesity in any
study. We identified seven P-values that were less than 0.05: three
studies indicated obese individuals had a lower richness, two studies
indicated a significantly lower diversity, one study indicated a
significantly lower evenness, and one study indicated a significantly
higher relative abundance of \emph{Bacteroidetes} (Figures 2 and S1).
These results largely match those of the Walters and Finucane
re-analysis studies. Interestingly, although only two of the ten studies
observed the previously reported association between lower diversity and
obesity, the other studies appeared to have the same trend, albeit the
differences were not statistically significant. We used a random effects
linear model to combine the studies using the study as the random effect
and found statistical support for decreased richness, evenness, and
diversity among obese individuals (all P\textless{}0.011). Although
there was a significant relationship between these metrics and obesity
status, the effect size was quite small. The obese individuals averaged
7.47\% lower richness, 0.88\% lower evenness, and 2.07\% lower
diversity. There were no significant associations when we pooled the
phylum-level metrics across studies. These results indicate that obese
individuals do have a statistically significant lower diversity than
non-obese individuals; however, it is questionable whether the
difference is biologically significant.

\textbf{\emph{Relative risk.}} Building upon the alpha diversity
analysis we calculated the relative risk of being obese based on an
individual's alpha diversity metrics relative to the median metric for
that study. Inspection of funnel plots for each of the metrics suggested
that the studies included in our analysis were not biased (Figure S2).
The results using relative risk largely matched those of using the raw
alpha diversity data. Across the ten studies and six metrics, the only
significant relative risk values were the richness, evenness, and
diversity values from the Goodrich study (Figures 3 and S3). Again,
although the relative risk values were not significant for other
studies, the values tended to be above one. When we pooled the data
using a random effects model, the relative risk associated with having a
richness, evenness, or diversity below the median for the population was
significantly associated with obesity (all P\textless{}0.0044). The
relative risks associated with alpha diversity were small. The relative
risk of having a low richness was 1.30 (95\% CI: 1.13-1.49), low
evenness was 1.20 (95\% CI: 1.06-1.37), and low diversity was 1.27 (95\%
CI: 1.09-1.48). There were no significant differences in the
phylum-level metrics. Again, the relative risk results indicate that
individuals with a lower richness, evenness, or diversity are at
statistically significant increased risk of being obese, it is
questionable whether that risk is biologically or clinically relevant.

\textbf{\emph{Beta diversity analysis.}} Following the approach used by
the Walters and Finucane re-analysis studies, for each dataset we
calculated a Bray-Curtis distance matrix to measure the difference in
the membership and structure of the individuals from each study. We then
used AMOVA to test for significant differences between the structure of
non-obese and obese individuals (Table 1). The Escobar, Goodrich, and
Turnbaugh datasets indicated a significant difference in community
structure (all P\textless{}0.05). Because it was not possible to
ascertain the directionality of the difference in community structure
because the samples are arrayed in a non-dimensional space or perform a
pooled analysis using studies that had non-overlapping 16S rRNA gene
sequence regions, it is unclear whether these differences reflect a
broader, but perhaps small, shift in community structure between
non-obese and obese individuals.

\textbf{\emph{Development of a microbiome-based classifier of obesity.}}
The Walters re-analysis study suggested that it was possible to classify
individuals as being non-obese or obese based on the composition of
their microbiota. We repeated this analysis with additional datasets
using OTU and genus-level phylotype data. For each study we developed a
Random Forest machine learning model to classify individuals. Using
ten-fold cross validation, the cross-validated AUC values for the
OTU-based models varied between 0.52 and 0.69 indicating a relatively
poor ability to classify individuals (Figure 4A). To test models on
other datasets, we trained models using genus-level phylotype data for
each dataset. The cross-validated AUC values for the models applied to
the training datasets varied between 0.51 and 0.65, again indicating a
relatively poor ability to classify individuals from the original
dataset (Figure 4B). For each model we identified the probability where
the sum of the sensitivity and specificity was the highest. We then used
this probability to define a threshold for calculating the accuracy of
the models when applied to the other nine datasets (Figure 5). Although
there was considerable variation in accuracy values for each model, the
median accuracy for each model varied between 0.33 (Turnbaugh) and 0.65
(HMP) (median=0.57). We built similar models using taxonomic
representation based on phylum, class, order, and family assignments and
saw no improvement in the results (Figure S4). We also attempted to
predict individual BMI values as continuous variables based on the
relative abundance of OTUs and genera. The median percent of the
variance explained with the resulting models was 12.9\% for the
OTU-based models and 8.2\% for the genus-based models. When we
considered the number of samples, balance of non-obese and obese
individuals, and region within the 16S rRNA gene for each study it was
not possible to identify factors that predictably affected model
performance. The ability to predict obesity status using relative
abundance data from the communities was only marginally better than
random. These results suggest that given the large diversity of
microbiome compositions it is difficult to identify a taxonomic signal
that can be associated with obesity.

\textbf{\emph{Power and Sample Size Estimate Simulations.}} The
inability to detect a difference between non-obese and obese individuals
could be due to the lack of a true effect or because the study had
insufficient statistical power to detect a difference because of
insufficient sampling, large interpersonal variation, or unbalanced
sampling of non-obese and obese individuals. To assess these factors, we
calculated the power to detect differences of 1, 5, 10, and 15\% in each
of the alpha diversity metrics using the sample sizes used in each of
the studies (Figures 6, S5-S10). Although there is no biological
rationale for these effect sizes, they represent a range that includes
effect sizes that would be generally considered to be biologically
significant. Only the Goodrich study had power greater than 0.80 to
detect a 5\% difference in Shannon diversity and six of the studies had
enough power to detect a 10\% difference (Figure 6A). None of the
studies had sufficient power to detect a 15\% difference between B:F
values (Figure S5). In fact, the maximum power among any of the studies
to detect a 15\% difference in B:F values was 0.25. Among the tests for
relative risk, none of the studies had sufficient power to detect a
Cohen's d of 0.10 and only two studies had sufficient power to detect a
Cohen's d of 0.15. We next estimated how many individuals would need to
have been sampled to have sufficient power to detect the four effect
sizes assuming the observed interpersonal variation from each study and
balanced sampling between the two groups (Figure 6B). To detect a 1, 5,
10, or 15\% difference in Shannon diversity, the median required
sampling effort per group was approximately 3,400, 140, 35, or 16
individuals, respectively. To detect a 1, 5, 10, and 15\% difference in
B:F values, the median required sampling effort per group was
approximately 160,000, 6,300, 1,600, or 700 individuals, respectively.
To detect a 1, 5, 10, and 15\% difference in relative risk values using
Shannon diversity, the median required sampling effort per group was
approximately 39,000, 1,500, 380, or 170 individuals, respectively.
These estimates indicate that most microbiome studies are underpowered
to detect modest effect sizes using either metric. In the case of
obesity, the studies were underpowered to detect the 0.90 to 6\%
difference in diversity that was observed across the studies.

\subsection{Discussion}\label{discussion}

Our meta-analysis helps to provide clarity to the ongoing debate of
whether or not there are specific microbiome-based markers that can be
associated with obesity. We performed an extensive literature review of
the existing studies on the microbiome and obesity and performed a
meta-analysis on the studies that remained based on our inclusion and
exclusion criteria. By statistically pooling the data from ten studies,
we observed significant, but small, relationships between richness,
evenness, and diversity and obesity status as well as the relative risk
of being obese based on these metrics. We also generated Random Forest
machine learning models trained on each dataset and tested on the
remaining datasets. This analysis demonstrated that the ability to
reliably classify individuals as being obese based solely on the
composition of their microbiome was limited. Finally, we assessed the
ability of each study to detect defined differences in alpha diversity
and observed that most studies were underpowered to detect modest effect
sizes. Considering these datasets are among the largest published, it
appears that most human microbiome studies are underpowered to detect
differences in alpha diversity.

Alpha diversity metrics are attractive because they distill a complex
dataset to a single value. For example, Shannon diversity is a measure
of the entropy in a community and integrates richness and evenness
information. Two communities with little taxonomic similarity can have
the same diversity. Among ecologists the relevance of these metrics is
questioned because it is difficult to ascribe a mechanistic
interpretation to their relationship with stability or disease.
Regardless, the concept of a biologically significant effect size needs
to be developed among microbiome researchers. Alternative metrics could
include the ability to detect a defined difference in the relative
abundance of an OTU representing a defined relative abundance. What
makes for a biologically significant difference or relative abundance is
an important point that has yet to be discussed in the microbiome field.
The use of operationally defined effect sizes should be adequate until
it is possible to decide upon an accepted practice.

By selecting a range of possible effect sizes, we were able to
demonstrate that most studies are underpowered to detect modest
differences in alpha diversity metrics and phylum-level relative
abundances. Several factors interact to limit the power of microbiome
studies. There is wide interpersonal variation in the diversity and
structure of the human microbiome. Some factors such as relationship
between subjects could potentially decrease the amount variation (6) and
other factors such as whether one lives in a rural environment could
increase the amount of variation (28). In addition, the common
experimental designs limit their power. As we observed, most of the
studies included in our analysis were unbalanced for the variable that
we were interested in. This was also true of those studies that
originally sought to identify associations with obesity. Even with a
balanced design, we showed that it was necessary to obtain approximately
140 and 6,300 samples per group to detect a 5\% difference in Shannon
diversity or B:F, respectively. It was interesting that these sample
sizes agreed across studies regardless of their sequencing method,
region within the 16S rRNA gene, or subject population (Figure 6). This
suggests that regardless of the treatment or category, these sample
sizes represent a good starting point for subject recruitment when using
stool samples. Unfortunately, few studies have been published with this
level of subject recruitment. This is troubling since the positive
predictive rate of a significant finding in an underpowered study is
small leading to results that cannot be reproduced (29). Future
microbiome studies should articulate the basis for their experimental
design.

Two previous re-analysis studies have stated that there was not a
consistent association between alpha diversity and obesity (8, 9);
however, neither of these studies made an attempt to pool the existing
data together to try and harness the additional power that this would
give and they did not assess whether the studies were sufficiently
powered to detect a difference. Additionally, our analysis used 16S rRNA
gene sequence data from ten studies whereas the Finucane study used 16S
rRNA gene sequence data from three studies (7, 10, 21) and a metagenomic
study (30) and the Walters study used 16S rRNA gene sequence data from
five studies (10, 15, 20, 21, 28); two studies were included in both
analyses (10, 21). Our analysis included four of these studies (10, 15,
20, 21) and excluded three of the studies because they were too small
(7), only utilized metagenomic data (30), or used short single read
Illumina HiSeq data that has a high error rate making it intractable for
\emph{de novo} OTU clustering (28). The additional seven datasets were
published after the two reviews were performed and include datasets with
more samples than were found in the original studies. Our collection of
ten studies allowed us to largely use the same sequence analysis
pipeline for all datasets and relied heavily on the availability of
public data and access to metadata that included variables beyond the
needs of the original study. To execute this analysis, we created an
automated data analysis pipeline, which can be easily updated to add
additional studies as they become available
(\url{https://github.com/SchlossLab/Sze_Obesity_mBio_2016/}). Similarly,
it would be possible to adapt this pipeline to other body sites and
treatment or variables (e.g.~subject's sex or age).

Similar to our study, the Walters study generated Random Forest machine
learning models to differentiate between non-obese and obese individuals
(8). They obtained similar AUC values to our analysis; however, they did
not attempt to test these models on the other studies in their analysis.
When we performed the inter-dataset cross validation the median accuracy
across datasets was only 56.68\% indicating that the models did a poor
job when applied to other datasets. This could be due to differences in
subject populations and methods. Furthermore, others have reported
improved classification at broader taxonomic levels (31); we did not
find this to be the case across the studies in our analysis (Figure S4).
Considering the median AUC for models trained and tested on the same
data with ten-fold cross validation only varied between 0.51 and 0.65
and that there was not a strong signal in the alpha diversity data, we
suspect that there is insufficient signal to reliably classify
individuals to a BMI category based on their microbiota.

Although we failed to find an effect this does not necessarily mean that
there is no role for the microbiome in obesity. There is strong evidence
in murine models of obesity that the microbiome and level of adiposity
can be manipulated via genetic manipulation of the animal and
manipulation of the community through antibiotics or colonizing germ
free mice with diverse fecal material from human donors (5, 10--13).
These studies appear to conflict with the observations using human
subjects. Recalling the large interpersonal variation in the structure
of the microbiome, it is possible that each individual has their own
signatures of obesity. Alternatively, it could be that the involvement
of the microbiome in obesity is not apparent based on the taxonomic
information provided by 16S rRNA gene sequence data. Rather, the
differences could become more apparent at the level of a common set of
gene transcripts or metabolites that can be produced from different
structures of the microbiome.

\subsection{Methods}\label{methods}

\textbf{\emph{In-Depth Overview of Search Strategy.}} The initial search
strategy included looking for all papers that initially fit under the
below NCBI PubMed advanced search criteria: ``(((((((((Bacterial
Microbiome) AND (Obesity or bmi or body mass index or BMI or obesity)
AND''last 10 years``{[}PDat{]} AND Humans{[}Mesh{]})) NOT
review{[}ptyp{]}) AND''last 10 years``{[}PDat{]} AND Humans{[}Mesh{]}))
AND''last 10 years``{[}PDat{]} AND Humans{[}Mesh{]})) AND''last 10
years``{[}PDat{]} AND Humans{[}Mesh{]})''. The terms included in this
criteria were that the manuscript had to have ``Bacterial Microbiome''
and ``Obesity, BMI, bmi, obesity'' in their manuscript criteria, it was
not published more than 10 years ago, they were not review articles, and
it contained research on humans only. This search yielded a total of 187
manuscripts. From the Finucane and Walters re-analysis studies along
with knowledge of other published papers that included BMI information
in their sequence metadata we obtained 10 additional studies. This
brought our total number of records to 197. We browsed abstracts of
these studies and included studies that mentioned stool or feces
examination, did not involve children, was not a clinical trial for
probiotics or other diet related treatments, did not have participants
with inflammatory bowel disease, were in English, and were based on the
analysis of sequence data from the 454 or MiSeq platforms. This
ultimately excluded all but a total of 11 studies. Because we decided
\emph{a priori} to use the standard definition for BMI group
classification one study was excluded because it did not have any
individuals who were obese.

\textbf{\emph{Sequence Analysis Pipeline.}} All sequence data were
publicly available and were downloaded from the NCBI Sequence Read
Archive, the European Nucleotide Archive, or the investigators' personal
website
(\url{https://gordonlab.wustl.edu/TurnbaughSE/_10/_09/STM/_2009.html}).
In total seven studies used 454 (6, 15, 16, 18, 20--22) and three
studies used Illumina sequencing (17, 19, 23). All of these studies used
amplification-based 16S rRNA gene sequencing. Among the studies that
sequenced the 16S rRNA gene, the researchers targeted the V1-V2 (20),
V1-V3 (15, 16, 18), V3-V5 (21, 22), V4 {[}(19); (23); {]}, and V3-4 (17)
regions. For those studies where multiple regions were sequenced, we
selected the region that corresponded to the largest number of subjects
(6, 21). We processed the 16S rRNA gene sequence data using a
standardized mothur pipeline. Briefly, our pipelines attempted to follow
previously recommended approaches for 454 and Illumina sequencing data
(24, 25). All sequences were screened for chimeras using UCHIME and
assigned to operational taxonomic units (OTUs) using the average
neighbor algorithm using a 3\% distance threshold (26, 32). All sequence
processing was performed using mothur (v.1.37.0) (33).

\textbf{\emph{Data Analysis.}} We split the overall meta-analysis into
three general strategies using R (3.3.0). First, we followed the
approach employed by Finucane et al (9) and Walters et al (8) where each
study was re-analyzed separately to identify associations between BMI
and the relative abundance of \emph{Bacteroidetes} and Firmicutes, the
ratio of \emph{Bacteroidetes} and \emph{Firmicutes} relative abundances
(B:F), Shannon diversity, observed richness, and Shannon evenness (34).
After each variable was transformed to fit a normal distribution a
two-tailed t-test was performed for comparison of non-obese and obese
individuals (i.e.~BMI \textgreater{} 35.0). We performed a pooled
analysis on these measured variables using linear random effect models
to correct for study effect to asses differences on the combined dataset
between non-obese and obese groups using the lme4 (v.1.1-12) R package
(35). Next, we compared the community structure from non-obese and obese
individuals using analysis of molecular variance (AMOVA) with
Bray-Curtis distance matrices (36). This analysis was performed using
the vegan (v.2.3-5) R package. For both analyses, the datasets were
rarefied (N=1000) so that each study had the same number of sequences.
Second, for each study we partitioned the subjects into a low or high
group depending on whether their alpha diversity metrics were below or
above the median value for the study. The relative risk (RR) was then
calculated as the ratio of the number of obese individuals in the low
group to the number of obese individuals in the high group. We then
performed a Fisher exact-test to investigate whether the RR was
significantly different from 1.0 within each study and across all of the
studies using the epiR (0.9-77) and metafor (1.9-8) packages. Third, we
used the AUCRF (1.1) R package to generate Random Forest models (37).
For each study we developed models using either OTUs or genus-level
phylotypes. The quality of each model was assessed by measuring the area
under the curve (AUC) of the Receiver Operating Characteristic (ROC)
using ten-fold cross validation. Because the genus-level phylotype
models were developed using a common reference, it was possible to use
one study's model (i.e.~the training set) to classify the samples from
the other studies (i.e.~the testing sets). The optimum threshold for the
training set was set as the probability threshold that had the highest
combined sensitivity and specificity. This threshold was then used to
calculate the accuracy of the model applied to the test studies. To
generate ROC curves and calculate the accuracy of the models we used the
pROC (1.8) R package (38). Finally, we performed power and sample number
simulations for different effect sizes for each study using the pwr
(1.1-3) R package and base R functions. We also calculated the actual
sample size needed based on the effect size of each individual study.

\textbf{\emph{Reproducible methods.}} A detailed and reproducible
description of how the data were processed and analyzed can be found at
\url{https://github.com/SchlossLab/Sze_Obesity_mBio_2016/}.

\subsection{Acknowledgements}\label{acknowledgements}

The authors would like to thank Nielson Baxter and Shawn Whitefield for
their suggestions on the development of the manuscript. We are grateful
to the authors of the studies used in our meta-analysis who have made
their data publicly available or available to us directly. Without their
forethought studies such as this would not be possible. This work was
supported in part by funding from the National Institutes of Health to
PDS (U01AI2425501 and P30DK034933).

\newpage

\textbf{Figure 1: PRISMA flow diagram of total records searched (39).}

\textbf{Figure 2: Individual and combined comparison of obese and
non-obese groups for Shannon diversity (A) and B:F (B).}

\textbf{Figure 3: Meta analysis of the relative risk of obesity based on
Shannon diversity (A) or B:F (B).}

\textbf{Figure 4: ROC curves for each study based on classification of
non-obese or obese groups using OTUs (A) or genus-level classification
(B).}

\textbf{Figure 5: Overall accuracy of each study to predict non-obese
and obese individuals based on that study's Random Forest machine
learning model applied to each of the other studies.}

\textbf{Figure 6: Power (A) and sample size simulations (B) for Shannon
diversity for differentiating between non-obese versus obese for effect
sizes of 1, 5, 10, and 15\%.} Power calculations use the sampling
distribution from the original studies and the sample size estimations
assume an equal amount of sampling from each treatment group.

\newpage

\textbf{Figure S1: Individual and Combined comparison of Obese and
Non-Obese groups Based on Evenness (A), Richness (B), or the Relative
Abundance of \emph{Bacteroidetes} (C) and Firmictues (D).}

\textbf{Figure S2: Funnel plots depicting the general lack of bias in
the selection of datasets included in the analysis.}

\textbf{Figure S3: Meta Analysis of the Relative Risk of Obesity Based
on Evenness (A), Richness (B), or the Relative Abundance of
\emph{Bacteroidetes} (C) and Firmictues (D).}

\textbf{Figure S4: Overall accuracy of each study to predict non-obese
and obese individuals based on that study's Random Forest machine
learning model applied to each of the other studies when trained using
relative abundance of each phylum, class, order, family, or genus.} The
cross-validated AUC values for the training model are provided for each
study and taxonomic level.

\textbf{Figure S5: Power (A) and sample size simulations (B) for B:F for
differentiating between non-obese versus obese for effect sizes of 1, 5,
10, and 15\%.} Power calculations use the sampling distribution from the
original studies and the sample size estimations assume an equal amount
of sampling from each treatment group.

\textbf{Figure S6: Power (A) and sample size simulations (B) for
richness for differentiating between non-obese versus obese for effect
sizes of 1, 5, 10, and 15\%.} Power calculations use the sampling
distribution from the original studies and the sample size estimations
assume an equal amount of sampling from each treatment group.

\textbf{Figure S7: Power (A) and sample size simulations (B) for
evenness for differentiating between non-obese versus obese for effect
sizes of 1, 5, 10, and 15\%.} Power calculations use the sampling
distribution from the original studies and the sample size estimations
assume an equal amount of sampling from each treatment group.

\textbf{Figure S8: Power (A) and sample size simulations (B) for the
relative abundance of \emph{Bacteroidetes} for differentiating between
non-obese versus obese for effect sizes of 1, 5, 10, and 15\%.} Power
calculations use the sampling distribution from the original studies and
the sample size estimations assume an equal amount of sampling from each
treatment group.

\textbf{Figure S9: Power (A) and sample size simulations (B) for the
relative abundance of \emph{Firmicutes} for differentiating between
non-obese versus obese for effect sizes of 1, 5, 10, and 15\%.} Power
calculations use the sampling distribution from the original studies and
the sample size estimations assume an equal amount of sampling from each
treatment group.

\textbf{Figure S10: Power (A) and sample size simulations (B) for
relative risk of obesity based on Shannon diversity.} Power calculations
use the sampling distribution from the original studies and the sample
size estimations assume an equal amount of sampling from each treatment
group.

\newpage

\hyperdef{}{references}{\label{references}}
\subsection*{References}\label{references}
\addcontentsline{toc}{subsection}{References}

\hyperdef{}{ref-ogdenux5fprevalenceux5f2014}{\label{ref-ogdenux5fprevalenceux5f2014}}
1. \textbf{Ogden CL}, \textbf{Carroll MD}, \textbf{Kit BK},
\textbf{Flegal KM}. 2014. Prevalence of childhood and adult obesity in
the United States, 2011-2012. JAMA \textbf{311}:806--814.
doi:\url{http://doi.org/10.1001/jama.2014.732}.

\hyperdef{}{ref-lichtashux5fbodyux5f2013}{\label{ref-lichtashux5fbodyux5f2013}}
2. \textbf{Lichtash CT}, \textbf{Cui J}, \textbf{Guo X}, \textbf{Chen
Y-DI}, \textbf{Hsueh WA}, \textbf{Rotter JI}, \textbf{Goodarzi MO}.
2013. Body adiposity index versus body mass index and other
anthropometric traits as correlates of cardiometabolic risk factors.
PloS One \textbf{8}:e65954.
doi:\url{http://doi.org/10.1371/journal.pone.0065954}.

\hyperdef{}{ref-braheux5fcanux5f2016}{\label{ref-braheux5fcanux5f2016}}
3. \textbf{Brahe LK}, \textbf{Astrup A}, \textbf{Larsen LH}. 2016. Can
We Prevent Obesity-Related Metabolic Diseases by Dietary Modulation of
the Gut Microbiota? Advances in Nutrition (Bethesda, Md)
\textbf{7}:90--101. doi:\url{http://doi.org/10.3945/an.115.010587}.

\hyperdef{}{ref-drorux5fmicrobiotaux5f2016}{\label{ref-drorux5fmicrobiotaux5f2016}}
4. \textbf{Dror T}, \textbf{Dickstein Y}, \textbf{Dubourg G},
\textbf{Paul M}. 2016. Microbiota manipulation for weight change.
Microbial Pathogenesis.
doi:\url{http://doi.org/10.1016/j.micpath.2016.01.002}.

\hyperdef{}{ref-leyux5fobesityux5f2005}{\label{ref-leyux5fobesityux5f2005}}
5. \textbf{Ley RE}, \textbf{BÃ¤ckhed F}, \textbf{Turnbaugh P},
\textbf{Lozupone CA}, \textbf{Knight RD}, \textbf{Gordon JI}. 2005.
Obesity alters gut microbial ecology. Proceedings of the National
Academy of Sciences of the United States of America
\textbf{102}:11070--11075.
doi:\url{http://doi.org/10.1073/pnas.0504978102}.

\hyperdef{}{ref-turnbaughux5fcoreux5f2009}{\label{ref-turnbaughux5fcoreux5f2009}}
6. \textbf{Turnbaugh PJ}, \textbf{Hamady M}, \textbf{Yatsunenko T},
\textbf{Cantarel BL}, \textbf{Duncan A}, \textbf{Ley RE}, \textbf{Sogin
ML}, \textbf{Jones WJ}, \textbf{Roe BA}, \textbf{Affourtit JP},
\textbf{Egholm M}, \textbf{Henrissat B}, \textbf{Heath AC},
\textbf{Knight R}, \textbf{Gordon JI}. 2009. A core gut microbiome in
obese and lean twins. Nature \textbf{457}:480--484.
doi:\url{http://doi.org/10.1038/nature07540}.

\hyperdef{}{ref-leyux5fmicrobialux5f2006}{\label{ref-leyux5fmicrobialux5f2006}}
7. \textbf{Ley RE}, \textbf{Turnbaugh PJ}, \textbf{Klein S},
\textbf{Gordon JI}. 2006. Microbial ecology: Human gut microbes
associated with obesity. Nature \textbf{444}:1022--1023.
doi:\url{http://doi.org/10.1038/4441022a}.

\hyperdef{}{ref-waltersux5fmeta-analysesux5f2014}{\label{ref-waltersux5fmeta-analysesux5f2014}}
8. \textbf{Walters WA}, \textbf{Xu Z}, \textbf{Knight R}. 2014.
Meta-analyses of human gut microbes associated with obesity and IBD.
FEBS letters \textbf{588}:4223--4233.
doi:\url{http://doi.org/10.1016/j.febslet.2014.09.039}.

\hyperdef{}{ref-finucaneux5ftaxonomicux5f2014}{\label{ref-finucaneux5ftaxonomicux5f2014}}
9. \textbf{Finucane MM}, \textbf{Sharpton TJ}, \textbf{Laurent TJ},
\textbf{Pollard KS}. 2014. A taxonomic signature of obesity in the
microbiome? Getting to the guts of the matter. PloS One
\textbf{9}:e84689.
doi:\url{http://doi.org/10.1371/journal.pone.0084689}.

\hyperdef{}{ref-turnbaughux5fobesity-associatedux5f2006}{\label{ref-turnbaughux5fobesity-associatedux5f2006}}
10. \textbf{Turnbaugh PJ}, \textbf{Ley RE}, \textbf{Mahowald MA},
\textbf{Magrini V}, \textbf{Mardis ER}, \textbf{Gordon JI}. 2006. An
obesity-associated gut microbiome with increased capacity for energy
harvest. Nature \textbf{444}:1027--31.
doi:\url{http://doi.org/10.1038/nature05414}.

\hyperdef{}{ref-Koren2012}{\label{ref-Koren2012}}
11. \textbf{Koren O}, \textbf{Goodrich JK}, \textbf{Cullender TC},
\textbf{Spor A}, \textbf{Laitinen K}, \textbf{BÃ¤ckhed HK},
\textbf{Gonzalez A}, \textbf{Werner JJ}, \textbf{Angenent LT},
\textbf{Knight R}, \textbf{BÃ¤ckhed F}, \textbf{Isolauri E},
\textbf{Salminen S}, \textbf{Ley RE}. 2012. Host remodeling of the gut
microbiome and metabolic changes during pregnancy. Cell
\textbf{150}:470--480.
doi:\url{http://doi.org/10.1016/j.cell.2012.07.008}.

\hyperdef{}{ref-Cox2014}{\label{ref-Cox2014}}
12. \textbf{Cox LM}, \textbf{Yamanishi S}, \textbf{Sohn J},
\textbf{Alekseyenko AV}, \textbf{Leung JM}, \textbf{Cho I}, \textbf{Kim
SG}, \textbf{Li H}, \textbf{Gao Z}, \textbf{Mahana D}, \textbf{Rodriguez
JGZ}, \textbf{Rogers AB}, \textbf{Robine N}, \textbf{Loke P},
\textbf{Blaser MJ}. 2014. Altering the intestinal microbiota during a
critical developmental window has lasting metabolic consequences. Cell
\textbf{158}:705--721.
doi:\url{http://doi.org/10.1016/j.cell.2014.05.052}.

\hyperdef{}{ref-Mahana2016}{\label{ref-Mahana2016}}
13. \textbf{Mahana D}, \textbf{Trent CM}, \textbf{Kurtz ZD},
\textbf{Bokulich NA}, \textbf{Battaglia T}, \textbf{Chung J},
\textbf{MÃ¼ller CL}, \textbf{Li H}, \textbf{Bonneau RA}, \textbf{Blaser
MJ}. 2016. Antibiotic perturbation of the murine gut microbiome enhances
the adiposity, insulin resistance, and liver disease associated with
high-fat diet. Genome Medicine \textbf{8}.
doi:\url{http://doi.org/10.1186/s13073-016-0297-9}.

\hyperdef{}{ref-moherux5fpreferredux5f2010}{\label{ref-moherux5fpreferredux5f2010}}
14. \textbf{Moher D}, \textbf{Liberati A}, \textbf{Tetzlaff J},
\textbf{Altman DG}, \textbf{PRISMA Group}. 2010. Preferred reporting
items for systematic reviews and meta-analyses: The PRISMA statement.
International Journal of Surgery (London, England) \textbf{8}:336--341.
doi:\url{http://doi.org/10.1016/j.ijsu.2010.02.007}.

\hyperdef{}{ref-zupancicux5fanalysisux5f2012}{\label{ref-zupancicux5fanalysisux5f2012}}
15. \textbf{Zupancic ML}, \textbf{Cantarel BL}, \textbf{Liu Z},
\textbf{Drabek EF}, \textbf{Ryan KA}, \textbf{Cirimotich S},
\textbf{Jones C}, \textbf{Knight R}, \textbf{Walters WA},
\textbf{Knights D}, \textbf{Mongodin EF}, \textbf{Horenstein RB},
\textbf{Mitchell BD}, \textbf{Steinle N}, \textbf{Snitker S},
\textbf{Shuldiner AR}, \textbf{Fraser CM}. 2012. Analysis of the gut
microbiota in the old order Amish and its relation to the metabolic
syndrome. PloS One \textbf{7}:e43052.
doi:\url{http://doi.org/10.1371/journal.pone.0043052}.

\hyperdef{}{ref-escobarux5fgutux5f2014}{\label{ref-escobarux5fgutux5f2014}}
16. \textbf{Escobar JS}, \textbf{Klotz B}, \textbf{Valdes BE},
\textbf{Agudelo GM}. 2014. The gut microbiota of Colombians differs from
that of Americans, Europeans and Asians. BMC microbiology
\textbf{14}:311. doi:\url{http://doi.org/10.1186/s12866-014-0311-6}.

\hyperdef{}{ref-zeeviux5fpersonalizedux5f2015}{\label{ref-zeeviux5fpersonalizedux5f2015}}
17. \textbf{Zeevi D}, \textbf{Korem T}, \textbf{Zmora N},
\textbf{Israeli D}, \textbf{Rothschild D}, \textbf{Weinberger A},
\textbf{Ben-Yacov O}, \textbf{Lador D}, \textbf{Avnit-Sagi T},
\textbf{Lotan-Pompan M}, \textbf{Suez J}, \textbf{Mahdi JA},
\textbf{Matot E}, \textbf{Malka G}, \textbf{Kosower N}, \textbf{Rein M},
\textbf{Zilberman-Schapira G}, \textbf{DohnalovÃ¡ L},
\textbf{Pevsner-Fischer M}, \textbf{Bikovsky R}, \textbf{Halpern Z},
\textbf{Elinav E}, \textbf{Segal E}. 2015. Personalized Nutrition by
Prediction of Glycemic Responses. Cell \textbf{163}:1079--1094.
doi:\url{http://doi.org/10.1016/j.cell.2015.11.001}.

\hyperdef{}{ref-rossux5f16sux5f2015}{\label{ref-rossux5f16sux5f2015}}
18. \textbf{Ross MC}, \textbf{Muzny DM}, \textbf{McCormick JB},
\textbf{Gibbs RA}, \textbf{Fisher-Hoch SP}, \textbf{Petrosino JF}. 2015.
16S gut community of the Cameron County Hispanic Cohort. Microbiome
\textbf{3}:7. doi:\url{http://doi.org/10.1186/s40168-015-0072-y}.

\hyperdef{}{ref-goodrichux5fhumanux5f2014}{\label{ref-goodrichux5fhumanux5f2014}}
19. \textbf{Goodrich JK}, \textbf{Waters JL}, \textbf{Poole AC},
\textbf{Sutter JL}, \textbf{Koren O}, \textbf{Blekhman R},
\textbf{Beaumont M}, \textbf{Van Treuren W}, \textbf{Knight R},
\textbf{Bell JT}, \textbf{Spector TD}, \textbf{Clark AG}, \textbf{Ley
RE}. 2014. Human genetics shape the gut microbiome. Cell
\textbf{159}:789--799.
doi:\url{http://doi.org/10.1016/j.cell.2014.09.053}.

\hyperdef{}{ref-wuux5flinkingux5f2011}{\label{ref-wuux5flinkingux5f2011}}
20. \textbf{Wu GD}, \textbf{Chen J}, \textbf{Hoffmann C},
\textbf{Bittinger K}, \textbf{Chen Y-Y}, \textbf{Keilbaugh SA},
\textbf{Bewtra M}, \textbf{Knights D}, \textbf{Walters WA},
\textbf{Knight R}, \textbf{Sinha R}, \textbf{Gilroy E}, \textbf{Gupta
K}, \textbf{Baldassano R}, \textbf{Nessel L}, \textbf{Li H},
\textbf{Bushman FD}, \textbf{Lewis JD}. 2011. Linking long-term dietary
patterns with gut microbial enterotypes. Science (New York, NY)
\textbf{334}:105--108. doi:\url{http://doi.org/10.1126/science.1208344}.

\hyperdef{}{ref-humanux5fmicrobiomeux5fprojectux5fconsortiumux5fstructureux5f2012}{\label{ref-humanux5fmicrobiomeux5fprojectux5fconsortiumux5fstructureux5f2012}}
21. \textbf{Human Microbiome Project Consortium}. 2012. Structure,
function and diversity of the healthy human microbiome. Nature
\textbf{486}:207--214. doi:\url{http://doi.org/10.1038/nature11234}.

\hyperdef{}{ref-schubertux5fmicrobiomeux5f2014}{\label{ref-schubertux5fmicrobiomeux5f2014}}
22. \textbf{Schubert AM}, \textbf{Rogers MAM}, \textbf{Ring C},
\textbf{Mogle J}, \textbf{Petrosino JP}, \textbf{Young VB},
\textbf{Aronoff DM}, \textbf{Schloss PD}. 2014. Microbiome data
distinguish patients with Clostridium difficile infection and non-C.
difficile-associated diarrhea from healthy controls. mBio
\textbf{5}:e01021--01014.
doi:\url{http://doi.org/10.1128/mBio.01021-14}.

\hyperdef{}{ref-baxterux5fmicrobiota-basedux5f2016}{\label{ref-baxterux5fmicrobiota-basedux5f2016}}
23. \textbf{Baxter NT}, \textbf{Ruffin MT}, \textbf{Rogers MAM},
\textbf{Schloss PD}. 2016. Microbiota-based model improves the
sensitivity of fecal immunochemical test for detecting colonic lesions.
Genome Medicine \textbf{8}:37.
doi:\url{http://doi.org/10.1186/s13073-016-0290-3}.

\hyperdef{}{ref-Kozichux5f2013}{\label{ref-Kozichux5f2013}}
24. \textbf{Kozich JJ}, \textbf{Westcott SL}, \textbf{Baxter NT},
\textbf{Highlander SK}, \textbf{Schloss PD}. 2013. Development of a
dual-index sequencing strategy and curation pipeline for analyzing
amplicon sequence data on the MiSeq Illumina sequencing platform.
Applied and environmental microbiology \textbf{79}:5112--5120.

\hyperdef{}{ref-Schloss2011}{\label{ref-Schloss2011}}
25. \textbf{Schloss PD}, \textbf{Gevers D}, \textbf{Westcott SL}. 2011.
Reducing the effects of PCR amplification and sequencing artifacts on
16S rRNA-based studies. PLoS ONE \textbf{6}:e27310.
doi:\url{http://doi.org/10.1371/journal.pone.0027310}.

\hyperdef{}{ref-Westcott2015}{\label{ref-Westcott2015}}
26. \textbf{Westcott SL}, \textbf{Schloss PD}. 2015. De novo clustering
methods outperform reference-based methods for assigning 16S rRNA gene
sequences to operational taxonomic units. PeerJ \textbf{3}:e1487.
doi:\url{http://doi.org/10.7717/peerj.1487}.

\hyperdef{}{ref-Wang2007}{\label{ref-Wang2007}}
27. \textbf{Wang Q}, \textbf{Garrity GM}, \textbf{Tiedje JM},
\textbf{Cole JR}. 2007. Naive bayesian classifier for rapid assignment
of rRNA sequences into the new bacterial taxonomy. Applied and
Environmental Microbiology \textbf{73}:5261--5267.
doi:\url{http://doi.org/10.1128/aem.00062-07}.

\hyperdef{}{ref-yatsunenkoux5fhumanux5f2012}{\label{ref-yatsunenkoux5fhumanux5f2012}}
28. \textbf{Yatsunenko T}, \textbf{Rey FE}, \textbf{Manary MJ},
\textbf{Trehan I}, \textbf{Dominguez-Bello MG}, \textbf{Contreras M},
\textbf{Magris M}, \textbf{Hidalgo G}, \textbf{Baldassano RN},
\textbf{Anokhin AP}, \textbf{Heath AC}, \textbf{Warner B},
\textbf{Reeder J}, \textbf{Kuczynski J}, \textbf{Caporaso JG},
\textbf{Lozupone CA}, \textbf{Lauber C}, \textbf{Clemente JC},
\textbf{Knights D}, \textbf{Knight R}, \textbf{Gordon JI}. 2012. Human
gut microbiome viewed across age and geography. Nature
\textbf{486}:222--227. doi:\url{http://doi.org/10.1038/nature11053}.

\hyperdef{}{ref-Ioannidis2005}{\label{ref-Ioannidis2005}}
29. \textbf{Ioannidis JPA}. 2005. Why most published research findings
are false. PLoS Med \textbf{2}:e124.
doi:\url{http://doi.org/10.1371/journal.pmed.0020124}.

\hyperdef{}{ref-Qin2010}{\label{ref-Qin2010}}
30. \textbf{Qin J}, \textbf{Li R}, \textbf{Raes J}, \textbf{Arumugam M},
\textbf{Burgdorf KS}, \textbf{Manichanh C}, \textbf{Nielsen T},
\textbf{Pons N}, \textbf{Levenez F}, \textbf{Yamada T}, \textbf{Mende
DR}, \textbf{Li J}, \textbf{Xu J}, \textbf{Li S}, \textbf{Li D},
\textbf{Cao J}, \textbf{Wang B}, \textbf{Liang H}, \textbf{Zheng H},
\textbf{Xie Y}, \textbf{Tap J}, \textbf{Lepage P}, \textbf{Bertalan M},
\textbf{Batto J-M}, \textbf{Hansen T}, \textbf{Paslier DL},
\textbf{Linneberg A}, \textbf{Nielsen HB}, \textbf{Pelletier E},
\textbf{Renault P}, \textbf{Sicheritz-Ponten T}, \textbf{Turner K},
\textbf{Zhu H}, \textbf{Yu C}, \textbf{Li S}, \textbf{Jian M},
\textbf{Zhou Y}, \textbf{Li Y}, \textbf{Zhang X}, \textbf{Li S},
\textbf{Qin N}, \textbf{Yang H}, \textbf{Wang J}, \textbf{Brunak S},
\textbf{DorÃ© J}, \textbf{Guarner F}, \textbf{Kristiansen K},
\textbf{Pedersen O}, \textbf{Parkhill J}, \textbf{Weissenbach J},
\textbf{Antolin M}, \textbf{Artiguenave F}, \textbf{Blottiere H},
\textbf{Borruel N}, \textbf{Bruls T}, \textbf{Casellas F},
\textbf{Chervaux C}, \textbf{Cultrone A}, \textbf{Delorme C},
\textbf{Denariaz G}, \textbf{Dervyn R}, \textbf{Forte M}, \textbf{Friss
C}, \textbf{Guchte M van de}, \textbf{Guedon E}, \textbf{Haimet F},
\textbf{Jamet A}, \textbf{Juste C}, \textbf{Kaci G}, \textbf{Kleerebezem
M}, \textbf{Knol J}, \textbf{Kristensen M}, \textbf{Layec S},
\textbf{Roux KL}, \textbf{Leclerc M}, \textbf{Maguin E}, \textbf{Minardi
RM}, \textbf{Oozeer R}, \textbf{Rescigno M}, \textbf{Sanchez N},
\textbf{Tims S}, \textbf{Torrejon T}, \textbf{Varela E}, \textbf{Vos W
de}, \textbf{Winogradsky Y}, \textbf{Zoetendal E}, \textbf{Bork P},
\textbf{Ehrlich SD}, \textbf{Wang J}. 2010. A human gut microbial gene
catalogue established by metagenomic sequencing. Nature
\textbf{464}:59--65. doi:\url{http://doi.org/10.1038/nature08821}.

\hyperdef{}{ref-Sun2010}{\label{ref-Sun2010}}
31. \textbf{Sun Y}, \textbf{Cai Y}, \textbf{Mai V}, \textbf{Farmerie W},
\textbf{Yu F}, \textbf{Li J}, \textbf{Goodison S}. 2010. Advanced
computational algorithms for microbial community analysis using massive
16S rRNA sequence data. Nucleic Acids Research \textbf{38}:e205--e205.
doi:\url{http://doi.org/10.1093/nar/gkq872}.

\hyperdef{}{ref-Edgar2011}{\label{ref-Edgar2011}}
32. \textbf{Edgar RC}, \textbf{Haas BJ}, \textbf{Clemente JC},
\textbf{Quince C}, \textbf{Knight R}. 2011. UCHIME improves sensitivity
and speed of chimera detection. Bioinformatics \textbf{27}:2194--2200.
doi:\url{http://doi.org/10.1093/bioinformatics/btr381}.

\hyperdef{}{ref-mothur}{\label{ref-mothur}}
33. \textbf{Schloss PD}, \textbf{Westcott SL}, \textbf{Ryabin T},
\textbf{Hall JR}, \textbf{Hartmann M}, \textbf{Hollister EB},
\textbf{Lesniewski RA}, \textbf{Oakley BB}, \textbf{Parks DH},
\textbf{Robinson CJ}, \textbf{others}. 2009. Introducing mothur:
open-source, platform-independent, community-supported software for
describing and comparing microbial communities. Applied and
environmental microbiology \textbf{75}:7537--7541.

\hyperdef{}{ref-Magurran2003}{\label{ref-Magurran2003}}
34. \textbf{Magurran AE}. 2003. Measuring biological diversity 264.

\hyperdef{}{ref-lme4ux5f2015}{\label{ref-lme4ux5f2015}}
35. \textbf{Bates D}, \textbf{MÃ¤chler M}, \textbf{Bolker B},
\textbf{Walker S}. 2015. Fitting linear mixed-effects models using lme4.
Journal of Statistical Software \textbf{67}:1--48.
doi:\url{http://doi.org/10.18637/jss.v067.i01}.

\hyperdef{}{ref-Anderson2001}{\label{ref-Anderson2001}}
36. \textbf{Anderson MJ}. 2001. A new method for non-parametric
multivariate analysis of variance. Austral Ecology \textbf{26}:32--46.
doi:\url{http://doi.org/10.1111/j.1442-9993.2001.01070.pp.x}.

\hyperdef{}{ref-Calle2011}{\label{ref-Calle2011}}
37. \textbf{Calle ML}, \textbf{Urrea V}, \textbf{Boulesteix A-L},
\textbf{Malats N}. 2011. AUC-RF: A new strategy for genomic profiling
with random forest. Human Heredity \textbf{72}:121--132.
doi:\url{http://doi.org/10.1159/000330778}.

\hyperdef{}{ref-pROCux5f2011}{\label{ref-pROCux5f2011}}
38. \textbf{Robin X}, \textbf{Turck N}, \textbf{Hainard A},
\textbf{Tiberti N}, \textbf{Lisacek F}, \textbf{Sanchez J-C},
\textbf{MÃ¼ller M}. 2011. PROC: An open-source package for r and s+ to
analyze and compare rOC curves. BMC Bioinformatics \textbf{12}:77.

\hyperdef{}{ref-Moher2009}{\label{ref-Moher2009}}
39. \textbf{Moher D}, \textbf{Liberati A}, \textbf{Tetzlaff J},
\textbf{Altman DG}. 2009. Preferred reporting items for systematic
reviews and meta-analyses: The PRISMA statement. PLoS Med
\textbf{6}:e1000097.
doi:\url{http://doi.org/10.1371/journal.pmed.1000097}.

\end{document}
