---
title: "Looking for a Signal in the Noise: Revisiting Obesity and the Microbiome"
bibliography: references.bib
output:
  pdf_document:
    keep_tex: true
    includes:
      in_header: header.tex
csl: msystems.csl
fontsize: 11pt
geometry: margin=1.0in
---

# Lack of power and small effect size confounds the ability to differentiate obese and non-obese individuals using gut microbiome data


```{r knitr_settings, eval=TRUE, echo=FALSE, cache=FALSE}
opts_chunk$set("tidy" = TRUE)
opts_chunk$set("echo" = FALSE)
opts_chunk$set("eval" = TRUE)
opts_chunk$set("warning" = FALSE)
opts_chunk$set("cache" = FALSE)

inline_hook <- function(x){
	print(x)
	if(is.numeric(x)){
		if(abs(x - round(x)) < .Machine$double.eps^0.5){
			paste(format(x,big.mark=',', digits=0, scientific=FALSE))
		} else {
			paste(format(x,big.mark=',', digits=2, nsmall=2, scientific=FALSE))
		}

	} else {
    	paste(x)      
	}
}
knitr::knit_hooks$set(inline=inline_hook)
```
\begin{center}
\vspace{25mm}

**Running Title:** The Human Microbiome and Obesity
<br>
Marc A Sze and Patrick D Schloss${^\dagger}$
<br>
**Contributions:** Both authors contributed to the planning, design, execution, interpretation, and writing of the analyses.


\vspace{30mm}

$\dagger$ To whom correspondence should be addressed: pschloss@umich.edu

Department of Microbiology and Immunology, University of Michigan, Ann Arbor, MI
\end{center}


\newpage
\linenumbers


```{r datasets}
alpha <- read.table(file="data/process/alpha_composite.summary", header=T)
shannon_effect_size <- 100*abs(1-alpha[alpha$metric=="shannon","obese"]/alpha[alpha$metric=="shannon","non_obese"])
beta <- read.table(file="data/process/beta_tests.summary", header=T)

genus <- read.table(file="data/process/random_forest.genus.summary", header=T)
otu <- read.table(file="data/process/random_forest.otu.summary", header=T)

test <- read.table(file="data/process/random_forest.genus.train_test", header=T)
test_summary <- aggregate(accuracy~train, data=test, median)

power <- read.table(file="data/process/alpha_power.predicted", header=T)
well_powered <- sum(power[power$metric=="shannon" &
 												power$effect_size==0.05, "power"] > 0.8)



number_string <- c("one", "two", "three", "four", "five", "six", "seven", "eight", "nine")
```

## Abstract
Two recent studies have re-analyzed published data and found that when datasets are analyzed independently there was limited support the widely accepted hypothesis that changes in the microbiome are associated with obesity. This hypothesis was reconsidered by increasing the number of data sets and pooling the results across the individual datasets. The Preferred Reporting Items for Systematic Reviews and Meta-Analyses (PRISMA) guidelines were applied to identify 10 studies for an updated and more synthetic analysis. Alpha-diversity metrics and the relative risk of obesity based on those metrics were used to identify a limited number of significant associations with obesity; however, when the results of the studies were pooled using a random effects model significant associations were observed between Shannon diversity, number of observed OTUs, and Shannon evenness and obesity status They were not observed for the ratio of *Bacteroidetes* and *Firmicutes* or their individual relative abundances. Although these tests yielded small P-values, the difference between the Shannon diversity index of obese and non-obese individuals was `r shannon_effect_size`%. A power analysis demonstrated that `r number_string[well_powered]` of the studies had sufficient power to detect a 5% difference in diversity. When models trained on one dataset were then tested using the other 9 datasets, the median accuracy varied between `r range(test_summary$accuracy)[1]` and `r range(test_summary$accuracy)[2]` (median=`r median(test_summary$accuracy)`). Although there is statistical support for a relationship between the microbial communities found in human feces and obesity status, this association is relatively weak and its detection is confounded by large interpersonal variation and insufficient sample sizes.


## Importance
As interest in the human microbiome grows there is an increasing number of studies that can be used to test numerous hypotheses across human populations. The hypothesis that variation in the gut microbiota can explain or be used to predict obesity status has received considerable attention and is frequently mentioned as an example for the role of the microbiome in human health. Here we assess this hypothesis using ten independent studies and find that although there is an association, it is smaller than can be detected by most microbiome studies. Furthermore, we directly tested the ability to predict obesity status based on the composition of an individual's microbiome and find that the median classification accuracy is between `r range(test_summary$accuracy)[1]` and `r range(test_summary$accuracy)[2]`%. This type of analysis can be used to design future studies and expanded to explore other hypotheses.



\newpage

## Introduction

Obesity is a growing health concern with approximately 20% of the youth (aged 2-19) in the United States classified as either overweight or obese [@ogden_prevalence_2014]. This number increases to approximately 35% in adults (aged 20 or older) and these statistics have seen little change since 2003 [@ogden_prevalence_2014]. Traditionally the body mass index (BMI) has been used as the traditional method of classifying individuals as non-obese or obese [@lichtash_body_2013]. Recently, there has been increased interest in the role of the microbiome in modulating obesity [@brahe_can_2016; @dror_microbiota_2016]. If the microbiome does affect obesity status, then manipulating the microbiome could have a significant role in the future treatment of obesity and in helping to stem the current epidemic.

There have been several studies that report observing a link between the composition of microbiome and obesity in animal models and in humans. The first such study used genetically obese mice and observed the ratio of the relative abundances of Bacteroidetes to Firmicutes (B:F) was lower in obese mice than lean mice [@ley_obesity_2005]. Translation of this result to humans by the same researchers did not observe this effect, but did find that obese individuals had a lower diversity than lean individuals [@turnbaugh_core_2009]. They also showed that the relative abundance of Bacteroidetes and Firmicutes increased and decreased, respectively, as obese individuals lost weight while on a fat or carbohydrate restricted diet [@ley_microbial_2006]. Two re-analysis studies interrogated previously published microbiome and obesity data and concluded that the previously reported differences in community diversity and B:F among non-obese and obese individuals could not be generalized [@walters_meta-analyses_2014; @finucane_taxonomic_2014]. Regardless of the results using human populations, mechanistic studies using animal models that were manipulated with antibiotics or colonization with varied communities appears to support the association since different communities yield variation in animal weight [REFERENCES]. The purported association between the differences in the microbiome and obesity have been widely repeated with little attention given to the lack of a clear signal in human cohort studies.

The recent publication of additional studies that collected BMI data for each subject as well as other studies that were not included in the earlier re-analysis studies offered the opportunity to revisit the question relating the structure of the human microbiome to obesity [@ross_16s_2015; @zupancic_analysis_2012; @nam_comparative_2011; @arumugam_enterotypes_2011; @goodrich_human_2014; @wu_linking_2011; @escobar_gut_2014; @baxter_microbiota-based_2016; @schubert_microbiome_2014]. One critique of the prior re-analysis studies is that the authors did not aggregate the results across studies to increase the effective sample size. It is possible that there were small associations within each study that were not statistically significant because the individual studies lacked sufficient power. Alternatively, diversity metrics may mask the appropriate signal and it is necessary to measure the association at the level of microbial populations. Walters et al. [-@walters_meta-analyses_2014] demonstrated that Random Forest machine learning models were capable of predicting obesity status within a single cohort, but did not attempt to test the models on other cohorts. The purpose of this study was to perform a meta-analysis of the association between differences in the microbiome and obesity status by analyzing and applying a more systematic and synthetic approach than was used previously.


## Methods

***Literature Review and Study Inclusion***
We followed the Preferred Reporting Items for Systematic Reviews and Meta-Analyses (PRISMA) guidelines to identify studies to include in our meta-analysis [@moher_preferred_2010]. A detailed description of our selection process and the exact search terms are provided in the Supplement and in figure 1. Briefly, we searched PubMed for original research studies that involved studying obesity and the human microbiome. The initial search yielded 187 studies. We identified ten additional studies that were not designed to explicitly test for an association between the microbiome and obesity. We then manually curated the 196 studies to select those studies that included BMI and sequence data. This yielded 10 eligible studies. An additional study was removed from our analysis because no individuals in the study had a BMI over 30. Among the final 10 studies, 3 were from identified from our PubMed search [@turnbaugh_core_2009; @zupancic_analysis_2012; @escobar_gut_2014], 5 were originally identified from the 9 studies that did not explicitly investigate obesity but included BMI data [@zeevi_personalized_2015; @ross_16s_2015; @goodrich_human_2014; @wu_linking_2011; @human_microbiome_project_consortium_structure_2012], and two datasets were used [@schubert_microbiome_2014; @baxter_microbiota-based_2016] because these publications did not specifically look for any metabolic or obesity conditions but had control populations and enabled us to help mitigate against publication biases associated with the bacterial microbiome and obesity. The ten studies are summarized in Tables 1 and 2.


***Sequence Analysis Pipeline***
All sequence data were publicly available and were downloaded from the NCBI Sequence Read Archive, the European Nucleotide Archive, or the investigators' personal website (https://gordonlab.wustl.edu/TurnbaughSE\_10\_09/STM\_2009.html). In total seven studies used 454 [@turnbaugh_core_2009; @zupancic_analysis_2012; @ross_16s_2015; @escobar_gut_2014; @wu_linking_2011; @human_microbiome_project_consortium_structure_2012; @schubert_microbiome_2014] and three studies used Illumina sequencing [@goodrich_human_2014; @zeevi_personalized_2015; @baxter_microbiota-based_2016]. All of these studies used amplification-based 16S rRNA gene sequencing. Among the studies that sequenced the 16S rRNA gene, the researchers targeted the V1-V2  [@wu_linking_2011], V1-V3 [@ross_16s_2015; @escobar_gut_2014; @zupancic_analysis_2012], V3-V5 [@human_microbiome_project_consortium_structure_2012; @schubert_microbiome_2014], V4 [@goodrich_human_2014; @baxter_microbiota-based_2016; ], and V3-4 [@zeevi_personalized_2015] regions. For those studies where multiple regions were sequenced, we selected the region that corresponded to the largest number of subjects [@turnbaugh_core_2009; @human_microbiome_project_consortium_structure_2012]. We processed the 16S rRNA gene sequence data using a standardized mothur pipeline. Briefly, our pipelines attempted to follow previously recommended approaches for 454 and Illumina sequencing data [@Kozich_2013; @Schloss2011]. All sequences were screened for chimeras using UCHIME and assigned to operational taxonomic units (OTUs) using the average neighbor algorithm using a 3% distance threshold [@Edgar2011; @Westcott2015]. All sequence processing was performed using mothur (v.1.37.0) [@mothur].

***Data Analysis***
We split the overall meta-analysis into three general strategies using R (`r paste0(R.Version()$major,'.', R.Version()$minor)`). First, we followed the approach employed by Finucane et al [-@finucane_taxonomic_2014] and Walters et al [-@walters_meta-analyses_2014] where each study was re-analyzed separately to identify associations between BMI and the relative abundance of Bacteroidetes and Firmicutes, the ratio of Bacteroidetes and Firmicutes relative abundances (B:F), Shannon diversity, observed richness, and Shannon evenness. After each variable was transformed to fit a normal distribution a two-tailed t-test was performed for comparison of non-obese and obese individuals (i.e. BMI > 35.0). We performed a pooled analysis on these measured variables using linear random effect models to correct for study effect to asses differences on the combined dataset between non-obese and obese groups using the lme4 (v.`r packageDescription("lme4")$Version`) R package. Next, we compared the community structure from non-obese and obese individuals using PERMANOVA analysis of Bray-Curtis distance matrices. This analysis was performed using the vegan (v.`r packageDescription("vegan")$Version`) R package. For both analyses, the datasets were rarefied (N=1000) so that each study within a study had the same number of sequences. Second, for each study we partitioned the subjects into a low or high group depending on whether their alpha-diversity metrics were below or above the median value for the study. The relative risk (RR) was then calculated as the ratio of the number of obese individuals in the low group to the number of obese individuals in the high group. We then performed a Fisher exact-test to investigate whether the RR was significantly different from 1.0 within each study and across all of the studies using the epiR (`r packageDescription("epiR")$Version`) and metafor (`r packageDescription("metafor")$Version`) packages. Third, we used the AUCRF (`r packageDescription("AUCRF")$Version`) R package to generate Random Forest models. For each study we developed models using either OTUs or genus-level phylotypes. The quality of each model was assessed by measuring the area under the curve (AUC) of the Receive Operating Characteristic (ROC) using ten-fold cross validation. Because the genus-level phylotype models were developed using a common reference, it was possible to use one study's model (i.e. the training set) to classify the samples from the other studies (i.e. the testing sets). The optimum threshold for the training set was set as the probability threshold that had the highest combined sensitivity and specificity. This threshold was then used to calculate the accuracy of the model applied to the test studies. To generate Receiver Operator Characteristic (ROC) curves and calculate the accuracy of the models we used the pROC (`r packageDescription("pROC")$Version`) R package. Finally, we performed power and sample number simulations for different effect sizes for each study using the pwr (`r packageDescription("pwr")$Version`) R package and base R functions. We also calculated the actual sample size needed based on the effect size of each individual study.

***Reproducible methods***
A detailed and reproducible description of how the data were processed and analyzed can be found at https://github.com/SchlossLab/Sze_ObesityMicrobiomeMeta_mBio_2016/.



## Results

***Separate and Pooled Analysis***
We used the categorical variable of obese or non-obese to analyze each data set separately for significant correlations to Bacteroidetes, Firmicutes, Bacteroidetes/Firmicutes (B:F) ratio, Shannon diversity, OTU richness, evenness, and the Bray-Curtis distance matrix which was a similar approach taken by two previous reviews [@walters_meta-analyses_2014; @finucane_taxonomic_2014]. Using a P-value cutoff for significance as less than 0.05 we found a total of `r length(rownames(alpha_tests.summary[which(alpha_tests.summary$p.value < 0.05), ]))` significant results; two for Shannon diversity [Figure 2A], three for observed richness [Figure S1], one for evenness [Figure S2], and one for Bacteroidetes [Figure S3]. There were no significant results for Firmicutes [Figure S4] or the B:F ratio [Figure 2B]. Using the linear mixed-effect model to analyze the data pooled together while also controling study effects we found that for Shannon diversity  there was a significant difference between the obese and non-obese groups (P-value = `r format(round(alpha_composite.summary['shannon', 'composite_p'], 4), nsmall=4)`) [Figure 2A]. Additionally,  there was also a significant difference in obseverd richness (P-value = `r format(round(alpha_composite.summary['sobs', 'composite_p'], 9), nsmall=9)`) [Figure S1].  However, we found that the there was no significant difference in evenness (P-value = `r format(round(alpha_composite.summary['shannoneven', 'composite_p'], 2), nsmall=2)`), Bacteroidetes (P-value = `r format(round(alpha_composite.summary['bacteroidetes', 'composite_p'], 2), nsmall=2)`), Firmicutes (P-value = `r format(round(alpha_composite.summary['firmicutes', 'composite_p'], 2), nsmall=2)`), and B:F ratio (P-value = `r format(round(alpha_composite.summary['bf_ratio', 'composite_p'], 2), nsmall=2)`)  [Figure 2B & S2-S4] using this approach, between obese and non-obese individuals.



***Classical Pooled Meta-analysis***

We next moved on to the pooled meta-analysis. For the Shannon diversity RR, all studies but Wu, Goodrich, and Turnbaugh had those with a lower Shannon diversity then the median as being higher risk for obesity [Figure 3A]. The pooled analysis showed an increased RR of `r format(round(RR_composite['shannon', 'rr'],2), nsmall=2)` (CI `r format(round(RR_composite['shannon', 'ci_lb'],2), nsmall=2)` - `r format(round(RR_composite['shannon', 'ci_ub'], 2), nsmall=2)`) for obesity in the lower then median Shannon diversity group that was significant (P-value = `r format(round(RR_composite['shannon', 'p_value']), nsmall=2)`). In contrast, when we investigated the B:F ratio RR  the pooled result from all studies showed that there was no significant difference between obese and non-obese individuals (P-value = `r format(round(RR_composite['bf_ratio', 'p_value']), nsmall=2)`) [Figure 3B].  The pooled RR was `r format(round(RR_composite['bf_ratio', 'rr'],2), nsmall=2)` (CI `r format(round(RR_composite['bf_ratio', 'ci_lb'], 2), nsmall=2)` - `r format(round(RR_composite['bf_ratio', 'ci_ub'], 2), nsmall=2)`).  The differences in RR that was observed also do not seem to be related to the variable region chosen [Figure 3].  

We also used a funnel plot to asses the bias of the 9 data sets used in this meta-analysis. From the graph we can observe that there does not seem to be any significant  bias [Figure S5]. Both funnel plots have almost equal number of studies scattered on either side of the estimated true value.



***Random Forest Classification and Predicition Accuracy***

Next we investigated how well the OTUs in each study were able to classify obese and non-obese individuals using the random forest algorithm. We found that the average Out Of Bag (OOB) Area Under the Curve (AUC) was `r format(round(mean(rand_forest.genus.summary$auc), 2), nsmall=2)` +/- `r format(round(sd(rand_forest.genus.summary$auc), 2), nsmall=2)` for the genus level analysis and `r format(round(mean(rand_forest.otu.summary$auc), 2), nsmall=2)` +/- `r format(round(sd(rand_forest.otu.summary$auc), 2), nsmall=2)` for the OTU level analysis.  We found that the range for the total number of variables used for the classification using genus ranged from `r min(rand_forest.genus.summary$k_opt)` (HMP and Wu, et al. study) to `r max(rand_forest.genus.summary$k_opt)` (Goodrich, et al. Study).  For the OTU level the number of variables ranged from `r min(rand_forest.otu.summary$k_opt)` (Wu, et al. study) to `r max(rand_forest.otu.summary$k_opt)` (Goodrich, et al. study). The total number of trees and node size was standardized to 500 and 20 respectively for each data set. Overall there was no significant difference between the ROC curves of each data set (P-value > 0.05) regardless if it was using genus or OTU level classification [Figure 4].  Overall most studies performed poorly when used on other data sets for classification [Figure 5] with no significant difference between studies for their mean accuracy for prediction of non-obese and obese indivdiuals (P-value > 0.05).    



***Power and Sample Size Estimates and Simulations***

For Shannon Diversity the measured difference between both obese and non-obese groups was relatively small [Figure 6A] compared to the B:F ratio [Figure 7A] differences.  When simulating different effect size differences of 1%, 5%, 10%, and 15% we found that many studies would be powered sufficiently to find a 15% difference in Shannon diversity (predicted percent power = `r format(round(mean(alpha_power.predicted[which(alpha_power.predicted$metric == 'shannon' & alpha_power.predicted$effect_size == 0.15), ]$power*100), 2), nsmall=2)`, SD = `r format(round(sd(alpha_power.predicted[which(alpha_power.predicted$metric == 'shannon' & alpha_power.predicted$effect_size == 0.15), ]$power*100), 2), nsmall=2)`) [Figure 6A].  This predicted power was also similar for evenness (`r format(round(mean(alpha_power.predicted[which(alpha_power.predicted$metric == 'shannoneven' & alpha_power.predicted$effect_size == 0.15), ]$power*100), 2), nsmall=2)`, SD = `r format(round(sd(alpha_power.predicted[which(alpha_power.predicted$metric == 'shannoneven' & alpha_power.predicted$effect_size == 0.15), ]$power*100), 2), nsmall=2)`).   However, this was not the case for the B:F ratio (`r format(round(mean(alpha_power.predicted[which(alpha_power.predicted$metric == 'bf_ratio' & alpha_power.predicted$effect_size == 0.15), ]$power*100), 2), nsmall=2)` +/- `r format(round(sd(alpha_power.predicted[which(alpha_power.predicted$metric == 'bf_ratio' & alpha_power.predicted$effect_size == 0.15), ]$power*100), 2), nsmall=2)`%) [Figure 7A] or the other measures that were tested (observed richness, Bacteroidetes, Firmicutes) where even a 15% difference between non-obese and obese groups only resulted in a relatively poor average power between studies.   Looking at the sample sizes showed that on average to detect a 5% difference in Shannon diversity at 80% power between non-obese and obese individuals a total n of `r format(round(mean(alpha_power.predicted[which(alpha_power.predicted$metric == 'shannon' & alpha_power.predicted$effect_size == 0.05), ]$balanced_n*100), 0), nsmall=0)` (SD = `r format(round(sd(alpha_power.predicted[which(alpha_power.predicted$metric == 'shannon' & alpha_power.predicted$effect_size == 0.05), ]$balanced_n*100), 0), nsmall=0)`) would be needed for each group [Figure 6B].  In contrast in order to detect a 5% difference with 80% power in the B:F ratio a sample size of `r format(round(mean(alpha_power.predicted[which(alpha_power.predicted$metric == 'bf_ratio' & alpha_power.predicted$effect_size == 0.05), ]$balanced_n*100), 0), nsmall=0)` (SD = `r format(round(sd(alpha_power.predicted[which(alpha_power.predicted$metric == 'bf_ratio' & alpha_power.predicted$effect_size == 0.05), ]$balanced_n*100), 0), nsmall=0)`) per group [Figure 7B].  A complete breakdown of the power simulations and sample size estimates for the other variables investigated and RR can be found in the online supplement [Figure S6-S11].   

We also investigated if the combined total n for the linear mixed effect model was sufficiently powered.  Using a small (0.2), medium (0.5), and large (0.8) cohen value we estimated the power of the combined data sets as well as the average cohen value from across studies for Shannon Diversity (0.06) and BF ratio (0.01).  For the small effect for combined data power calculation a value of `r format(round(pwr.t2n.test(n1=length(which(alpha.data.ALL$obese == FALSE)), n2=length(which(alpha.data.ALL$obese == TRUE)), d=0.2, sig.level = 0.05, alternative="two.sided")$power*100,2), nsmall=2)`% was found.  For the medium effect for combined data power calculation a value of `r format(round(pwr.t2n.test(n1=length(which(alpha.data.ALL$obese == FALSE)), n2=length(which(alpha.data.ALL$obese == TRUE)), d=0.5, sig.level = 0.05, alternative="two.sided")$power*100,2), nsmall=2)`% was found.  For the large effect for combined data power calculation a value of
`r format(round(pwr.t2n.test(n1=length(which(alpha.data.ALL$obese == FALSE)), n2=length(which(alpha.data.ALL$obese == TRUE)), d=0.8, sig.level = 0.05, alternative="two.sided")$power*100,2), nsmall=2)`% was found.  The average actual cohens d value for Shannon Diversity for all `r length(rownames(beta_tests.summary))` studies of `r format(round(shannond, 2), nsmall=2)` gave a power of `r format(round(pwr.t2n.test(n1=length(which(alpha.data.ALL$obese == FALSE)), n2=length(which(alpha.data.ALL$obese == TRUE)), d=shannond, sig.level = 0.05, alternative="two.sided")$power*100,2), nsmall=2)`%.  Finally, the average actual cohens d value for B:F ratio for all `r length(rownames(beta_tests.summary))` studies of `r format(round(bf_ratiod, 2), nsmall=2)` yieled a power of `r format(round(pwr.t2n.test(n1=length(which(alpha.data.ALL$obese == FALSE)), n2=length(which(alpha.data.ALL$obese == TRUE)), d=bf_ratiod, sig.level = 0.05, alternative="two.sided")$power*100,2), nsmall=2)`%.  


## Discussion:

Our meta-analysis helps to provide a little more clarity to the on going debate of whether or not there are specific microbiome variables that are influenced by obesity. Two previous reviews [@walters_meta-analyses_2014; @finucane_taxonomic_2014] have stated that the data do not support that there is a microbiome difference between non-obese and obese individuals. However, neither of these studies made an attempt to pool the existing data together to try and harness the additional power that this would give. Here we perform an extensive literature review of the existing studies on the microbiome and obesity and perform a meta-analysis on the studies that remained based on our inclusion and exclusion criteria. We analyzed the data using three different approaches and also performed post-hoc power and sample size calculations and simulations. The first approach uses the method taken by the previous reviews [@walters_meta-analyses_2014; @finucane_taxonomic_2014], the second approach takes a meta-analysis approach using RR representation by forest plots and funnel plots, the final approach uses linear mixed effect models to adjust for study effects and invetigates both predicted power and sample number needed for the observed effect sizes.

Overall, the first approach in this study by and large agrees with the prevous two reviews[@walters_meta-analyses_2014; @finucane_taxonomic_2014], in that for the majority of studies, no significant correlation between the microbiome and obesityy could be observed. Further, even though random forest could identify specific variables with similar taxonomic classifications that resulted in good ROC curves [Figure 1] these variables varied widely by both their actual sequence and directionality.  Similarily, for the B:F ratio we found that the other two approaches, that used the pooled data approach, were also not able to identify a correlation between measures of the microbiome and obesity [Figure 3 & 4]. For these approaches both the B:F ratio failed to have a significant correlation to obesity.  However, a small but significant differences was observed between non-obese and obese individauls  for Shannon diversity [Figure 3 & 4].  In order to also assess the amount of our results that could be due to insufficient power we performed power and sample size analysis.  Overall, we found that most studies for Shannon Diversity would be powered to find a 15% difference, however, the vast majority of reported effect sizes between studies ranged from `r format(round(min(alpha_effectSize[, 'shannon']), 2), nsmall=2)`-`r format(round(max(alpha_effectSize[, 'shannon']), 2), nsmall=2)` percent.  Although the effect sizes for the BF ratio were larger than that for Shannon Diversity [Figure 5A & 6A] the estimated power remained low.  Additionaly, we would also need a relatively large sample size to be able to detect the actual differences in each study for both Shannon Diversity and B:F ratio [Figure 5B & 6B].  Taken together we believe that this analysis builds on the previous two re-anlaysis and provides additional information as to whether there is a difference between non-obese and obese human individuals with respect to both Shannon diversity and the B:F ratio.    


We do acknowledge that this study does have a few limitations that are worth mentioning. First,  it has been documented that in a number of groups BMI is not a good classification of obesity [@who_expert_consultation_appropriate_2004; @rahman_accuracy_2010]. Of particular note, the asian population has a separate set of guidelines for obesity [@who_expert_consultation_appropriate_2004]. Although the asain population in the studies analyzed are very small [Table 1] it is possible that this may contribute to some of the noise in the data. However, it is important to keep in mind that for these measures, specifically the RR, there does not seem to be any study bias due to studies having more ethnic diversity [Figure 4]. Additionally, we are also limited in our BMI classifications by what the previous studies published and performed. In general, not enough studies have been done to investigate the microbiome in the context of other markers for obesity or on different ethnicities to be able to perform the analysis that we did. Thus as more specific studies are published that address ethnicity and studies on the microbiome that use better markers of obesity are completed it is entirely possible that the noise associated with these measurements will decrease.    

Second, although we make attempts to correct for study effects in our pooled analysis  so that we can compare the microbiome across studies it is possible that the different variable regions sampled, sequencing machines used, and the quality of the output data could have an impact on the overall conclusions reached.  To help mitigate some of these problems we analyzed each data set separately and processed all samples through a uniform pipeline in an attempt to mitigate a portion of the variation that might occur due to sequence processing.  It should also be noted that we also do not see any specific bias in the forest plots [Figure 3] based on hypervariable region.  Further many of the studies show very similar actual effect sizes for Shannon Diversity difference between non-obese and obese individuals.  However, even with all these precautions and safety checks we do realize that it may not be possible to compare these studies together and that further studies need to be completed to verify the observations made in this report.  

Third, although we may not have detected a signal between non-obese and obese individuals our results indicate that we may not have been sufficiently powered to do so. However, taking both the sample size calculations and power calculations together would indicate that if there is a true difference to be found between non-obese and obese individuals this difference would be very small.  Additionally, due to the very small effect size potentially the more important question that does need to be raised is whether or not this difference is truly biologically meaningful.

Fourth, although the ROC curves for each study were relatively good there was very little actual overlap between studies based on sequence similarity and directionality.  Further none of the selected variables from each indivdiual study did a respectable job of classification when applied to the other data sets.  This would suggest that within each indivdiual study classification of obese and non-obese is relatively strong but not generalizable across studies.  This might indicate an ancestry, ethnicity, or a specific environmental effect such as diet difference between study populations that might cause this dissimilarity.  Additionally, an alternative explanation that may be more reasonable is that the random forest model is overfit for some of the studies examined which explains why the variables selected are less generalizable across studies.  In actuality it may be a combination of all of the factors mentioned as to why generalizability is problematic.      

Although we failed to find an effect it is not realistic to necessarily state that there is no microbiome impact on obesity.  There is a large body of very strong evidence in mice on the effect the microbiome has on obesity [@turnbaugh_effect_2009; @koren_host_2012].  It is not possible to discount these findings and it may be the case that in humans there is just too many other variables and factors that effect the microbiome which makes the signal too hard to detect in a cross sectional manner.  Some well designed studies in humans and diet, in the past, have shown that the noise inherent to microbiome studies can be overcome with very careful attention made to controlling the study conditions [@ley_microbial_2006].   

Overall, this study provides a reasonable follow up to the two previous re-analysis studies to assess the association between the microbiome and obesity. In doing so we believe that we have provided evidence that if there is an effect size present it is small and will take many more studies and samples to be analyzed to confirm a difference between non-obese and obese individuals.


## Acknowledgements

The authors would like to thank Nielson Baxter and Shawn Whitefield for their suggestions on the manuscript. Finally, we would like to thank all the authors who have made their data publicly available.  Without this initial forethought studies such as these would not be possible.  



\newpage

## Tables

**Table 1. Summary Demographics of Individuals used in the Meta-analysis.**
```{r Table_1, echo=FALSE, message=FALSE, warning=FALSE}

table1 <- read.table("../results/tables/table1.demographics", header = T)

colnames(table1) <- c("Study", "Age (SD)", "Sex (F|M)", "European Ancestry (%)", "Obese (No|Yes)", "BMI (SD)", "Min BMI","Max BMI")

kable(table1, format = "markdown", align = 'c', padding = 2)


```

* For Turnbaugh only BMI group information was provided and only an age range was given.

\newpage

**Table 2. Summary of Important Study Characteristics.**

```{r Table_2, echo=FALSE, message=FALSE, warning=FALSE}

Study <- as.character(table1$Study)
#New Order
  #baxter, escobar, hmp, ross, schubert, turnbaugh, wu, zupancic, goodrich, zeevi

Type <- c("Observational", "Observational", "Observational", "Observational", "Observational", "Observational", "Observational", "Observational", "Observational", "Observational")

Population <- c("Adults without Cancer from Texas, Ontario, Michigan, Massachusetts", "Healthy adults from Columbia", "Healthy Adults from Pennsylvania", "Hispanic adults with and without type 2 diabetes from Texas
", "Nondiarrheal indivdiuals without infection history", "Adult twins and mothers from Missouri", "Healthy Adults from Texas and Missouri", "Amish Adults with and without metabolic syndrome from Pennsylvania", "Adult twins and mothers from the United Kingdom", "Healthy Adults from Tel Aviv")



Published <- c("Yes", "Yes", "Yes", "Yes", "Yes", "Yes", "Yes","Yes", "Yes", "Yes")

Outcomes <- c("Did not investigate control populations for relationship with BMI", "Firmicutes less abundant at higher BMI", "No significant correlations", "Mentions obesity, no specific correlations made though", "Did not investigate control populations for relationship with BMI", "Unweighted UniFrac Correlation with BMI", "Decrease diversity, Bacteroidetes, and Actinobacteria correlated with increased BMI ", "Bacteroidetes/Firmicutes ratio adjusted BMI correlation", "Christensenellaceae higher in low BMI individuals. No other correlations found.", "Found associations with postprandial (post meal) glycemic responses, which correlates with BMI, and Proteobacteria, Actinobacteria, and Enterobacteriaceae")

table2 <- cbind(Study, Type, Population, Published, Outcomes)

colnames(table2) <- c("Study", "Study Type", "Population", "Published", "Outcomes Reported")


kable(table2, format = "markdown", align = 'c', padding = 2)
```






*******
##### Figures:


**Figure 1: PRISMA Flow Diagram of Total Records Searched.**  This was adapted from:  Moher D, Liberati A, Tetzlaff J, Altman DG, The PRISMA Group (2009). Preferred Reporting Items for Systematic Reviews and Meta-Analyses: The PRISMA Statement. PLoS Med 6(7): e1000097. doi:10.1371/journal.pmed1000097.



**Figure 2: Individual and Combined comparison of Obese and Non-Obese groups for Shannon Diversity and B:F Ratio.** **A)** For Shannon diversity only Baxter (P-value = `r format(round(alpha_tests.summary[which(alpha_tests.summary$dataset == 'baxter' & alpha_tests.summary$metric == 'shannon'), 'p.value'], 2), nsmall = 2)` and Goodrich (P-value = `r format(round(alpha_tests.summary[which(alpha_tests.summary$dataset == 'goodrich' & alpha_tests.summary$metric == 'shannon'), 'p.value'], 5), nsmall = 5)`) studies indivdiually were signficantly different for obese and non-obese groups.  When the data sets were combined and study effect was corrected for,  there was a significant difference between Non-Obese and Obese groups (P-value = `r format(round(alpha_composite.summary['shannon', 'composite_p'], 5), nsmall = 5)`). **B)** For the B:F ratio there was no individual study that was significant nor was their a difference when the data was combined and study effect corrected for between Non-Obese and Obese groups (P > 0.05).    



**Figure 3: Meta Analysis of the Relative Risk of Obesity Based on Shannon Diversity or Bacteroidetes/Firmicutes Ratio.**  Groups were divided for each study on high and low Shannon Diversity or Bacteroidetes/Firmicutes (B:F) ratio groups based on the median for that study. Red represents studies that used the 16S rRNA V4 region, yellow represents studies that used a V1-V3 approach, blue represents studies that used either a V1-V2 or just a V2 approach, green represents studies that used a V3-V4 region, and orange represents studies that used the V3-V5 region.  **A)** The overall pooled relative risk for Shannon Diversity was `r format(round(RR_composite['shannon', 'rr'],2), nsmall=2)` with a P-value of `r format(round(RR_composite['shannon', 'p_value']), nsmall=2)`.  **B)** The overall pooled relative risk for the B:F ratio was  `r format(round(RR_composite['bf_ratio', 'rr'],2), nsmall=2)` with a P-value of `r format(round(RR_composite['bf_ratio', 'p_value'], 2), nsmall=2)`.




**Figure 4: ROC Curves for Each Study based on Classification of Non-Obese or Obese Groups at the Genus or OTU level.**  **A)** No significant difference was found between the different ROC curves based on Study for the genus level.  The overall AUCs ranged from `r round(min(rand_forest.genus.summary$auc), 2)` to `r round(max(rand_forest.genus.summary$auc), 2)` with a mean of `r round(mean(rand_forest.genus.summary$auc), 2)`. **B)** Similarly no difference was found between the ROC curves based on Study at the OTU level.  The overall mean of the AUCs were slightly higher overall than the genus level `r round(mean(rand_forest.otu.summary$auc), 2)` but the range was greater going from `r round(min(rand_forest.otu.summary$auc), 2)` to `r round(max(rand_forest.otu.summary$auc), 2)`.


**Figure 5: Overall Accuracy of Each Study to Predict Non-Obese and Obese Individuals based on that Study's Classification Variables** Overall there is no significant difference between the studes in mean accuracy.  Additionally, the accuracy for each study group to predict Non-Obese and Obese individuals was relatively low.



**Figure 6: Summary of Power and Sample Size Simulations for Shannon Diversity for Non-obese versus Obese**  P-values listed in the legend represent the outcome of a wilcoxson rank sum test between non-obese and obese for each specific data set.  **A)** The dotted lines represent the actual effect size for each specific study.  **B)** The dotted lines represent the needed n to achieve an 80% power with the actual study effect size.




**Figure 7. Summary of Power and Sample Size Simulations for Bacteroidetes/Firmicutes Ratio for Non-obese versus Obese**  P-values listed in the legend represent the outcome of a wilcoxson rank sum test between non-obese and obese for each specific data set.  **A)** The dotted lines represent the actual effect size for each specific study.  **B)** The dotted lines represent the needed n to achieve an 80% power with the actual study effect size.

*******
##### References:
