---
title: "Bacterial Microbiome and Obesity Meta-analysis Supplemental"
author: "Marc Sze"
date: "January 26, 2016"
output:
  html_document:
    keep_md: yes
bibliography: metaAnalysis.obesity.microbiome.bib
---
<br><br>

```{r Loading_data_tables, echo=FALSE, message=FALSE, warning=FALSE}

OverallPValue <- read.csv("results/tables/denovoOverallPTable.csv")
ShannonRRTable <- read.csv("results/tables/denovoShannonRRTable.csv")
BFRatioRRTable <- read.csv("results/tables/denovoBFRatioRRTable.csv")

```



###In-Depth Overview of Search Strategy:

The initial search strategy included looking for all papers that initially fit under the below NCBI PubMed advanced search criteria.  The terms included in this criteria were that the manuscript had to have "Bacterial Microbiome" and "Obesity, BMI, bmi, obesity" in their manuscript criteria, it was not published more than 10 years ago, they were not review articles, and it contained research on humans only.  The below formula when put into PubMed should recapitulate our initial search on the website.

```{bash, eval=FALSE}
(((((((((Bacterial Microbiome) AND (Obesity or bmi or body mass index or BMI or obesity) AND "last 10 years"[PDat] AND Humans[Mesh])) NOT review[ptyp]) AND "last 10 years"[PDat] AND Humans[Mesh])) AND "last 10 years"[PDat] AND Humans[Mesh])) AND "last 10 years"[PDat] AND Humans[Mesh])
```

This search yielded a total of 187 manuscripts.  From two previous other reviews of obesity and the bacterial microbiome along with knowledge of two other published papers that investigated obesity but were missed by the database search we obtained a total of 7 more articles.  We also had access to normal healthy individuals from an unpublished dataset.  This brought our total number of records to 196.  

From this total we browsed abstracts for mention of stool or feces examination, that did not involve children, was not a clinical trial for probiotics or other diet related treatments, did not only have participants with inflammatory bowel disease, the articles were in English, did not only use PCR, qPCR, or RT-PCR only for their analysis, and sequencing that used only clone libraries.  This ultimately excluded all but a total of 11 studies.

From this total of 12 studies the full text was reviewed for whether or not sequencing data was publicly available, BMI information (either categorical or continuous) was available in a supplement or, if it was not available, whether authors upon contact were willing to share this information or direct us to repositories that stored this specific information.  One study was excluded [@yatsunenko_human_2012] because it contained children and their sequencing of the 16S rRNA gene involved amplicons of only 100bp in length.  They also did not have obesity as part of their results in the actual published manuscript.  The second study was excluded because they did not have BMI information available and when contacted the authors never returned any correspondence [@zeevi_personalized_2015].  A third study was excluded since it did not use 16S rRNA gene sequencing for their bacterial microbiome analysis [@arumugam_enterotypes_2011]  

Once these 3 studies were excluded there was a total of 9 studies in the qualitative synthesis of the analysis.  Because we decided a prioi to use the standard definition for BMI group classification one study from this ten did not have any individuals who were obese by this criteria [@nam_comparative_2011] and was excluded from the final quantitative synthesis and analysis.  

*Inclusion Criteria:*

* Contains mention of Bacterial Microbiome and Obesity
* BMI, bmi, or obesity could be referenced instead of Obesity
* Not published more than 10 years ago
* Research on Humans only
* At least one specific result examining obesity and a bacterial microbiome measure
* Participants did not have Inflammatory Bowel Disease or Cancer
* Greater than 100bp single or dual end reads for 16S Sequencing
* DNA obtained from stool or feces

*Exclusion Criteria:*

* PCR, qPCR, metagenomic sequencing, or RT-PCR used as main analysis Tool
* TRFLP or clone sequencing used to asses the bacterial community
* Utilization of 100bp or less single end reads for sequencing
* Sequencing Data not publicly available for download
* BMI not available and authors do not return correspondence
* Samples were not stool or feces
* Study contained children
* Study was a review





###Supplemental Results:


**Table S1. Summary of P-values for Measurements of Interest for each Individual Study for Obese versus Normal**

```{r Table_S1, echo=FALSE, message=FALSE, warning=FALSE}

library(knitr)

Study <- as.character(OverallPValue$X)
Bacteroidetes <- format(round(OverallPValue$Bacteroidetes, 2), nsmall=2)
Firmicutes <- format(round(OverallPValue$Firmicutes, 2), nsmall=2)
BFRatio <- format(round(OverallPValue$BFRatio, 2), nsmall=2)
Shannon <- format(round(OverallPValue$Shannon, 2), nsmall=2)
OTURich <- format(round(OverallPValue$OTURich, 2), nsmall=2)
Evenness <- format(round(OverallPValue$Evenness, 2), nsmall=2)
BrayC <- format(round(OverallPValue$BrayC, 2), nsmall=2)

tableS1 <- cbind(Study, Bacteroidetes, Firmicutes, BFRatio, Shannon, OTURich, Evenness, BrayC)
colnames(tableS1) <- c("Study", "Bacteroidetes", "Firmicutes", "B/F Ratio", "Shannon Diversity", "OTU Richness", "Evenness", "Bray Curtis")

kable(tableS1, format = "markdown", align = 'c', padding = 2)

```



![Figure S1](results/figures/suppFigClostridiales.png)\


**Figure S1: Boxplots of All Clostridiales OTUs Important for Non-Obese and Obese Classification.** A total of 6/8 studies had at least 1 OTU that was classified to Clostridiales.  However, overall there was a wide range in direction both between studies and within studies.


![Figure S2](results/figures/suppFigLachno.png)\


**Figure S2: Boxplots of All Lachnospiraceae OTUs Important for Non-Obese and Obese Classification.** A total of 6/8 studies had at least 1 OTU that was classified to Lachnospiraceae.  However, overall there was a wide range in direction both between studies and within studies.


![Figure S3](results/figures/suppFigRumi.png)\

**Figure S3: Boxplots of All Ruminococcaceae OTUs Important for Non-Obese and Obese Classification.** A total of 7/8 studies had at least 1 OTU that was classified to Ruminococcaceae  However, overall there was a wide range in direction both between studies and within studies.


**Table 2. Summary of BLAST alignment of Representative OTU for Clostridiales, Lachnospiraceae, and Ruminococcaceae classification OTUs**  

```{r Table_S2, echo=FALSE, message=FALSE, warning=FALSE}

Study <- c("Wu", "Turnbaugh", "Ross", "Escobar", "Zupancic", "Baxter", "Goodrich", "HMP")
VariableRegion <- c("V1-V2", "V2", "V1-V3", "V1-V3", "V1-V3", "V4", "V4", "V3-V5")
ClosMostPrev <- c("OTU191:3 Clostridium", "None", "OTU102:8 Ruminococcus", "None", "OTU322:3 Parvimonas | OTU62:3 Parvimonas | OTU256:4 Clostridium", "OTU479:4 Anaerostipes", "OTU287:5 Clostridium | OTU662:8 Clostridium | OTU145:6 Clostridium | OTU244:3 Clostridium | OTU442:4 Clostridium", "OTU293:7 Clostridium")
ClostHighMatch <- c("Anaerovorax odorimutans(81|95)", "None", "Ruminococcus champanellensis(100|100)", "None", "OTU322:Eubacterium hallii(92|100) | OTU62:Eubacterium hallii(92|99) | OTU256:Romboutsia lituseburensis(92|99)", "Anaerosporbacter mobilis(100|94)", "OTU287:Clostridium fimetarium(100|94) |  OTU662:Ruminiclostridium thermocellum(100|91) | OTU145:Ruminococcus lactaris(100|100) | OTU244:Anaerosporobacter mobilis(100|94) | OTU442:Clostridium propionicum(100|92)", "Anaerobacterium chartisolvens (100|91)")
LachMostPrev <- c("OTU75:6 Clostridium | OTU229:5 Peptostreptococcus | OTU5: 4 Blautia", "OTU318:", "OTU177:3 Roseburia", "OTU67:6 Eubacterium | OTU74:5 Blautia | OTU80:4 Blautia", "None", "OTU36:8 Clostridium", "OTU180:4 Clostridium | OTU75: 5 Clostridium | OTU122:5 Roseburia | OTU92:8 Clostridium | OTU673:7 Clostridium", "None")
LachHighMatch <- c("Clostridium bolteae(100|96)", "", "Clostridium Spp(100|98)", "OTU67:Dorea formicigenerans(100|96) | OTU74:Blautia luti(100|96) | OTU80:Blautia producta(100|93)", "None", "Lachnobacterium bovis(100|96)", "OTU180:Hungatella effluvii(100|95) | OTU75:Ruminococcus gnavus(100|100) | OTU122:Eubacterium ramulus(100|96) | OTU92:Fusicatenibacter saccharivorans(100|94) | OTU673:Bacteroides xylanolyticus(92|97) | OTU523:Eubacterium rectale(100|97)", "None")
RumiMostPrev <- c("OTU123:4 Clostridium", "OTU30:", "OTU22:3 Oscillibacter", "None", "OTU398:3 Eubacterium | OTU850:3 Ethanoligenens | OTU817:8 Ruminococcus", "OTU320:4 Clostridium", "OTU68:5 Ruminococcus | OTU255:3 Blautia | OTU1145:3 Ethanoligenens | OTU558:8 Ruminococcus | OTU297:3 Oscillibacter | OTU6:4 Clostridium | OTU132:2 Ruminoclostridium", "OTU252:10 Clostridium")
RumiHighMatch <- c("Anaerotruncus colihominis(85|95)", "", "Oscillibacter Spp(100|97)", "None", "OTU398:Oscillibacter valericigene(77|96) | OTU850:Papillibacter cinnamivorans(90|91) | OTU817:Ruminococcus callidus(91|95)", "Sporobacter termitidis(100|94)", "OTU68:Acetanaerobacterium elongatum(100|89) | OTU255:Pseudoflaconifractor capillosus(100|91) | OTU1145:Subdoligranulum variable(100|95) | OTU558:Ruminococcus champanellensis(92|98) | OTU297:Flavonifractor plautii(100|95) | OTU6:Oscillibacter valericigenes(100|93) | OTU132:Intestinimonas butyriciproducens(100|92)", "Clostridium Spp(100|97)")



tableS2 <- cbind(Study, VariableRegion, ClosMostPrev, ClostHighMatch, LachMostPrev, LachHighMatch, RumiMostPrev, RumiHighMatch)
colnames(tableS2) <- c("Study", "Hypervariable Region", "Clostridiales OTU Most Prevalent Match", "Clostridiales OTU Highest Match (Query Cover|Identity)", "Lachnospiraceae OTU Most Prevalent Match", "Lachnospiraceae OTU Highest Match (Query Cover|Identity)", "Ruminococcaceae OTU Most Prevalent Match", "Ruminococcaceae OTU Highest Match (Query Cover|Identity)")

kable(tableS2, format = "markdown", align = 'c', padding = 2)
```


**Table S3. Top Sequence Similarity by Variable Region Based on the Representative OTU for Clostridiales, Lachnospiraceae, and Ruminococcaceae**  

```{r Table_S3, echo=FALSE, message=FALSE, warning=FALSE}

VariableRegion <- c("V1-V2 vs V2", "V1-V3", "V4")

TopHitClos <- c("None", "50.3|39.1", "89.4|2.8")
AverageClos <- c("None", "48.0|42.4", "87.7|2.4")
TopHitLachno <- c("TBC", "40.1|47.9", "92.9|0.8")
AverageLachno <- c("TBC", "35.9|52.5", "85.8|5.6")
TopHitRumi <- c("TBC", "30.4|68.4", "86.6|0.8")
AverageRumi <- c("TBC", "29.2|69.1", "81.6|6.6")

tableS3 <- cbind(VariableRegion, TopHitClos, AverageClos, TopHitLachno, AverageLachno, TopHitRumi, AverageRumi)
colnames(tableS3) <- c("Variable Region", "Top Similarity Colstridiales (Id|Gaps)", "Average Similarity Colstridiales (Id|Gaps)", "Top Similarity Lachnospiraceae (Id|Gaps)", "Average Similarity Lachnospiraceae (Id|Gaps)", "Top Similarity Ruminococcaceae (Id|Gaps)", "Average Similarity Ruminococcaceae (Id|Gaps)")

kable(tableS3, format = "markdown", align = 'c', padding = 2)

```


```{r Figure_S4, echo=FALSE, message=FALSE, warning=FALSE, fig.path="results/figures/"}

library(metafor)

# Shannon Diversity Data for Funnel Plot
StudyNo <- as.numeric(ShannonRRTable$X)
Study <- as.character(ShannonRRTable$Study)
Year <- c("2015", "2015", "2014", "2014", "2012", "2011", "2012", "2009")
Total <- c("n=172", "n=63", "n=507", "n=30", "n=200", "n=256", "n=63", "n=146")
tpos <- as.numeric(ShannonRRTable$tposH) #obese in low diversity group
tneg <- as.numeric(ShannonRRTable$tnegH) #nonobese in low diversity group
cpos <- as.numeric(ShannonRRTable$cposH) #obese in high diversity group
cneg <- as.numeric(ShannonRRTable$cnegH) #nonobese in high diversity group
RR <- as.numeric(ShannonRRTable$RRH)
low <- as.numeric(ShannonRRTable$lowH)
high <- as.numeric(ShannonRRTable$highH)

numeric.data <-as.data.frame(cbind(tpos, tneg, cpos, cneg))
raw.overall <- as.data.frame(cbind(StudyNo, Study,Year, Total, numeric.data))

dat <- escalc(measure = "RR", ai = tpos, bi = tneg, ci = cpos, di = cneg, data=raw.overall, append=TRUE)
k <- length(dat$StudyNo)
dat.fm <- data.frame(study=factor(rep(1:k, each = 4)))
dat.fm$grp <- factor(rep(c("T", "T", "C", "C"), k), levels = c("T", "C"))
dat.fm$out <- factor(rep(c("+", "-", "+", "-"), k), levels = c("+", "-"))
dat.fm$freq <- with(raw.overall, c(rbind(tpos, tneg, cpos, cneg)))

#Used to generate the RE Model estimate based on all the studies
res <- rma(ai=tpos, bi=tneg, ci=cpos, di=cneg, data=raw.overall, measure="RR", slab=paste(Study, Year, Total, sep=", "), method="REML")


# B/F ratio data used to generate Funnel Plot
StudyNo <- as.numeric(BFRatioRRTable$X)
Study <- as.character(BFRatioRRTable$Study)
Year <- c("2015", "2015", "2014", "2014", "2012", "2011", "2012", "2009")
Total <- c("n=172", "n=63", "n=507", "n=30", "n=200", "n=256", "n=63", "n=146")
tpos <- as.numeric(BFRatioRRTable$tposBF) #obese in low diversity group
tneg <- as.numeric(BFRatioRRTable$tnegBF) #nonobese in low diversity group
cpos <- as.numeric(BFRatioRRTable$cposBF) #obese in high diversity group
cneg <- as.numeric(BFRatioRRTable$cnegBF) #nonobese in high diversity group
RR <- as.numeric(BFRatioRRTable$RRBF)
low <- as.numeric(BFRatioRRTable$lowBF)
high <- as.numeric(BFRatioRRTable$highBF)

numeric.data <-as.data.frame(cbind(tpos, tneg, cpos, cneg))
raw.overall <- as.data.frame(cbind(StudyNo, Study,Year, Total, numeric.data))

dat <- escalc(measure = "RR", ai = tpos, bi = tneg, ci = cpos, di = cneg, data=raw.overall, append=TRUE)
k <- length(dat$StudyNo)
dat.fm <- data.frame(study=factor(rep(1:k, each = 4)))
dat.fm$grp <- factor(rep(c("T", "T", "C", "C"), k), levels = c("T", "C"))
dat.fm$out <- factor(rep(c("+", "-", "+", "-"), k), levels = c("+", "-"))
dat.fm$freq <- with(raw.overall, c(rbind(tpos, tneg, cpos, cneg)))

#Used to generate the RE Model estimate based on all the studies
res2 <- rma(ai=tpos, bi=tneg, ci=cpos, di=cneg, data=raw.overall, measure="RR", slab=paste(Study, Year, Total, sep=", "), method="REML")


# Generate Funnel Plots for supplemental
par(mfrow=c(1,2))
funnel(res, main="A")
funnel(res2, main="B")
```

**Figure S4: Funnel Plot of the Shannon Diversity and Bacteroidetes/Firmicutes Ratio Relative Risk.**  **A)** Overall there does not seem to be any bias associated with the studies selected for the Shannon Diversity analysis with studies falling on either side of the predicted value and those with smaller total n falling further away from this. **B)** For the Bacteroidetes/Firmicutes ratio there does not seem to be any associated bias. The overall pattern is similar to that observed for the Shannon Diversity analysis.


```{r Figure_S5, echo=FALSE, message=FALSE, warning=FALSE, fig.path="results/figures/"}

source("code/MakeFiguresCodeSupplemental.R")
library(gridExtra)

grid.arrange(s2, sn2, nrow=2)




```

**Figure S5. Summary of Power and Sample Size Simulations for OTU Richness (S) for Non-obese versus Obese**  P-values listed in the legend represent the outcome of a wilcoxson rank sum test between non-obese and obese for each specific data set.  **A)** The dotted lines represent the actual effect size for each specific study.  **B)** The dotted lines represent the needed n to achieve an 80% power with the actual study effect size.

```{r Figure_S6, echo=FALSE, message=FALSE, warning=FALSE, fig.path="results/figures/"}

grid.arrange(j2, jn2, nrow=2)


```

**Figure S6. Summary of Power and Sample Size Simulations for Evenness (J) for Non-obese versus Obese**  P-values listed in the legend represent the outcome of a wilcoxson rank sum test between non-obese and obese for each specific data set.  **A)** The dotted lines represent the actual effect size for each specific study.  **B)** The dotted lines represent the needed n to achieve an 80% power with the actual study effect size.


```{r Figure_S7, echo=FALSE, message=FALSE, warning=FALSE, fig.path="results/figures/"}

grid.arrange(b2, bn2, nrow=2)
```

**Figure S7. Summary of Power and Sample Size Simulations for Bacteroidetes for Non-obese versus Obese**  P-values listed in the legend represent the outcome of a wilcoxson rank sum test between non-obese and obese for each specific data set.  **A)** The dotted lines represent the actual effect size for each specific study.  **B)** The dotted lines represent the needed n to achieve an 80% power with the actual study effect size.

```{r Figure_S8, echo=FALSE, message=FALSE, warning=FALSE, fig.path="results/figures/"}

grid.arrange(f2, fn2, nrow=2)
```

**Figure S8. Summary of Power and Sample Size Simulations for Firmicutes for Non-obese versus Obese**  P-values listed in the legend represent the outcome of a wilcoxson rank sum test between non-obese and obese for each specific data set.  **A)** The dotted lines represent the actual effect size for each specific study.  **B)** The dotted lines represent the needed n to achieve an 80% power with the actual study effect size.


```{r Figure_S9, echo=FALSE, message=FALSE, warning=FALSE, fig.path="results/figures/"}

grid.arrange(RRH2, HRRn2, nrow=2)
```

**Figure S9. Summary of Power and Sample Size Simulations for Relative Risk  for Obesity based on Shannon Diversity**  P-values listed in the legend represent the outcome of a wilcoxson rank sum test between non-obese and obese for each specific data set.  **A)** The dotted lines represent the actual effect size for each specific study.  **B)** The dotted lines represent the needed n to achieve an 80% power with the actual study effect size.

```{r Figure_S10, echo=FALSE, message=FALSE, warning=FALSE, fig.path="results/figures/"}

grid.arrange(RRBF2, BFRRn2, nrow=2)
```

**Figure S10. Summary of Power and Sample Size Simulations for Relative Risk  for Obesity based on B/F Ratio**  P-values listed in the legend represent the outcome of a wilcoxson rank sum test between non-obese and obese for each specific data set.  **A)** The dotted lines represent the actual effect size for each specific study.  **B)** The dotted lines represent the needed n to achieve an 80% power with the actual study effect size.


*******
##### References:
